<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>exploreBiocUniverse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="exploreBiocUniverse_files/libs/clipboard/clipboard.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/popper.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/anchor.min.js"></script>
<link href="exploreBiocUniverse_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="exploreBiocUniverse_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-b24c5370db45008d7a0c55e687ef2d35.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="exploreBiocUniverse_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="exploreBiocUniverse_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">exploreBiocUniverse</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="note" class="level1">
<h1>Note</h1>
<p>This was an initial exploration, which is now old. Please see <a href="comparison.qmd">comparison</a>.</p>
</section>
<section id="comparing-bioconductor-packages-in-r-universe-and-the-bbs" class="level1">
<h1>Comparing Bioconductor packages in R-Universe and the BBS</h1>
<p>Skip to <a href="#packages-not-in-agreement">Packages not in agreements</a> or go to <a href="windows.qmd">windows</a>.</p>
<section id="r-universe" class="level2">
<h2 class="anchored" data-anchor-id="r-universe">R Universe</h2>
<p>We can use the <a href="https://github.com/ropensci/universe/">universe</a> package to explore the R-Universe (RU) API. (We might need to increase the <code>max.print</code> to show all packages.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universe)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">max.print =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data for the Bioc Universe (BU).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs_i <span class="ot">&lt;-</span> <span class="fu">universe_all_packages</span>(<span class="st">"bioc"</span>, <span class="at">limit =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What packages are in bioc universe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs <span class="ot">&lt;-</span> <span class="fu">universe_ls</span>(<span class="st">"bioc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bbs" class="level2">
<h2 class="anchored" data-anchor-id="bbs">BBS</h2>
<p>Get Bioconductor packages in BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocPkgTools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bbs_pkgs_i <span class="ot">&lt;-</span> BiocPkgTools<span class="sc">::</span><span class="fu">biocBuildReport</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">version =</span> BiocManager<span class="sc">::</span><span class="fu">version</span>(),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgType =</span> <span class="st">"software"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>bbs_pkgs <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unique</span>(bbs_pkgs_i<span class="sc">$</span>pkg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Packages in RU vs BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">RU =</span> <span class="fu">length</span>(ru_pkgs), <span class="at">BBS =</span> <span class="fu">length</span>(bbs_pkgs),</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"Total"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        RU  BBS
Total 2347 2361</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>not_in_ru <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(bbs_pkgs, ru_pkgs)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>not_in_bbs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(ru_pkgs, bbs_pkgs)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># There's probably a better way to do this</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">Created =</span> <span class="fu">character</span>(), <span class="at">Published =</span> <span class="fu">character</span>(),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">RemoteUrl =</span> <span class="fu">character</span>())</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> not_in_bbs) {</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  pi <span class="ot">&lt;-</span> <span class="fu">universe_one_package</span>(<span class="st">"bioc"</span>, pkg)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  pkgs_not_in_bbs <span class="ot">&lt;-</span> pkgs_not_in_bbs <span class="sc">|&gt;</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> pi<span class="sc">$</span>Package, <span class="at">Version =</span> pi<span class="sc">$</span>Version,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">Created =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_created</span><span class="st">`</span>, <span class="at">Published =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_published</span><span class="st">`</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">RemoteUrl =</span> pi<span class="sc">$</span>RemoteUrl)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-not-in-bbs-but-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-not-in-bbs-but-in-ru">Packages not in BBS but in RU</h3>
<p>Packages in RU but not in the BBS are likely added by maintainers via <a href="https://github.com/r-universe/bioc/blob/master/.remotes.json" class="uri">https://github.com/r-universe/bioc/blob/master/.remotes.json</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="sc">|&gt;</span> knitr<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Created</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Published</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RemoteUrl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Aerith</td>
<td style="text-align: left;">0.99.11</td>
<td style="text-align: left;">2026-02-02T09:50:47.000Z</td>
<td style="text-align: left;">2026-02-02T10:26:58.425Z</td>
<td style="text-align: left;">https://github.com/bioc/Aerith</td>
</tr>
<tr class="even">
<td style="text-align: left;">BiocMaintainerApp</td>
<td style="text-align: left;">0.99.0</td>
<td style="text-align: left;">2026-02-14T02:47:08.000Z</td>
<td style="text-align: left;">2026-02-14T02:50:35.754Z</td>
<td style="text-align: left;">https://github.com/bioc/BiocMaintainerApp</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CellMentor</td>
<td style="text-align: left;">0.99.0</td>
<td style="text-align: left;">2026-02-14T05:27:03.000Z</td>
<td style="text-align: left;">2026-02-14T05:41:18.976Z</td>
<td style="text-align: left;">https://github.com/bioc/CellMentor</td>
</tr>
<tr class="even">
<td style="text-align: left;">ClusterGVis</td>
<td style="text-align: left;">0.99.9</td>
<td style="text-align: left;">2026-02-03T09:54:11.000Z</td>
<td style="text-align: left;">2026-02-03T10:39:51.964Z</td>
<td style="text-align: left;">https://github.com/bioc/ClusterGVis</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CompensAID</td>
<td style="text-align: left;">0.99.6</td>
<td style="text-align: left;">2026-01-23T03:37:31.000Z</td>
<td style="text-align: left;">2026-01-23T03:45:06.768Z</td>
<td style="text-align: left;">https://github.com/bioc/CompensAID</td>
</tr>
<tr class="even">
<td style="text-align: left;">DOTSeq</td>
<td style="text-align: left;">0.99.4</td>
<td style="text-align: left;">2026-02-14T05:38:45.000Z</td>
<td style="text-align: left;">2026-02-14T05:57:42.218Z</td>
<td style="text-align: left;">https://github.com/bioc/DOTSeq</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DenoIST</td>
<td style="text-align: left;">0.99.2</td>
<td style="text-align: left;">2026-02-17T09:37:52.000Z</td>
<td style="text-align: left;">2026-02-17T09:52:37.903Z</td>
<td style="text-align: left;">https://github.com/bioc/DenoIST</td>
</tr>
<tr class="even">
<td style="text-align: left;">LACHESIS</td>
<td style="text-align: left;">0.99.4</td>
<td style="text-align: left;">2026-01-22T19:24:18.000Z</td>
<td style="text-align: left;">2026-01-22T19:57:10.565Z</td>
<td style="text-align: left;">https://github.com/bioc/LACHESIS</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MDSvis</td>
<td style="text-align: left;">0.99.7</td>
<td style="text-align: left;">2026-01-23T03:38:39.000Z</td>
<td style="text-align: left;">2026-01-23T03:47:53.582Z</td>
<td style="text-align: left;">https://github.com/bioc/MDSvis</td>
</tr>
<tr class="even">
<td style="text-align: left;">MutSeqR</td>
<td style="text-align: left;">0.99.9</td>
<td style="text-align: left;">2026-02-14T07:40:58.000Z</td>
<td style="text-align: left;">2026-02-14T07:52:10.459Z</td>
<td style="text-align: left;">https://github.com/bioc/MutSeqR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PTMods</td>
<td style="text-align: left;">0.99.1</td>
<td style="text-align: left;">2026-02-19T06:02:38.000Z</td>
<td style="text-align: left;">2026-02-19T06:05:49.828Z</td>
<td style="text-align: left;">https://github.com/bioc/PTMods</td>
</tr>
<tr class="even">
<td style="text-align: left;">SpiecEasi</td>
<td style="text-align: left;">1.99.3</td>
<td style="text-align: left;">2026-01-22T18:48:38.000Z</td>
<td style="text-align: left;">2026-01-22T19:03:54.914Z</td>
<td style="text-align: left;">https://github.com/bioc/SpiecEasi</td>
</tr>
<tr class="odd">
<td style="text-align: left;">StatescopeR</td>
<td style="text-align: left;">0.99.34</td>
<td style="text-align: left;">2026-01-29T05:05:13.000Z</td>
<td style="text-align: left;">2026-01-29T05:57:31.608Z</td>
<td style="text-align: left;">https://github.com/bioc/StatescopeR</td>
</tr>
<tr class="even">
<td style="text-align: left;">annoLinker</td>
<td style="text-align: left;">0.99.7</td>
<td style="text-align: left;">2026-01-29T04:16:25.000Z</td>
<td style="text-align: left;">2026-01-29T04:30:27.182Z</td>
<td style="text-align: left;">https://github.com/bioc/annoLinker</td>
</tr>
<tr class="odd">
<td style="text-align: left;">asuri</td>
<td style="text-align: left;">0.99.19</td>
<td style="text-align: left;">2026-02-07T04:16:46.000Z</td>
<td style="text-align: left;">2026-02-07T04:26:07.854Z</td>
<td style="text-align: left;">https://github.com/bioc/asuri</td>
</tr>
<tr class="even">
<td style="text-align: left;">carnation</td>
<td style="text-align: left;">0.99.8</td>
<td style="text-align: left;">2026-02-14T02:51:30.000Z</td>
<td style="text-align: left;">2026-02-14T03:06:54.148Z</td>
<td style="text-align: left;">https://github.com/bioc/carnation</td>
</tr>
<tr class="odd">
<td style="text-align: left;">damidBind</td>
<td style="text-align: left;">0.99.12</td>
<td style="text-align: left;">2026-02-06T04:46:15.000Z</td>
<td style="text-align: left;">2026-02-07T03:51:31.988Z</td>
<td style="text-align: left;">https://github.com/bioc/damidBind</td>
</tr>
<tr class="even">
<td style="text-align: left;">decemedip</td>
<td style="text-align: left;">0.99.8</td>
<td style="text-align: left;">2026-02-14T08:38:34.000Z</td>
<td style="text-align: left;">2026-02-16T04:18:41.438Z</td>
<td style="text-align: left;">https://github.com/bioc/decemedip</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dominatR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2026-01-22T18:50:11.000Z</td>
<td style="text-align: left;">2026-01-22T19:02:08.820Z</td>
<td style="text-align: left;">https://github.com/bioc/dominatR</td>
</tr>
<tr class="even">
<td style="text-align: left;">drugfindR</td>
<td style="text-align: left;">0.99.2516</td>
<td style="text-align: left;">2026-02-15T06:00:42.000Z</td>
<td style="text-align: left;">2026-02-15T06:05:57.004Z</td>
<td style="text-align: left;">https://github.com/bioc/drugfindR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">epiSeeker</td>
<td style="text-align: left;">0.99.14</td>
<td style="text-align: left;">2026-02-14T05:28:01.000Z</td>
<td style="text-align: left;">2026-02-14T05:42:31.262Z</td>
<td style="text-align: left;">https://github.com/bioc/epiSeeker</td>
</tr>
<tr class="even">
<td style="text-align: left;">glycoTraitR</td>
<td style="text-align: left;">0.99.2</td>
<td style="text-align: left;">2026-02-14T02:48:11.000Z</td>
<td style="text-align: left;">2026-02-14T03:01:15.568Z</td>
<td style="text-align: left;">https://github.com/bioc/glycoTraitR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">immReferent</td>
<td style="text-align: left;">0.99.6</td>
<td style="text-align: left;">2026-02-04T04:28:11.000Z</td>
<td style="text-align: left;">2026-02-04T04:45:37.759Z</td>
<td style="text-align: left;">https://github.com/bioc/immReferent</td>
</tr>
<tr class="even">
<td style="text-align: left;">lcmsPlot</td>
<td style="text-align: left;">0.99.16</td>
<td style="text-align: left;">2026-02-19T06:04:56.000Z</td>
<td style="text-align: left;">2026-02-19T06:14:25.304Z</td>
<td style="text-align: left;">https://github.com/bioc/lcmsPlot</td>
</tr>
<tr class="odd">
<td style="text-align: left;">leapR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2026-02-19T08:28:52.000Z</td>
<td style="text-align: left;">2026-02-19T08:35:50.647Z</td>
<td style="text-align: left;">https://github.com/bioc/leapR</td>
</tr>
<tr class="even">
<td style="text-align: left;">plaid</td>
<td style="text-align: left;">0.99.19</td>
<td style="text-align: left;">2026-02-14T05:12:51.000Z</td>
<td style="text-align: left;">2026-02-14T05:19:45.013Z</td>
<td style="text-align: left;">https://github.com/bioc/plaid</td>
</tr>
<tr class="odd">
<td style="text-align: left;">posDemux</td>
<td style="text-align: left;">0.99.11</td>
<td style="text-align: left;">2026-02-03T04:18:17.000Z</td>
<td style="text-align: left;">2026-02-03T04:26:22.659Z</td>
<td style="text-align: left;">https://github.com/bioc/posDemux</td>
</tr>
<tr class="even">
<td style="text-align: left;">proBatch</td>
<td style="text-align: left;">1.99.4</td>
<td style="text-align: left;">2026-01-29T04:17:31.000Z</td>
<td style="text-align: left;">2026-01-29T04:30:23.542Z</td>
<td style="text-align: left;">https://github.com/bioc/proBatch</td>
</tr>
<tr class="odd">
<td style="text-align: left;">singIST</td>
<td style="text-align: left;">0.99.85</td>
<td style="text-align: left;">2026-02-14T08:44:05.000Z</td>
<td style="text-align: left;">2026-02-14T09:00:24.767Z</td>
<td style="text-align: left;">https://git.bioconductor.org/packages/singIST</td>
</tr>
<tr class="even">
<td style="text-align: left;">staRgate</td>
<td style="text-align: left;">0.99.6</td>
<td style="text-align: left;">2026-01-24T04:38:21.000Z</td>
<td style="text-align: left;">2026-01-24T04:43:03.992Z</td>
<td style="text-align: left;">https://github.com/bioc/staRgate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tidyexposomics</td>
<td style="text-align: left;">0.99.12</td>
<td style="text-align: left;">2026-02-14T05:36:22.000Z</td>
<td style="text-align: left;">2026-02-14T06:00:00.234Z</td>
<td style="text-align: left;">https://github.com/bioc/tidyexposomics</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="packages-in-bbs-but-not-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-in-bbs-but-not-in-ru">Packages in BBS but not in RU</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>not_in_ru</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "bgx"                    "BiGGR"                  "biodbHmdb"             
 [4] "biodbNcbi"              "biodbNci"               "biodbUniprot"          
 [7] "ccmap"                  "CellScore"              "CINdex"                
[10] "cisPath"                "CuratedAtlasQueryR"     "DEP"                   
[13] "GEOexplorer"            "gpuMagic"               "Harshlight"            
[16] "hiAnnotator"            "hiReadsProcessor"       "hmdbQuery"             
[19] "IMAS"                   "interactiveDisplay"     "interactiveDisplayBase"
[22] "lapmix"                 "LinTInd"                "lute"                  
[25] "MADSEQ"                 "netZooR"                "oppti"                 
[28] "PhenStat"               "qckitfastq"             "ReactomeGraph4R"       
[31] "Repitools"              "rGADEM"                 "RgnTX"                 
[34] "rRDP"                   "SARC"                   "seqArchR"              
[37] "seqArchRplus"           "seqTools"               "spatzie"               
[40] "Streamer"               "tidytof"                "TitanCNA"              
[43] "TransView"              "traviz"                 "XNAString"             </code></pre>
</div>
</div>
<p>The following packages are not deprecated but are not in RU.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>depre_pkgs <span class="ot">&lt;-</span> <span class="fu">filter</span>(bbs_pkgs_i, Deprecated <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(pkg) <span class="sc">|&gt;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(not_in_ru, depre_pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "biodbHmdb"          "biodbNcbi"          "biodbNci"          
 [4] "biodbUniprot"       "cisPath"            "CuratedAtlasQueryR"
 [7] "GEOexplorer"        "hiAnnotator"        "hmdbQuery"         
[10] "IMAS"               "RgnTX"              "spatzie"           
[13] "tidytof"            "traviz"            </code></pre>
</div>
</div>
</section>
</section>
<section id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc." class="level2">
<h2 class="anchored" data-anchor-id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc.">Making sense of R Universe’s build status in respect to R CMD build, check, etc.</h2>
<p>Looking at the status of linux x86_64 binary builds. The build “status” appears to get “success”, “failure”, or “cancelled” from the GitHub Action overall. The “check” value corresponds to <code>R CMD check</code>. I interpret the result of “status” to be “success” OK and “failure” ERROR for <code>R CMD build</code> but it may mean something else. We’ll use CANCELLED as it is. There’s also a special status for documentation, which is handled separately from build/check, which may account for some discrepancies. Not sure if the build status for vignettes is available in the API.</p>
<p>Packages with my comment “Missing R4.6 on Linux” did have data in the API but no x86_64 linux with R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Status =</span> <span class="fu">character</span>(), <span class="at">Build =</span> <span class="fu">character</span>(),</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Check =</span> <span class="fu">character</span>(), <span class="at">Comment =</span> <span class="fu">character</span>())</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#, Buildurl = character(),</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># Commit = character())</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> ru_pkgs_i) {</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> <span class="cn">NA</span>, <span class="at">build =</span> <span class="cn">NA</span>, <span class="at">check =</span> <span class="cn">NA</span>, <span class="at">buildurl =</span> <span class="cn">NA</span>)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  comment <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (b <span class="cf">in</span> p<span class="sc">$</span><span class="st">`</span><span class="at">_binaries</span><span class="st">`</span>) {</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (b<span class="sc">$</span>os <span class="sc">==</span> <span class="st">"linux"</span> <span class="sc">&amp;&amp;</span> b<span class="sc">$</span>r <span class="sc">==</span> <span class="st">"4.6.0"</span>) {</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>      build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> b<span class="sc">$</span>status, <span class="at">check =</span> b<span class="sc">$</span>check, <span class="at">buildurl =</span> b<span class="sc">$</span>buildurl)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(build[<span class="st">"status"</span>])) {</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> <span class="st">"Missing R4.6 on Linux"</span> </span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"success"</span>) {</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"failure"</span>) {</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"ERROR*"</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"cancelled"</span>) {</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"CANCELLED"</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  ru_status <span class="ot">&lt;-</span> ru_status <span class="sc">|&gt;</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> p<span class="sc">$</span>Package, <span class="at">Version =</span> p<span class="sc">$</span>Version, <span class="at">Status =</span> build[<span class="st">"status"</span>],</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">Build =</span> build[<span class="st">"build"</span>], <span class="at">Check =</span> build[<span class="st">"check"</span>], <span class="at">Comment =</span> comment)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>            <span class="co">#, Buildurl = build["buildurl"], Commit = p$`_commit`$id)</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a><span class="co"># order by Package</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> ru_status[<span class="fu">order</span>(ru_status<span class="sc">$</span>Package), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>DEVEL_LINUX_BUILDER <span class="ot">=</span> <span class="st">"nebbiolo2"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper functions</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>get_result <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  entry <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pkg <span class="sc">==</span> x[<span class="st">"pkg"</span>], stage <span class="sc">==</span> x[<span class="st">"stage"</span>], node <span class="sc">==</span> x[<span class="st">"node"</span>])</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(entry)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    entry <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    entry<span class="sc">$</span>result</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>get_install <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"install"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>get_build <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"buildsrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>get_check <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"checksrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>bbs_status <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(node <span class="sc">==</span> DEVEL_LINUX_BUILDER) <span class="sc">|&gt;</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version, Deprecated) <span class="sc">|&gt;</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(pkg, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">install =</span> <span class="fu">map_chr</span>(pkg, get_install)) <span class="sc">|&gt;</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">build =</span> <span class="fu">map_chr</span>(pkg, get_build)) <span class="sc">|&gt;</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">check =</span> <span class="fu">map_chr</span>(pkg, get_check))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-the-status-of-packages-in-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-status-of-packages-in-ru-and-bbs">Compare the status of packages in RU and BBS</h3>
<p>We can only compare <code>check</code> and <code>build</code>, so we’ll check if those columns agree. Note that in the BBS, <code>check</code> can show “WARNINGS” whereas R Universe can show “WARNING” (no s) as well as “OK”, “NOTE”, “ERROR”, or NA.</p>
<p>Note: - Deprecated packages are removed immedately in R Universe. - Packages may have different versions since R-Universe builds new versions after detecting a “push”.</p>
<p>The following includes packages not deprecated in BBS where build and check don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ru_status) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pkg"</span>, <span class="st">"version"</span>, <span class="st">"status"</span>, <span class="st">"build"</span>, <span class="st">"check"</span>, <span class="st">"comment"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>bbs_ru_comparison <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> bbs_status, <span class="at">y =</span> ru_status, <span class="at">by =</span> <span class="st">"pkg"</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">&amp;</span> Deprecated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(pkg, install, build.x, build.y, check.x, check.y, version.x,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                  version.y, status, comment)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>, <span class="st">"RU_build"</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"BBS_version"</span>, <span class="st">"RU_version"</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-without-x86_64-linux-builds-for-r-4.6" class="level4">
<h4 class="anchored" data-anchor-id="packages-without-x86_64-linux-builds-for-r-4.6">Packages without x86_64 Linux builds for R 4.6</h4>
<p>The following are packages in bioc R Universe but without an x86_64 Linux build for R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="sc">|&gt;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Comment <span class="sc">==</span> <span class="st">"Missing R4.6 on Linux"</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages in RU without an x86_64 linux build for R 4.6"</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages in RU without an x86_64 linux build for R 4.6</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BgeeCall</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.26.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">casper</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.44.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ChIPQC</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.46.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">chromVAR</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CONFESS</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">consensusDE</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.28.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ddPCRclust</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.30.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">debCAM</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.28.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">esATAC</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">GEOmetadb</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.72.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HilbertVisGUI</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.68.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">nearBynding</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.20.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">profileplyr</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.26.1</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">SamSPECTRAL</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.64.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SwathXtend</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">wavClusteR</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.44.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">zellkonverter</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.20.1</td>
<td style="text-align: left;">1.21.1</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Notes: - Retrying to refresh the build on the dashboard failed for some packages. How can we rebuild a package that hasn’t received a push in 3 months? - paxtoolsr should be in Bioc R Universe, but it isn’t. Why?</p>
</section>
</section>
</section>
<section id="packages-not-in-agreement" class="level2">
<h2 class="anchored" data-anchor-id="packages-not-in-agreement">Packages not in agreement</h2>
<p>Looking at packages with the same version and removing deprecated packages, we have the following packages where the BBS and R Universe’s results don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Deprecated <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> version.x <span class="sc">==</span> version.y <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(status)) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version.x, install, build.x, build.y, check.x, check.y,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>         status, comment)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_version"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"RU_build"</span>, <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="sc">|&gt;</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages with different build results between BBS and RU"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages with different build results between BBS and RU</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Note: <code>ERROR*</code> could be an <code>ERROR</code>, <code>TIMEOUT</code>, or <code>skipped</code>.</p>
<section id="total-packages-not-in-agreement" class="level3">
<h3 class="anchored" data-anchor-id="total-packages-not-in-agreement">Total packages not in agreement</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bbs_ru_not_ok2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="other" class="level1">
<h1>Other</h1>
<p>If a package repo isn’t accessible or an error exists in the DESCRIPTION file, there will be an error in the registry for that submodule and it will show up at <a href="https://github.com/r-universe/bioc/actions/workflows/sync.yml" class="uri">https://github.com/r-universe/bioc/actions/workflows/sync.yml</a>. The red triangle on <a href="https://bioc.r-universe.dev/" class="uri">https://bioc.r-universe.dev/</a> will take you to the specific workflow error.</p>
<p>Currently, this information is available on the BBS but I wonder if this information is available somewhere for R Universe OR if it would be necessary given that developers would be able to inspect their builds sooner.</p>
<section id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api" class="level3">
<h3 class="anchored" data-anchor-id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api">Important: Use the dbdump endpoint with read_bson to dump the API</h3>
<p>Use <code>read_bson</code> to dump API data into a db</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("mongolite")</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mongolite)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>bioc <span class="ot">&lt;-</span> mongolite<span class="sc">::</span><span class="fu">read_bson</span>(<span class="st">'https://bioc.r-universe.dev/api/dbdump'</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: it returns 2268 records rather than 2276 (fewer retrieved through</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># another endpoint)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-a-tarball-from-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="comparing-a-tarball-from-ru-and-bbs">Comparing a tarball from RU and BBS</h3>
<p>Lastly, when I look at the builds, they’re using various repositories, such p3m, r-universe, etc. This might be okay as long as the resulting binaries/tarballs are the generally the same. Looking at XVector for both, both the BBS and R Universe inject some additional information. R Universe also creates an <code>extra</code> directory that has citations, manuals, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2 ruXVector_0.47.2</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2/XVector/DESCRIPTION ruXVector_0.47.2/XVector/DESCRIPTION</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="ex">5c5</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>   <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span>         <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="ex">28,33c28,31</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_url: <span class="ex">https://git.bioconductor.org/packages/XVector</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_branch: <span class="ex">devel</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit: <span class="ex">9f44218</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit_date: <span class="ex">2025-01-07</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Repository: <span class="ex">Bioconductor</span> 3.21</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Date/Publication: <span class="ex">2025-01-08</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Repository: <span class="ex">https://bioc.r-universe.dev</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteUrl: <span class="ex">https://github.com/bioc/XVector</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteRef: <span class="ex">HEAD</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteSha: <span class="ex">9f44218308f9d5867f0b9df5fae60d217f4e7d65</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="ex">35c33</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Packaged: <span class="ex">2025-01-09</span> 00:57:29 UTC<span class="kw">;</span> <span class="ex">biocbuild</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Packaged: <span class="ex">2025-01-08</span> 03:05:41 UTC<span class="kw">;</span> <span class="ex">root</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: extra</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: manual.pdf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="structure-of-a-package-through-universe" class="level3">
<h3 class="anchored" data-anchor-id="structure-of-a-package-through-universe">Structure of a package through universe</h3>
<p>The structure of a package looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">str</span>(ru_pkgs_i[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ :List of 74
  ..$ Package           : chr "ggcyto"
  ..$ Type              : chr "Package"
  ..$ Title             : chr "Visualize Cytometry data with ggplot"
  ..$ Version           : chr "1.39.2"
  ..$ Date              : chr "2015-11-02"
  ..$ Author            : chr "Mike Jiang"
  ..$ Maintainer        : chr "Mike Jiang &lt;mike@ozette.com&gt;"
  ..$ Description       : chr "With the dedicated fortify method implemented for flowSet,\nncdfFlowSet and GatingSet classes, both raw and gat"| __truncated__
  ..$ VignetteBuilder   : chr "knitr"
  ..$ License           : chr "file LICENSE"
  ..$ URL               : chr "https://github.com/RGLab/ggcyto/issues"
  ..$ biocViews         : chr "ImmunoOncology, FlowCytometry, CellBasedAssays,\nInfrastructure, Visualization"
  ..$ Collate           : chr "'AllClasses.R' 'autoplot.R' 'axis_inverse_trans.R'\n'compute_stats.R' 'faust_gating_plot.R' 'fortify.R'\n'forti"| __truncated__
  ..$ RoxygenNote       : chr "7.2.1"
  ..$ Roxygen           : chr "list(markdown=TRUE)"
  ..$ Config/pak/sysreqs: chr "libxml2-dev zlib1g-dev"
  ..$ Repository        : chr "https://bioc.r-universe.dev"
  ..$ Date/Publication  : chr "2026-02-19 02:57:34 UTC"
  ..$ RemoteUrl         : chr "https://github.com/bioc/ggcyto"
  ..$ RemoteRef         : chr "HEAD"
  ..$ RemoteSha         : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  ..$ NeedsCompilation  : chr "no"
  ..$ Packaged          :List of 2
  .. ..$ Date: chr "2026-02-19 06:04:01 UTC"
  .. ..$ User: chr "root"
  ..$ MD5sum            : chr "35264c4ea98a9830f09f86104b84acab"
  ..$ _user             : chr "bioc"
  ..$ _type             : chr "src"
  ..$ _file             : chr "ggcyto_1.39.2.tar.gz"
  ..$ _fileid           : chr "103e6093018429c7f5b77b23f38644e81ac61307e09250d7a9aed44f9405fc72"
  ..$ _filesize         : int 22981041
  ..$ _sha256           : chr "103e6093018429c7f5b77b23f38644e81ac61307e09250d7a9aed44f9405fc72"
  ..$ _created          : chr "2026-02-19T06:04:01.000Z"
  ..$ _published        : chr "2026-02-19T06:10:20.528Z"
  ..$ _jobs             :List of 10
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 180
  .. .. ..$ config  : chr "bioc-checks"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "5567936797"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 315
  .. .. ..$ config  : chr "linux-devel-x86_64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567953399"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 308
  .. .. ..$ config  : chr "linux-release-x86_64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567952556"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 181
  .. .. ..$ config  : chr "macos-devel-arm64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567936959"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 170
  .. .. ..$ config  : chr "macos-release-arm64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567935421"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 311
  .. .. ..$ config  : chr "source"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "5567912749"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 147
  .. .. ..$ config  : chr "wasm-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "5567932410"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 268
  .. .. ..$ config  : chr "windows-devel"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567948083"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 287
  .. .. ..$ config  : chr "windows-oldrel"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567950786"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.41e+10
  .. .. ..$ time    : int 276
  .. .. ..$ config  : chr "windows-release"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5567948718"
  ..$ _bioccheck        :List of 3
  .. ..$ error  : int 1
  .. ..$ warning: int 3
  .. ..$ note   : int 19
  ..$ _buildurl         : chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  ..$ _status           : chr "success"
  ..$ _host             : chr "GitHub-Actions"
  ..$ _upstream         : chr "https://github.com/bioc/ggcyto"
  ..$ _commit           :List of 5
  .. ..$ id       : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. ..$ author   : chr "Mike Jiang &lt;mike@ozette.com&gt;"
  .. ..$ committer: chr "Mike Jiang &lt;mike@ozette.com&gt;"
  .. ..$ message  : chr "bump\n"
  .. ..$ time     : int 1771469854
  ..$ _maintainer       :List of 4
  .. ..$ name : chr "Mike Jiang"
  .. ..$ email: chr "mike@ozette.com"
  .. ..$ login: chr "mikejiang"
  .. ..$ uuid : int 1385649
  ..$ _distro           : chr "noble"
  ..$ _registered       : logi TRUE
  ..$ _dependencies     :List of 21
  .. ..$ :List of 2
  .. .. ..$ package: chr "methods"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "ggplot2"
  .. .. ..$ version: chr "&gt;= 3.5.0"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "flowCore"
  .. .. ..$ version: chr "&gt;= 1.41.5"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "ncdfFlow"
  .. .. ..$ version: chr "&gt;= 2.17.1"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "flowWorkspace"
  .. .. ..$ version: chr "&gt;= 4.3.1"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "plyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scales"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "hexbin"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "data.table"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "RColorBrewer"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "gridExtra"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rlang"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "testthat"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "flowWorkspaceData"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "knitr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rmarkdown"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "flowStats"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "openCyto"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "flowViz"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggridges"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "vdiffr"
  .. .. ..$ role   : chr "Suggests"
  ..$ _owner            : chr "bioc"
  ..$ _selfowned        : logi TRUE
  ..$ _usedby           : int 7
  ..$ _updates          :List of 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-27"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-38"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2026-01"
  .. .. ..$ n   : int 3
  .. ..$ :List of 2
  .. .. ..$ week: chr "2026-08"
  .. .. ..$ n   : int 2
  ..$ _tags             : list()
  ..$ _bioc             :List of 2
  .. ..$ :List of 3
  .. .. ..$ branch : chr "devel"
  .. .. ..$ version: chr "1.39.1"
  .. .. ..$ bioc   : chr "3.23"
  .. ..$ :List of 3
  .. .. ..$ branch : chr "release"
  .. .. ..$ version: chr "1.38.1"
  .. .. ..$ bioc   : chr "3.22"
  ..$ _topics           :List of 5
  .. ..$ : chr "immunooncology"
  .. ..$ : chr "flowcytometry"
  .. ..$ : chr "cellbasedassays"
  .. ..$ : chr "infrastructure"
  .. ..$ : chr "visualization"
  ..$ _stars            : int 65
  ..$ _contributors     :List of 12
  .. ..$ :List of 3
  .. .. ..$ user : chr "jacobpwagner"
  .. .. ..$ count: int 50
  .. .. ..$ uuid : int 43548373
  .. ..$ :List of 3
  .. .. ..$ user : chr "nturaga"
  .. .. ..$ count: int 22
  .. .. ..$ uuid : int 2746443
  .. ..$ :List of 3
  .. .. ..$ user : chr "mikejiang"
  .. .. ..$ count: int 16
  .. .. ..$ uuid : int 1385649
  .. ..$ :List of 3
  .. .. ..$ user : chr "jwokaty"
  .. .. ..$ count: int 14
  .. .. ..$ uuid : int 1744257
  .. ..$ :List of 3
  .. .. ..$ user : chr "teunbrand"
  .. .. ..$ count: int 8
  .. .. ..$ uuid : int 49372158
  .. ..$ :List of 3
  .. .. ..$ user : chr "djhammill"
  .. .. ..$ count: int 6
  .. .. ..$ uuid : int 125331387
  .. ..$ :List of 3
  .. .. ..$ user : chr "gfinak"
  .. .. ..$ count: int 6
  .. .. ..$ uuid : int 1385593
  .. ..$ :List of 3
  .. .. ..$ user : chr "hpages"
  .. .. ..$ count: int 5
  .. .. ..$ uuid : int 8810451
  .. ..$ :List of 3
  .. .. ..$ user : chr "dtenenba"
  .. .. ..$ count: int 4
  .. .. ..$ uuid : int 2286826
  .. ..$ :List of 3
  .. .. ..$ user : chr "vobencha"
  .. .. ..$ count: int 4
  .. .. ..$ uuid : int 2466173
  .. ..$ :List of 3
  .. .. ..$ user : chr "kayla-morrell"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 32339524
  .. ..$ :List of 3
  .. .. ..$ user : chr "saflatfish508"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 67445939
  ..$ _userbio          :List of 4
  .. ..$ uuid       : int 2286807
  .. ..$ type       : chr "organization"
  .. ..$ name       : chr "Bioconductor"
  .. ..$ description: chr "Software for the analysis and comprehension of high-throughput genomic data"
  ..$ _downloads        :List of 2
  .. ..$ count : int 2036
  .. ..$ source: chr "https://www.bioconductor.org/packages/stats/bioc/ggcyto"
  ..$ _mentions         : int 13
  ..$ _devurl           : chr "https://github.com/rglab/ggcyto"
  ..$ _searchresults    : int 466
  ..$ _rbuild           : chr "4.5.2"
  ..$ _assets           :List of 11
  .. ..$ : chr "extra/citation.cff"
  .. ..$ : chr "extra/citation.html"
  .. ..$ : chr "extra/citation.json"
  .. ..$ : chr "extra/citation.txt"
  .. ..$ : chr "extra/contents.json"
  .. ..$ : chr "extra/ggcyto.html"
  .. ..$ : chr "extra/NEWS.html"
  .. ..$ : chr "extra/NEWS.txt"
  .. ..$ : chr "extra/readme.html"
  .. ..$ : chr "extra/readme.md"
  .. ..$ : chr "manual.pdf"
  ..$ _homeurl          : chr "https://github.com/rglab/ggcyto"
  ..$ _realowner        : chr "bioc"
  ..$ _cranurl          : logi FALSE
  ..$ _exports          :List of 45
  .. ..$ : chr ".fr2dt"
  .. ..$ : chr ".fs2dt"
  .. ..$ : chr "%+%"
  .. ..$ : chr "as.ggplot"
  .. ..$ : chr "autoplot"
  .. ..$ : chr "axis_x_inverse_trans"
  .. ..$ : chr "axis_y_inverse_trans"
  .. ..$ : chr "compute_stats"
  .. ..$ : chr "faust_gating_plot"
  .. ..$ : chr "flowCore_asinht_trans"
  .. ..$ : chr "fortify_fs"
  .. ..$ : chr "gate_null"
  .. ..$ : chr "geom_gate"
  .. ..$ : chr "geom_hvline"
  .. ..$ : chr "geom_multi_range"
  .. ..$ : chr "geom_overlay"
  .. ..$ : chr "geom_stats"
  .. ..$ : chr "getFlowFrame"
  .. ..$ : chr "ggcyto"
  .. ..$ : chr "ggcyto_arrange"
  .. ..$ : chr "ggcyto_par_default"
  .. ..$ : chr "ggcyto_par_set"
  .. ..$ : chr "is.ggcyto"
  .. ..$ : chr "is.ggcyto_flowSet"
  .. ..$ : chr "is.ggcyto_par"
  .. ..$ : chr "labs_cyto"
  .. ..$ : chr "marginalFilter"
  .. ..$ : chr "print"
  .. ..$ : chr "rescale_gate"
  .. ..$ : chr "scale_x_flowCore_fasinh"
  .. ..$ : chr "scale_x_flowjo_biexp"
  .. ..$ : chr "scale_x_flowJo_biexp"
  .. ..$ : chr "scale_x_flowjo_fasinh"
  .. ..$ : chr "scale_x_flowJo_fasinh"
  .. ..$ : chr "scale_x_logicle"
  .. ..$ : chr "scale_y_flowCore_fasinh"
  .. ..$ : chr "scale_y_flowjo_biexp"
  .. ..$ : chr "scale_y_flowJo_biexp"
  .. ..$ : chr "scale_y_flowjo_fasinh"
  .. ..$ : chr "scale_y_flowJo_fasinh"
  .. ..$ : chr "scale_y_logicle"
  .. ..$ : chr "show"
  .. ..$ : chr "stat_position"
  .. ..$ : chr "stats_null"
  .. ..$ : chr "transform"
  ..$ _help             :List of 41
  .. ..$ :List of 3
  .. .. ..$ page  : chr "as.ggplot"
  .. .. ..$ title : chr "It fortifies the data, fills some default settings and returns a regular ggplot object."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "as.ggplot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "autoplot"
  .. .. ..$ title : chr "Plot cytometry data in one or two dimension with the ggcyto package."
  .. .. ..$ topics:List of 9
  .. .. .. ..$ : chr "autoplot"
  .. .. .. ..$ : chr "autoplot.cytoframe"
  .. .. .. ..$ : chr "autoplot.cytoset"
  .. .. .. ..$ : chr "autoplot.flowFrame"
  .. .. .. ..$ : chr "autoplot.flowSet"
  .. .. .. ..$ : chr "autoplot.GatingHierarchy"
  .. .. .. ..$ : chr "autoplot.GatingSet"
  .. .. .. ..$ : chr "autoplot.GatingSetList"
  .. .. .. ..$ : chr "autoplot.ncdfFlowList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "axis_x_inverse_trans"
  .. .. ..$ title : chr "Display ggcyto axis labels using their raw values (as stored in the data structure)"
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "axis_x_inverse_trans"
  .. .. .. ..$ : chr "axis_y_inverse_trans"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "compute_stats"
  .. .. ..$ title : chr "compute the statistics of the cell population defined by gates"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "compute_stats"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "faust_gating_plot"
  .. .. ..$ title : chr "plot faust gating schemes"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "faust_gating_plot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "flowCore_asinht_trans"
  .. .. ..$ title : chr "Inverse hyperbolic sine transformation(flowCore version)."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "flowCore_asinht_trans"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify_fs"
  .. .. ..$ title : chr "Fortify a model into flowSet object"
  .. .. ..$ topics:List of 7
  .. .. .. ..$ : chr "fortify_fs"
  .. .. .. ..$ : chr "fortify_fs.cytoframe"
  .. .. .. ..$ : chr "fortify_fs.default"
  .. .. .. ..$ : chr "fortify_fs.flowFrame"
  .. .. .. ..$ : chr "fortify_fs.flowSet"
  .. .. .. ..$ : chr "fortify_fs.GatingSet"
  .. .. .. ..$ : chr "fortify_fs.GatingSetList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.flowSet"
  .. .. ..$ title : chr "Convert a flowFrame/flowSet/GatingSet to a ggplot-compatible data.table"
  .. .. ..$ topics:List of 8
  .. .. .. ..$ : chr "fortify"
  .. .. .. ..$ : chr "fortify.cytoframe"
  .. .. .. ..$ : chr "fortify.cytoset"
  .. .. .. ..$ : chr "fortify.flowFrame"
  .. .. .. ..$ : chr "fortify.flowSet"
  .. .. .. ..$ : chr "fortify.GatingSet"
  .. .. .. ..$ : chr "fortify.GatingSetList"
  .. .. .. ..$ : chr "fortify.ncdfFlowList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.ellipsoidGate"
  .. .. ..$ title : chr "Convert a ellipsoidGate to a data.table useful for ggplot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "fortify.ellipsoidGate"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.filterList"
  .. .. ..$ title : chr "Convert a filterList to a data.table useful for ggplot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "fortify.filterList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.multiRangeGate"
  .. .. ..$ title : chr "Convert a multiRangeGate to a data.table useful for ggplot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "fortify.multiRangeGate"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.polygonGate"
  .. .. ..$ title : chr "Convert a polygonGate to a data.table useful for ggplot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "fortify.polygonGate"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "fortify.rectangleGate"
  .. .. ..$ title : chr "Convert a rectangleGate to a data.table useful for ggplot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "fortify.rectangleGate"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "gate_null"
  .. .. ..$ title : chr "clear all the geom_gate() layer previously added"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "gate_null"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "geom_gate"
  .. .. ..$ title : chr "Add a gate layer to a ggcyto plot."
  .. .. ..$ topics:List of 11
  .. .. .. ..$ : chr "geom_gate"
  .. .. .. ..$ : chr "geom_gate.character"
  .. .. .. ..$ : chr "geom_gate.default"
  .. .. .. ..$ : chr "geom_gate.filter"
  .. .. .. ..$ : chr "geom_gate.filterList"
  .. .. .. ..$ : chr "geom_gate.filters"
  .. .. .. ..$ : chr "geom_gate.filtersList"
  .. .. .. ..$ : chr "geom_gate.list"
  .. .. .. ..$ : chr "geom_gate.logical"
  .. .. .. ..$ : chr "geom_gate.logicalFilterResult"
  .. .. .. ..$ : chr "geom_gate.quadGate"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "geom_hvline"
  .. .. ..$ title : chr "Vertical or horizontal line."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "geom_hvline"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "geom_multi_range"
  .. .. ..$ title : chr "Draw multi-ranges as multiple rectangles on 1D or 2D plot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "geom_multi_range"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "geom_overlay"
  .. .. ..$ title : chr "Overlay a population on an existing ggcyto plot analogous to backgating."
  .. .. ..$ topics:List of 6
  .. .. .. ..$ : chr "geom_overlay"
  .. .. .. ..$ : chr "geom_overlay.character"
  .. .. .. ..$ : chr "geom_overlay.default"
  .. .. .. ..$ : chr "geom_overlay.flowFrame"
  .. .. .. ..$ : chr "geom_overlay.flowSet"
  .. .. .. ..$ : chr "geom_overlay.ncdfFlowList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "geom_stats"
  .. .. ..$ title : chr "Add a popluation statistics layer to a ggcyto plot."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "geom_stats"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getFlowFrame"
  .. .. ..$ title : chr "extract flowFrame data structure from the given R object"
  .. .. ..$ topics:List of 6
  .. .. .. ..$ : chr "getFlowFrame"
  .. .. .. ..$ : chr "getFlowFrame.flowSet"
  .. .. .. ..$ : chr "getFlowFrame.GatingHierarchy"
  .. .. .. ..$ : chr "getFlowFrame.GatingSet"
  .. .. .. ..$ : chr "getFlowFrame.GatingSetList"
  .. .. .. ..$ : chr "getFlowFrame.ncdfFlowList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ggcyto_add"
  .. .. ..$ title : chr "overloaded '+' method for ggcyto"
  .. .. ..$ topics:List of 12
  .. .. .. ..$ : chr "+"
  .. .. .. ..$ : chr "+,ggcyto_flowSet,ANY-method"
  .. .. .. ..$ : chr "+,ggcyto_flowSet-method"
  .. .. .. ..$ : chr "+,ggcyto_GatingLayout,ANY-method"
  .. .. .. ..$ : chr "+,ggcyto_GatingLayout-method"
  .. .. .. ..$ : chr "+,ggcyto_GatingSet,ANY-method"
  .. .. .. ..$ : chr "+,ggcyto_GatingSet-method"
  .. .. .. ..$ : chr "+.ggcyto_flowSet"
  .. .. .. ..$ : chr "+.ggcyto_GatingLayout"
  .. .. .. ..$ : chr "+.ggcyto_GatingSet"
  .. .. .. ..$ : chr "+.ggcyto_ncdfFlowList"
  .. .. .. ..$ : chr "ggcyto_add"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ggcyto_arrange"
  .. .. ..$ title : chr "Arrange a list of ggplot objects into gtable"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "ggcyto_arrange"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ggcyto_par_default"
  .. .. ..$ title : chr "Return The default ggcyto settings"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "ggcyto_par_default"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ggcyto_par_set"
  .. .. ..$ title : chr "Set some default parameters for ggcyto"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "ggcyto_par_set"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ggcyto"
  .. .. ..$ title : chr "Plot cytometry data using the ggcyto API"
  .. .. ..$ topics:List of 10
  .. .. .. ..$ : chr "ggcyto"
  .. .. .. ..$ : chr "ggcyto-class"
  .. .. .. ..$ : chr "ggcyto.default"
  .. .. .. ..$ : chr "ggcyto.flowSet"
  .. .. .. ..$ : chr "ggcyto.GatingHierarchy"
  .. .. .. ..$ : chr "ggcyto.GatingSet"
  .. .. .. ..$ : chr "ggcyto.GatingSetList"
  .. .. .. ..$ : chr "ggcyto_flowSet-class"
  .. .. .. ..$ : chr "ggcyto_GatingLayout-class"
  .. .. .. ..$ : chr "ggcyto_GatingSet-class"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "is.ggcyto"
  .. .. ..$ title : chr "Reports whether x is a ggcyto object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "is.ggcyto"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "is.ggcyto_flowSet"
  .. .. ..$ title : chr "Reports whether x is a ggcyto_flowSet object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "is.ggcyto_flowSet"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "is.ggcyto_par"
  .. .. ..$ title : chr "Reports whether x is a ggcyto_par object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "is.ggcyto_par"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "labs_cyto"
  .. .. ..$ title : chr "Change axis labels and legend titles"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "labs_cyto"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "marginalFilter"
  .. .. ..$ title : chr "Generate a marginal gate."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "marginalFilter"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "merge.quad.gates"
  .. .. ..$ title : chr "extend the original flowWorkspace:::.mergeGates function to restore quadGate when applicable"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "merge.quad.gates"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "print.ggcyto"
  .. .. ..$ title : chr "Draw ggcyto on current graphics device."
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "plot.ggcyto"
  .. .. .. ..$ : chr "print,ggcyto-method"
  .. .. .. ..$ : chr "print.ggcyto"
  .. .. .. ..$ : chr "show,ggcyto-method"
  .. .. .. ..$ : chr "show.ggcyto"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "print.ggcyto_GatingLayout"
  .. .. ..$ title : chr "print method for ggcyto_gate_layout class"
  .. .. ..$ topics:List of 3
  .. .. .. ..$ : chr "print.ggcyto_GatingLayout"
  .. .. .. ..$ : chr "show,ggcyto_GatingLayout-method"
  .. .. .. ..$ : chr "show.ggcyto_GatingLayout"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "replace_data"
  .. .. ..$ title : chr "replace current cytometry data"
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "%+%"
  .. .. .. ..$ : chr "%+%,ggcyto-method"
  .. .. .. ..$ : chr "%+%,ggcyto_GatingLayout,ANY-method"
  .. .. .. ..$ : chr "%+%,ggcyto_GatingLayout-method"
  .. .. .. ..$ : chr "replace_data"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "scale_x_flowCore_fasinh"
  .. .. ..$ title : chr "Add a flowCore inverse hyperbolic sine scale to the x or y axes of a ggcyto plot."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "scale_x_flowCore_fasinh"
  .. .. .. ..$ : chr "scale_y_flowCore_fasinh"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "scale_x_logicle"
  .. .. ..$ title : chr "Add a logicle scale to the x or y axes of a ggcyto plot."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "scale_x_logicle"
  .. .. .. ..$ : chr "scale_y_logicle"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "scales_flowjo_biexp"
  .. .. ..$ title : chr "Add a flowJo biexponential scale to the x or y axes of a ggcyto plot."
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "scales_flowjo_biexp"
  .. .. .. ..$ : chr "scale_x_flowJo_biexp"
  .. .. .. ..$ : chr "scale_x_flowjo_biexp"
  .. .. .. ..$ : chr "scale_y_flowJo_biexp"
  .. .. .. ..$ : chr "scale_y_flowjo_biexp"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "scales_flowjo_fasinh"
  .. .. ..$ title : chr "Add a flowJo inverse hyperbolic sine scale to the x or y axes of a ggcyto plot."
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "scales_flowjo_fasinh"
  .. .. .. ..$ : chr "scale_x_flowJo_fasinh"
  .. .. .. ..$ : chr "scale_x_flowjo_fasinh"
  .. .. .. ..$ : chr "scale_y_flowJo_fasinh"
  .. .. .. ..$ : chr "scale_y_flowjo_fasinh"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "stat_position"
  .. .. ..$ title : chr "Compute the positions of the population statistics based on the geometic gate centroid for a ggcyto plot."
  .. .. ..$ topics:List of 4
  .. .. .. ..$ : chr "stat_position"
  .. .. .. ..$ : chr "stat_position.filter"
  .. .. .. ..$ : chr "stat_position.filterList"
  .. .. .. ..$ : chr "stat_position.list"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "stats_null"
  .. .. ..$ title : chr "clear all the geom_stats() layer previously added"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "stats_null"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "transform-gate"
  .. .. ..$ title : chr "rescale methods for gates"
  .. .. ..$ topics:List of 9
  .. .. .. ..$ : chr "rescale_gate"
  .. .. .. ..$ : chr "rescale_gate.ellipsoidGate"
  .. .. .. ..$ : chr "rescale_gate.polygonGate"
  .. .. .. ..$ : chr "rescale_gate.quadGate"
  .. .. .. ..$ : chr "rescale_gate.rectangleGate"
  .. .. .. ..$ : chr "transform"
  .. .. .. ..$ : chr "transform,filter-method"
  .. .. .. ..$ : chr "transform,filterList-method"
  .. .. .. ..$ : chr "transform-gate"
  ..$ _pkglogo          : chr "https://github.com/bioc/ggcyto/raw/HEAD/logo.png"
  ..$ _readme           : chr "https://github.com/bioc/ggcyto/raw/HEAD/README.md"
  ..$ _rundeps          :List of 54
  .. ..$ : chr "abind"
  .. ..$ : chr "BH"
  .. ..$ : chr "Biobase"
  .. ..$ : chr "BiocGenerics"
  .. ..$ : chr "cli"
  .. ..$ : chr "cpp11"
  .. ..$ : chr "cytolib"
  .. ..$ : chr "data.table"
  .. ..$ : chr "DelayedArray"
  .. ..$ : chr "dplyr"
  .. ..$ : chr "farver"
  .. ..$ : chr "flowCore"
  .. ..$ : chr "flowWorkspace"
  .. ..$ : chr "generics"
  .. ..$ : chr "ggplot2"
  .. ..$ : chr "glue"
  .. ..$ : chr "graph"
  .. ..$ : chr "gridExtra"
  .. ..$ : chr "gtable"
  .. ..$ : chr "hexbin"
  .. ..$ : chr "IRanges"
  .. ..$ : chr "isoband"
  .. ..$ : chr "labeling"
  .. ..$ : chr "lattice"
  .. ..$ : chr "lifecycle"
  .. ..$ : chr "magrittr"
  .. ..$ : chr "Matrix"
  .. ..$ : chr "MatrixGenerics"
  .. ..$ : chr "matrixStats"
  .. ..$ : chr "ncdfFlow"
  .. ..$ : chr "pillar"
  .. ..$ : chr "pkgconfig"
  .. ..$ : chr "plyr"
  .. ..$ : chr "R6"
  .. ..$ : chr "RBGL"
  .. ..$ : chr "RColorBrewer"
  .. ..$ : chr "Rcpp"
  .. ..$ : chr "Rgraphviz"
  .. ..$ : chr "Rhdf5lib"
  .. ..$ : chr "rlang"
  .. ..$ : chr "RProtoBufLib"
  .. ..$ : chr "S4Arrays"
  .. ..$ : chr "S4Vectors"
  .. ..$ : chr "S7"
  .. ..$ : chr "scales"
  .. ..$ : chr "SparseArray"
  .. ..$ : chr "tibble"
  .. ..$ : chr "tidyselect"
  .. ..$ : chr "utf8"
  .. ..$ : chr "vctrs"
  .. ..$ : chr "viridisLite"
  .. ..$ : chr "withr"
  .. ..$ : chr "XML"
  .. ..$ : chr "XVector"
  ..$ _vignettes        :List of 4
  .. ..$ :List of 8
  .. .. ..$ source  : chr "Top_features_of_ggcyto.Rmd"
  .. .. ..$ filename: chr "Top_features_of_ggcyto.html"
  .. .. ..$ title   : chr "ggcyto : Visualize Cytometry data with ggplot"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 16
  .. .. .. ..$ : chr "1: support 3 types of plot constructor"
  .. .. .. ..$ : chr "low level: ggplot"
  .. .. .. ..$ : chr "GatingSet"
  .. .. .. ..$ : chr "flowSet/ncdfFlowSet/flowFrame"
  .. .. .. ..$ : chr "gates"
  .. .. .. ..$ : chr "medium level: ggcyto"
  .. .. .. ..$ : chr "top level: autoplot"
  .. .. .. ..$ : chr "2: in-line transformation"
  .. .. .. ..$ : chr "3: generic geom_gate layer"
  .. .. .. ..$ : chr "4: geom_stats"
  .. .. .. ..$ : chr "5: axis_inverse_trans"
  .. .. .. ..$ : chr "6: auto limits"
  .. .. .. ..$ : chr "7: labs_cyto"
  .. .. .. ..$ : chr "8: ggcyto_par_set"
  .. .. .. ..$ : chr "9: as.ggplot"
  .. .. .. ..$ : chr "10: ggcyto_layout"
  .. .. ..$ created : chr "2016-05-16 22:36:36"
  .. .. ..$ modified: chr "2023-08-20 22:53:18"
  .. .. ..$ commits : int 11
  .. ..$ :List of 8
  .. .. ..$ source  : chr "autoplot.Rmd"
  .. .. ..$ filename: chr "autoplot.html"
  .. .. ..$ title   : chr "Quick plot for cytometry data"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 5
  .. .. .. ..$ : chr "flowSet"
  .. .. .. ..$ : chr "flowFrame"
  .. .. .. ..$ : chr "GatingSet"
  .. .. .. ..$ : chr "GatingHierarchy"
  .. .. .. ..$ : chr "ggcyto_arrange"
  .. .. ..$ created : chr "2015-05-05 23:37:51"
  .. .. ..$ modified: chr "2019-05-02 18:34:19"
  .. .. ..$ commits : int 13
  .. ..$ :List of 8
  .. .. ..$ source  : chr "ggcyto.flowSet.Rmd"
  .. .. ..$ filename: chr "ggcyto.flowSet.html"
  .. .. ..$ title   : chr "Visualize flowSet with ggcyto"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 4
  .. .. .. ..$ : chr "1d histogram/densityplot"
  .. .. .. ..$ : chr "stacked density plot"
  .. .. .. ..$ : chr "2d scatter/dot plot"
  .. .. .. ..$ : chr "Add geom_gate and geom_stats layers"
  .. .. ..$ created : chr "2015-05-15 19:07:04"
  .. .. ..$ modified: chr "2019-05-09 19:03:01"
  .. .. ..$ commits : int 13
  .. ..$ :List of 8
  .. .. ..$ source  : chr "ggcyto.GatingSet.Rmd"
  .. .. ..$ filename: chr "ggcyto.GatingSet.html"
  .. .. ..$ title   : chr "Visualize GatingSet with ggcyto"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 6
  .. .. .. ..$ : chr "ggcyto_par_set"
  .. .. .. ..$ : chr "geom_gate"
  .. .. .. ..$ : chr "geom_stats"
  .. .. .. ..$ : chr "geom_overlay"
  .. .. .. ..$ : chr "subset"
  .. .. .. ..$ : chr "axis_x_inverse_trans"
  .. .. ..$ created : chr "2015-05-15 19:07:04"
  .. .. ..$ modified: chr "2016-07-20 23:59:03"
  .. .. ..$ commits : int 23
  ..$ _score            : num 12.1
  ..$ _indexed          : logi TRUE
  ..$ _nocasepkg        : chr "ggcyto"
  ..$ _universes        :List of 3
  .. ..$ : chr "bioc"
  .. ..$ : chr "mikejiang"
  .. ..$ : chr "rglab"
  ..$ _previous         : chr "1.39.1"
  ..$ _binaries         :List of 8
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:07:42.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "be1b394a498b15b1feae729cdc30bc5ba454dcd30b4dd8cabf5e4e7b1c9303db"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:07:29.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "0710085f14cc1b212343602c72001ff9502ff69690510db92be1db95832c15ea"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:06:09.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "533ad6c1c410cde1980f63f3e57c9f907f948673e54dc8124b651243378a113f"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:06:08.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "ed7ce67e44ba879a551eb9c3d7ff36da78c8782b61cf9f828655ecbb6359a104"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 8
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "wasm"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:07:11.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "f91fcb0a190e4f53802d165d202759b5aacf34ab9e1826e75c622f5a0807bfc4"
  .. .. ..$ status  : chr "success"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:06:26.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "dc0963faa15e51663618d8555f6d05b478527f97ada4d7ee524e35820ae9db65"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:06:46.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "4a78f1b69f7712c0d6e0a5eb5f00b2db98a885e9cb08f6df046633188fb26f4e"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.39.2"
  .. .. ..$ date    : chr "2026-02-19T06:06:29.000Z"
  .. .. ..$ commit  : chr "a6bc5e45e4e31585d81e5952e8e12d1db956c30e"
  .. .. ..$ fileid  : chr "0ae54af8aae416685b0568eafc2cdaa2c3c63484e9d89ecfad0b948a5f366b0f"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/22170470815"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>