<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>exploreBiocUniverse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="exploreBiocUniverse_files/libs/clipboard/clipboard.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/popper.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/anchor.min.js"></script>
<link href="exploreBiocUniverse_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="exploreBiocUniverse_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-6650302f782765824de81675458e0c02.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="exploreBiocUniverse_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="exploreBiocUniverse_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">exploreBiocUniverse</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="comparing-bioconductor-packages-in-r-universe-and-the-bbs" class="level1">
<h1>Comparing Bioconductor packages in R-Universe and the BBS</h1>
<p>Skip to <a href="#packages-not-in-agreement">Packages not in agreements</a> or go to <a href="windows.qmd">windows</a>.</p>
<section id="r-universe" class="level2">
<h2 class="anchored" data-anchor-id="r-universe">R Universe</h2>
<p>We can use the <a href="https://github.com/ropensci/universe/">universe</a> package to explore the R-Universe (RU) API. (We might need to increase the <code>max.print</code> to show all packages.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universe)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">max.print =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data for the Bioc Universe (BU).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs_i <span class="ot">&lt;-</span> <span class="fu">universe_all_packages</span>(<span class="st">"bioc"</span>, <span class="at">limit =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What packages are in bioc universe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs <span class="ot">&lt;-</span> <span class="fu">universe_ls</span>(<span class="st">"bioc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bbs" class="level2">
<h2 class="anchored" data-anchor-id="bbs">BBS</h2>
<p>Get Bioconductor packages in BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocPkgTools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bbs_pkgs_i <span class="ot">&lt;-</span> BiocPkgTools<span class="sc">::</span><span class="fu">biocBuildReport</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">version =</span> BiocManager<span class="sc">::</span><span class="fu">version</span>(),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgType =</span> <span class="st">"software"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>bbs_pkgs <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unique</span>(bbs_pkgs_i<span class="sc">$</span>pkg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Packages in RU vs BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">RU =</span> <span class="fu">length</span>(ru_pkgs), <span class="at">BBS =</span> <span class="fu">length</span>(bbs_pkgs),</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"Total"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        RU  BBS
Total 2327 2361</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>not_in_ru <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(bbs_pkgs, ru_pkgs)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>not_in_bbs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(ru_pkgs, bbs_pkgs)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># There's probably a better way to do this</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">Created =</span> <span class="fu">character</span>(), <span class="at">Published =</span> <span class="fu">character</span>(),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">RemoteUrl =</span> <span class="fu">character</span>())</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> not_in_bbs) {</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  pi <span class="ot">&lt;-</span> <span class="fu">universe_one_package</span>(<span class="st">"bioc"</span>, pkg)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  pkgs_not_in_bbs <span class="ot">&lt;-</span> pkgs_not_in_bbs <span class="sc">|&gt;</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> pi<span class="sc">$</span>Package, <span class="at">Version =</span> pi<span class="sc">$</span>Version,</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">Created =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_created</span><span class="st">`</span>, <span class="at">Published =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_published</span><span class="st">`</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">RemoteUrl =</span> pi<span class="sc">$</span>RemoteUrl)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-not-in-bbs-but-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-not-in-bbs-but-in-ru">Packages not in BBS but in RU</h3>
<p>Packages in RU but not in the BBS are likely added by maintainers via <a href="https://github.com/r-universe/bioc/blob/master/.remotes.json" class="uri">https://github.com/r-universe/bioc/blob/master/.remotes.json</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="sc">|&gt;</span> knitr<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Created</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Published</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RemoteUrl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LACHESIS</td>
<td style="text-align: left;">0.99.4</td>
<td style="text-align: left;">2025-11-13T03:46:38.000Z</td>
<td style="text-align: left;">2025-11-13T04:21:22.709Z</td>
<td style="text-align: left;">https://github.com/bioc/LACHESIS</td>
</tr>
<tr class="even">
<td style="text-align: left;">SpiecEasi</td>
<td style="text-align: left;">1.99.3</td>
<td style="text-align: left;">2025-11-06T03:26:26.000Z</td>
<td style="text-align: left;">2025-11-06T03:38:14.161Z</td>
<td style="text-align: left;">https://github.com/bioc/SpiecEasi</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dominatR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2025-11-19T08:56:08.000Z</td>
<td style="text-align: left;">2025-11-19T09:02:29.790Z</td>
<td style="text-align: left;">https://github.com/bioc/dominatR</td>
</tr>
<tr class="even">
<td style="text-align: left;">ggpattern</td>
<td style="text-align: left;">1.1.4</td>
<td style="text-align: left;">2025-11-11T06:57:27.000Z</td>
<td style="text-align: left;">2025-11-11T07:10:04.396Z</td>
<td style="text-align: left;">https://github.com/coolbutuseless/ggpattern</td>
</tr>
<tr class="odd">
<td style="text-align: left;">leapR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2025-11-13T03:42:31.000Z</td>
<td style="text-align: left;">2025-11-13T03:50:07.840Z</td>
<td style="text-align: left;">https://github.com/bioc/leapR</td>
</tr>
<tr class="even">
<td style="text-align: left;">lmms</td>
<td style="text-align: left;">1.3.3</td>
<td style="text-align: left;">2025-10-23T03:46:27.000Z</td>
<td style="text-align: left;">2025-10-23T03:52:01.356Z</td>
<td style="text-align: left;">https://github.com/cran/lmms</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="packages-in-bbs-but-not-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-in-bbs-but-not-in-ru">Packages in BBS but not in RU</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>not_in_ru</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "bgx"                    "BiGGR"                  "biodbHmdb"             
 [4] "biodbNcbi"              "biodbNci"               "biodbUniprot"          
 [7] "ccmap"                  "CellScore"              "CINdex"                
[10] "cisPath"                "DEP"                    "gpuMagic"              
[13] "Harshlight"             "hiAnnotator"            "hiReadsProcessor"      
[16] "hypeR"                  "interactiveDisplay"     "interactiveDisplayBase"
[19] "lapmix"                 "LinTInd"                "lute"                  
[22] "MADSEQ"                 "netZooR"                "oppti"                 
[25] "PhenStat"               "qckitfastq"             "ReactomeGraph4R"       
[28] "Repitools"              "Rfastp"                 "rGADEM"                
[31] "rRDP"                   "SARC"                   "seqArchR"              
[34] "seqArchRplus"           "seqTools"               "Streamer"              
[37] "TitanCNA"               "TransView"              "traviz"                
[40] "XNAString"             </code></pre>
</div>
</div>
<p>The following packages are not deprecated but are not in RU.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>depre_pkgs <span class="ot">&lt;-</span> <span class="fu">filter</span>(bbs_pkgs_i, Deprecated <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(pkg) <span class="sc">|&gt;</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(not_in_ru, depre_pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "bgx"          "biodbHmdb"    "biodbNcbi"    "biodbNci"     "biodbUniprot"
[6] "cisPath"      "hiAnnotator"  "seqArchRplus" "traviz"      </code></pre>
</div>
</div>
</section>
</section>
<section id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc." class="level2">
<h2 class="anchored" data-anchor-id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc.">Making sense of R Universe’s build status in respect to R CMD build, check, etc.</h2>
<p>Looking at the status of linux x86_64 binary builds. The build “status” appears to get “success”, “failure”, or “cancelled” from the GitHub Action overall. The “check” value corresponds to <code>R CMD check</code>. I interpret the result of “status” to be “success” OK and “failure” ERROR for <code>R CMD build</code> but it may mean something else. We’ll use CANCELLED as it is. There’s also a special status for documentation, which is handled separately from build/check, which may account for some discrepancies. Not sure if the build status for vignettes is available in the API.</p>
<p>Packages with my comment “Missing R4.6 on Linux” did have data in the API but no x86_64 linux with R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Status =</span> <span class="fu">character</span>(), <span class="at">Build =</span> <span class="fu">character</span>(),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Check =</span> <span class="fu">character</span>(), <span class="at">Comment =</span> <span class="fu">character</span>())</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#, Buildurl = character(),</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># Commit = character())</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> ru_pkgs_i) {</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> <span class="cn">NA</span>, <span class="at">build =</span> <span class="cn">NA</span>, <span class="at">check =</span> <span class="cn">NA</span>, <span class="at">buildurl =</span> <span class="cn">NA</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  comment <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (b <span class="cf">in</span> p<span class="sc">$</span><span class="st">`</span><span class="at">_binaries</span><span class="st">`</span>) {</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (b<span class="sc">$</span>os <span class="sc">==</span> <span class="st">"linux"</span> <span class="sc">&amp;&amp;</span> b<span class="sc">$</span>r <span class="sc">==</span> <span class="st">"4.6.0"</span>) {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>      build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> b<span class="sc">$</span>status, <span class="at">check =</span> b<span class="sc">$</span>check, <span class="at">buildurl =</span> b<span class="sc">$</span>buildurl)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(build[<span class="st">"status"</span>])) {</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> <span class="st">"Missing R4.6 on Linux"</span> </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"success"</span>) {</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"failure"</span>) {</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"ERROR*"</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"cancelled"</span>) {</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"CANCELLED"</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  ru_status <span class="ot">&lt;-</span> ru_status <span class="sc">|&gt;</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> p<span class="sc">$</span>Package, <span class="at">Version =</span> p<span class="sc">$</span>Version, <span class="at">Status =</span> build[<span class="st">"status"</span>],</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">Build =</span> build[<span class="st">"build"</span>], <span class="at">Check =</span> build[<span class="st">"check"</span>], <span class="at">Comment =</span> comment)</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>            <span class="co">#, Buildurl = build["buildurl"], Commit = p$`_commit`$id)</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co"># order by Package</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> ru_status[<span class="fu">order</span>(ru_status<span class="sc">$</span>Package), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>DEVEL_LINUX_BUILDER <span class="ot">=</span> <span class="st">"nebbiolo2"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper functions</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>get_result <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  entry <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pkg <span class="sc">==</span> x[<span class="st">"pkg"</span>], stage <span class="sc">==</span> x[<span class="st">"stage"</span>], node <span class="sc">==</span> x[<span class="st">"node"</span>])</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(entry)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    entry <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    entry<span class="sc">$</span>result</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>get_install <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"install"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>get_build <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"buildsrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>get_check <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"checksrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>bbs_status <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(node <span class="sc">==</span> DEVEL_LINUX_BUILDER) <span class="sc">|&gt;</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version, Deprecated) <span class="sc">|&gt;</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(pkg, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">install =</span> <span class="fu">map_chr</span>(pkg, get_install)) <span class="sc">|&gt;</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">build =</span> <span class="fu">map_chr</span>(pkg, get_build)) <span class="sc">|&gt;</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">check =</span> <span class="fu">map_chr</span>(pkg, get_check))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-the-status-of-packages-in-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-status-of-packages-in-ru-and-bbs">Compare the status of packages in RU and BBS</h3>
<p>We can only compare <code>check</code> and <code>build</code>, so we’ll check if those columns agree. Note that in the BBS, <code>check</code> can show “WARNINGS” whereas R Universe can show “WARNING” (no s) as well as “OK”, “NOTE”, “ERROR”, or NA.</p>
<p>Note: - Deprecated packages are removed immedately in R Universe. - Packages may have different versions since R-Universe builds new versions after detecting a “push”.</p>
<p>The following includes packages not deprecated in BBS where build and check don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ru_status) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pkg"</span>, <span class="st">"version"</span>, <span class="st">"status"</span>, <span class="st">"build"</span>, <span class="st">"check"</span>, <span class="st">"comment"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>bbs_ru_comparison <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> bbs_status, <span class="at">y =</span> ru_status, <span class="at">by =</span> <span class="st">"pkg"</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">&amp;</span> Deprecated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(pkg, install, build.x, build.y, check.x, check.y, version.x,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                  version.y, status, comment)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>, <span class="st">"RU_build"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"BBS_version"</span>, <span class="st">"RU_version"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-without-x86_64-linux-builds-for-r-4.6" class="level4">
<h4 class="anchored" data-anchor-id="packages-without-x86_64-linux-builds-for-r-4.6">Packages without x86_64 Linux builds for R 4.6</h4>
<p>The following are packages in bioc R Universe but without an x86_64 Linux build for R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Comment <span class="sc">==</span> <span class="st">"Missing R4.6 on Linux"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages in RU without an x86_64 linux build for R 4.6"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages in RU without an x86_64 linux build for R 4.6</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CONFESS</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">ddPCRclust</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.30.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">debCAM</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.28.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">GEOmetadb</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.72.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HilbertVisGUI</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.68.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">motifbreakR</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.24.0</td>
<td style="text-align: left;">2.23.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MungeSumstats</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.18.0</td>
<td style="text-align: left;">1.19.1</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">SamSPECTRAL</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.64.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SwathXtend</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Notes: - Retrying to refresh the build on the dashboard failed for some packages. How can we rebuild a package that hasn’t received a push in 3 months? - paxtoolsr should be in Bioc R Universe, but it isn’t. Why?</p>
</section>
</section>
</section>
<section id="packages-not-in-agreement" class="level2">
<h2 class="anchored" data-anchor-id="packages-not-in-agreement">Packages not in agreement</h2>
<p>Looking at packages with the same version and removing deprecated packages, we have the following packages where the BBS and R Universe’s results don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Deprecated <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> version.x <span class="sc">==</span> version.y <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(status)) <span class="sc">|&gt;</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version.x, install, build.x, build.y, check.x, check.y,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         status, comment)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_version"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"RU_build"</span>, <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="sc">|&gt;</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages with different build results between BBS and RU"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages with different build results between BBS and RU</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Note: <code>ERROR*</code> could be an <code>ERROR</code>, <code>TIMEOUT</code>, or <code>skipped</code>.</p>
<section id="total-packages-not-in-agreement" class="level3">
<h3 class="anchored" data-anchor-id="total-packages-not-in-agreement">Total packages not in agreement</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bbs_ru_not_ok2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="other" class="level1">
<h1>Other</h1>
<p>If a package repo isn’t accessible or an error exists in the DESCRIPTION file, there will be an error in the registry for that submodule and it will show up at <a href="https://github.com/r-universe/bioc/actions/workflows/sync.yml" class="uri">https://github.com/r-universe/bioc/actions/workflows/sync.yml</a>. The red triangle on <a href="https://bioc.r-universe.dev/" class="uri">https://bioc.r-universe.dev/</a> will take you to the specific workflow error.</p>
<p>Currently, this information is available on the BBS but I wonder if this information is available somewhere for R Universe OR if it would be necessary given that developers would be able to inspect their builds sooner.</p>
<section id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api" class="level3">
<h3 class="anchored" data-anchor-id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api">Important: Use the dbdump endpoint with read_bson to dump the API</h3>
<p>Use <code>read_bson</code> to dump API data into a db</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("mongolite")</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mongolite)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>bioc <span class="ot">&lt;-</span> mongolite<span class="sc">::</span><span class="fu">read_bson</span>(<span class="st">'https://bioc.r-universe.dev/api/dbdump'</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: it returns 2268 records rather than 2276 (fewer retrieved through</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co"># another endpoint)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-a-tarball-from-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="comparing-a-tarball-from-ru-and-bbs">Comparing a tarball from RU and BBS</h3>
<p>Lastly, when I look at the builds, they’re using various repositories, such p3m, r-universe, etc. This might be okay as long as the resulting binaries/tarballs are the generally the same. Looking at XVector for both, both the BBS and R Universe inject some additional information. R Universe also creates an <code>extra</code> directory that has citations, manuals, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2 ruXVector_0.47.2</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2/XVector/DESCRIPTION ruXVector_0.47.2/XVector/DESCRIPTION</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">5c5</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>   <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span>         <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="ex">28,33c28,31</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_url: <span class="ex">https://git.bioconductor.org/packages/XVector</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_branch: <span class="ex">devel</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit: <span class="ex">9f44218</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit_date: <span class="ex">2025-01-07</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Repository: <span class="ex">Bioconductor</span> 3.21</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Date/Publication: <span class="ex">2025-01-08</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Repository: <span class="ex">https://bioc.r-universe.dev</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteUrl: <span class="ex">https://github.com/bioc/XVector</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteRef: <span class="ex">HEAD</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteSha: <span class="ex">9f44218308f9d5867f0b9df5fae60d217f4e7d65</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="ex">35c33</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Packaged: <span class="ex">2025-01-09</span> 00:57:29 UTC<span class="kw">;</span> <span class="ex">biocbuild</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Packaged: <span class="ex">2025-01-08</span> 03:05:41 UTC<span class="kw">;</span> <span class="ex">root</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: extra</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: manual.pdf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="structure-of-a-package-through-universe" class="level3">
<h3 class="anchored" data-anchor-id="structure-of-a-package-through-universe">Structure of a package through universe</h3>
<p>The structure of a package looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">str</span>(ru_pkgs_i[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ :List of 73
  ..$ Package           : chr "ORFik"
  ..$ Type              : chr "Package"
  ..$ Title             : chr "Open Reading Frames in Genomics"
  ..$ Version           : chr "1.31.1"
  ..$ Encoding          : chr "UTF-8"
  ..$ Authors@R         : chr "c(\nperson(given = \"Haakon\", family = \"Tjeldnes\", email = \"hauken_heyken@hotmail.com\", role = c(\"aut\", "| __truncated__
  ..$ Description       : chr "R package for analysis of transcript and translation\nfeatures through manipulation of sequence data and NGS da"| __truncated__
  ..$ biocViews         : chr "ImmunoOncology, Software, Sequencing, RiboSeq, RNASeq,\nFunctionalGenomics, Coverage, Alignment, DataImport"
  ..$ License           : chr "MIT + file LICENSE"
  ..$ BugReports        : chr "https://github.com/Roleren/ORFik/issues"
  ..$ URL               : chr "https://github.com/Roleren/ORFik"
  ..$ RoxygenNote       : chr "7.3.3"
  ..$ VignetteBuilder   : chr "knitr"
  ..$ Config/pak/sysreqs: chr "make libbz2-dev libicu-dev liblzma-dev libpng-dev\nlibxml2-dev libssl-dev libx11-dev xz-utils zlib1g-dev"
  ..$ Repository        : chr "https://bioc.r-universe.dev"
  ..$ Date/Publication  : chr "2025-11-21 11:05:35 UTC"
  ..$ RemoteUrl         : chr "https://github.com/bioc/ORFik"
  ..$ RemoteRef         : chr "HEAD"
  ..$ RemoteSha         : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  ..$ NeedsCompilation  : chr "yes"
  ..$ Packaged          :List of 2
  .. ..$ Date: chr "2025-11-21 15:48:44 UTC"
  .. ..$ User: chr "root"
  ..$ Author            : chr "Haakon Tjeldnes [aut, cre, dtc],\nKornel Labun [aut, cph],\nMichal Swirski [ctb],\nKatarzyna Chyzynska [ctb, dt"| __truncated__
  ..$ Maintainer        : chr "Haakon Tjeldnes &lt;hauken_heyken@hotmail.com&gt;"
  ..$ MD5sum            : chr "f14a16e878dd804341a2a8f36227ae94"
  ..$ _user             : chr "bioc"
  ..$ _type             : chr "src"
  ..$ _file             : chr "ORFik_1.31.1.tar.gz"
  ..$ _fileid           : chr "aa1742c6b38b5e545b4d8a0de20130affe64e839a34fedd46bd012458aee8e03"
  ..$ _filesize         : int 3944979
  ..$ _sha256           : chr "aa1742c6b38b5e545b4d8a0de20130affe64e839a34fedd46bd012458aee8e03"
  ..$ _created          : chr "2025-11-21T15:48:44.000Z"
  ..$ _published        : chr "2025-11-21T16:13:40.592Z"
  ..$ _distro           : chr "noble"
  ..$ _jobs             :List of 14
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 337
  .. .. ..$ config  : chr "bioc-checks"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: NULL
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 644
  .. .. ..$ config  : chr "linux-devel-arm64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642123660"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 650
  .. .. ..$ config  : chr "linux-devel-x86_64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642124277"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 606
  .. .. ..$ config  : chr "linux-release-arm64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642116868"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 639
  .. .. ..$ config  : chr "linux-release-x86_64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642122356"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 421
  .. .. ..$ config  : chr "macos-oldrel-arm64"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642082735"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 1308
  .. .. ..$ config  : chr "macos-oldrel-x86_64"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642256326"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 454
  .. .. ..$ config  : chr "macos-release-arm64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642088556"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 1141
  .. .. ..$ config  : chr "macos-release-x86_64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "4642221716"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 629
  .. .. ..$ config  : chr "source"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4642005364"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 252
  .. .. ..$ config  : chr "wasm-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4642051850"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 815
  .. .. ..$ config  : chr "windows-devel"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "4642155738"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 777
  .. .. ..$ config  : chr "windows-oldrel"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "4642148634"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.61e+10
  .. .. ..$ time    : int 797
  .. .. ..$ config  : chr "windows-release"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "4642152547"
  ..$ _buildurl         : chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  ..$ _status           : chr "success"
  ..$ _host             : chr "GitHub-Actions"
  ..$ _upstream         : chr "https://github.com/bioc/ORFik"
  ..$ _commit           :List of 5
  .. ..$ id       : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. ..$ author   : chr "Roleren &lt;hauken_heyken@hotmail.com&gt;"
  .. ..$ committer: chr "Roleren &lt;hauken_heyken@hotmail.com&gt;"
  .. ..$ message  : chr "Merge branch 'master' into devel\n"
  .. ..$ time     : int 1763723135
  ..$ _maintainer       :List of 4
  .. ..$ name : chr "Haakon Tjeldnes"
  .. ..$ email: chr "hauken_heyken@hotmail.com"
  .. ..$ login: chr "roleren"
  .. ..$ uuid : int 14344350
  ..$ _registered       : logi TRUE
  ..$ _dependencies     :List of 44
  .. ..$ :List of 3
  .. .. ..$ package: chr "R"
  .. .. ..$ version: chr "&gt;= 4.1.0"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "IRanges"
  .. .. ..$ version: chr "&gt;= 2.17.1"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GenomicRanges"
  .. .. ..$ version: chr "&gt;= 1.35.1"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GenomicAlignments"
  .. .. ..$ version: chr "&gt;= 1.19.0"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Rcpp"
  .. .. ..$ role   : chr "LinkingTo"
  .. ..$ :List of 3
  .. .. ..$ package: chr "AnnotationDbi"
  .. .. ..$ version: chr "&gt;= 1.45.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Biostrings"
  .. .. ..$ version: chr "&gt;= 2.51.1"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "biomaRt"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "biomartr"
  .. .. ..$ version: chr "&gt;= 1.0.7"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocFileCache"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "BiocGenerics"
  .. .. ..$ version: chr "&gt;= 0.29.1"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "BiocParallel"
  .. .. ..$ version: chr "&gt;= 1.19.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "cowplot"
  .. .. ..$ version: chr "&gt;= 1.0.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "data.table"
  .. .. ..$ version: chr "&gt;= 1.11.8"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "DESeq2"
  .. .. ..$ version: chr "&gt;= 1.24.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "fst"
  .. .. ..$ version: chr "&gt;= 0.9.2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GenomeInfoDb"
  .. .. ..$ version: chr "&gt;= 1.15.5"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GenomicFeatures"
  .. .. ..$ version: chr "&gt;= 1.31.10"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "ggplot2"
  .. .. ..$ version: chr "&gt;= 2.2.1"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "gridExtra"
  .. .. ..$ version: chr "&gt;= 2.3"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "httr"
  .. .. ..$ version: chr "&gt;= 1.3.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "jsonlite"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "methods"
  .. .. ..$ version: chr "&gt;= 3.6.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "qs"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "R.utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Rcpp"
  .. .. ..$ version: chr "&gt;= 1.0.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Rsamtools"
  .. .. ..$ version: chr "&gt;= 1.35.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "rtracklayer"
  .. .. ..$ version: chr "&gt;= 1.43.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "SummarizedExperiment"
  .. .. ..$ version: chr "&gt;= 1.14.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "S4Vectors"
  .. .. ..$ version: chr "&gt;= 0.21.3"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tools"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "txdbmaker"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "XML"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "xml2"
  .. .. ..$ version: chr "&gt;= 1.2.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "withr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "testthat"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rmarkdown"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "knitr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocStyle"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Hsapiens.UCSC.hg19"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GenomeInfoDbData"
  .. .. ..$ role   : chr "Suggests"
  ..$ _owner            : chr "bioc"
  ..$ _selfowned        : logi TRUE
  ..$ _usedby           : int 2
  ..$ _updates          :List of 16
  .. ..$ :List of 2
  .. .. ..$ week: chr "2024-48"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2024-49"
  .. .. ..$ n   : int 3
  .. ..$ :List of 2
  .. .. ..$ week: chr "2024-50"
  .. .. ..$ n   : int 8
  .. ..$ :List of 2
  .. .. ..$ week: chr "2024-51"
  .. .. ..$ n   : int 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-05"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-06"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-07"
  .. .. ..$ n   : int 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-08"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-16"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-19"
  .. .. ..$ n   : int 3
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-21"
  .. .. ..$ n   : int 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-22"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-25"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-29"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-38"
  .. .. ..$ n   : int 3
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-47"
  .. .. ..$ n   : int 4
  ..$ _tags             : list()
  ..$ _bioc             :List of 2
  .. ..$ :List of 3
  .. .. ..$ branch : chr "devel"
  .. .. ..$ version: chr "1.31.0"
  .. .. ..$ bioc   : chr "3.23"
  .. ..$ :List of 3
  .. .. ..$ branch : chr "release"
  .. .. ..$ version: chr "1.30.0"
  .. .. ..$ bioc   : chr "3.22"
  ..$ _topics           :List of 10
  .. ..$ : chr "immunooncology"
  .. ..$ : chr "software"
  .. ..$ : chr "sequencing"
  .. ..$ : chr "riboseq"
  .. ..$ : chr "rnaseq"
  .. ..$ : chr "functionalgenomics"
  .. ..$ : chr "coverage"
  .. ..$ : chr "alignment"
  .. ..$ : chr "dataimport"
  .. ..$ : chr "cpp"
  ..$ _stars            : int 34
  ..$ _contributors     :List of 12
  .. ..$ :List of 3
  .. .. ..$ user : chr "roleren"
  .. .. ..$ count: int 1141
  .. .. ..$ uuid : int 14344350
  .. ..$ :List of 3
  .. .. ..$ user : chr "jokinghero"
  .. .. ..$ count: int 50
  .. .. ..$ uuid : int 7996188
  .. ..$ :List of 3
  .. .. ..$ user : chr "nturaga"
  .. .. ..$ count: int 14
  .. .. ..$ uuid : int 2746443
  .. ..$ :List of 3
  .. .. ..$ user : chr "jwokaty"
  .. .. ..$ count: int 10
  .. .. ..$ uuid : int 1744257
  .. ..$ :List of 3
  .. .. ..$ user : chr "m-swirski"
  .. .. ..$ count: int 6
  .. .. ..$ uuid : int 38792637
  .. ..$ :List of 3
  .. .. ..$ user : chr "yeroslaviz"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 8927555
  .. ..$ :List of 3
  .. .. ..$ user : chr "hpages"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 8810451
  .. ..$ :List of 3
  .. .. ..$ user : chr "vobencha"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 2466173
  .. ..$ :List of 3
  .. .. ..$ user : chr "upgradedavid"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 43794473
  .. ..$ :List of 3
  .. .. ..$ user : chr "kayla-morrell"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 32339524
  .. ..$ :List of 3
  .. .. ..$ user : chr "przester"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 2692657
  .. ..$ :List of 3
  .. .. ..$ user : chr "suhassrinivasan"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 32346517
  ..$ _userbio          :List of 4
  .. ..$ uuid       : int 2286807
  .. ..$ type       : chr "organization"
  .. ..$ name       : chr "Bioconductor"
  .. ..$ description: chr "Software for the analysis and comprehension of high-throughput genomic data"
  ..$ _downloads        :List of 2
  .. ..$ count : int 602
  .. ..$ source: chr "https://www.bioconductor.org/packages/stats/bioc/ORFik"
  ..$ _mentions         : int 2
  ..$ _devurl           : chr "https://github.com/roleren/orfik"
  ..$ _searchresults    : int 160
  ..$ _rbuild           : chr "4.5.2"
  ..$ _assets           :List of 11
  .. ..$ : chr "extra/citation.cff"
  .. ..$ : chr "extra/citation.html"
  .. ..$ : chr "extra/citation.json"
  .. ..$ : chr "extra/citation.txt"
  .. ..$ : chr "extra/contents.json"
  .. ..$ : chr "extra/NEWS.html"
  .. ..$ : chr "extra/NEWS.txt"
  .. ..$ : chr "extra/ORFik.html"
  .. ..$ : chr "extra/readme.html"
  .. ..$ : chr "extra/readme.md"
  .. ..$ : chr "manual.pdf"
  ..$ _homeurl          : chr "https://github.com/roleren/orfik"
  ..$ _realowner        : chr "bioc"
  ..$ _cranurl          : logi FALSE
  ..$ _exports          :List of 247
  .. ..$ : chr "append_gene_symbols"
  .. ..$ : chr "artificial.orfs"
  .. ..$ : chr "assignTSSByCage"
  .. ..$ : chr "asTX"
  .. ..$ : chr "bamVarName"
  .. ..$ : chr "browseSRA"
  .. ..$ : chr "canonical_isoforms"
  .. ..$ : chr "codon_usage"
  .. ..$ : chr "codon_usage_exp"
  .. ..$ : chr "codon_usage_plot"
  .. ..$ : chr "collapse.fastq"
  .. ..$ : chr "collapseDuplicatedReads"
  .. ..$ : chr "combn.pairs"
  .. ..$ : chr "computeFeatures"
  .. ..$ : chr "config"
  .. ..$ : chr "config_file"
  .. ..$ : chr "config.exper"
  .. ..$ : chr "config.save"
  .. ..$ : chr "convert_bam_to_ofst"
  .. ..$ : chr "convert_to_bigWig"
  .. ..$ : chr "convert_to_covRle"
  .. ..$ : chr "convert_to_covRleList"
  .. ..$ : chr "convertLibs"
  .. ..$ : chr "convertToOneBasedRanges"
  .. ..$ : chr "countOverlapsW"
  .. ..$ : chr "countTable"
  .. ..$ : chr "countTable_regions"
  .. ..$ : chr "coverageByTranscriptFST"
  .. ..$ : chr "coverageHeatMap"
  .. ..$ : chr "coveragePerTiling"
  .. ..$ : chr "coverageScorings"
  .. ..$ : chr "covRle"
  .. ..$ : chr "covRleFromGR"
  .. ..$ : chr "covRleList"
  .. ..$ : chr "create.experiment"
  .. ..$ : chr "defineTrailer"
  .. ..$ : chr "DEG_gorilla"
  .. ..$ : chr "DEG_gorilla_copy_to_local"
  .. ..$ : chr "DEG_gorilla_local_load_data"
  .. ..$ : chr "DEG_gorilla_plot"
  .. ..$ : chr "DEG_model"
  .. ..$ : chr "DEG_model_results"
  .. ..$ : chr "DEG_model_simple"
  .. ..$ : chr "DEG.analysis"
  .. ..$ : chr "DEG.plot.static"
  .. ..$ : chr "design"
  .. ..$ : chr "detect_drive"
  .. ..$ : chr "detect_ribo_orfs"
  .. ..$ : chr "detectRibosomeShifts"
  .. ..$ : chr "disengagementScore"
  .. ..$ : chr "distToCds"
  .. ..$ : chr "distToTSS"
  .. ..$ : chr "download_gene_homologues"
  .. ..$ : chr "download_gene_info"
  .. ..$ : chr "download.SRA"
  .. ..$ : chr "download.SRA.metadata"
  .. ..$ : chr "DTEG.analysis"
  .. ..$ : chr "DTEG.plot"
  .. ..$ : chr "entropy"
  .. ..$ : chr "envExp"
  .. ..$ : chr "envExp&lt;-"
  .. ..$ : chr "exonsWithPseudoIntronsPerGroup"
  .. ..$ : chr "experiment"
  .. ..$ : chr "experiment.colors"
  .. ..$ : chr "export.bed12"
  .. ..$ : chr "export.bedoc"
  .. ..$ : chr "export.bigWig"
  .. ..$ : chr "export.fstwig"
  .. ..$ : chr "export.ofst"
  .. ..$ : chr "export.wiggle"
  .. ..$ : chr "extendLeaders"
  .. ..$ : chr "extendLeadersUntil"
  .. ..$ : chr "extendTrailers"
  .. ..$ : chr "extendTrailersUntil"
  .. ..$ : chr "f"
  .. ..$ : chr "file_ext_without_compression"
  .. ..$ : chr "filepath"
  .. ..$ : chr "filterExtremePeakGenes"
  .. ..$ : chr "filterTranscripts"
  .. ..$ : chr "fimport"
  .. ..$ : chr "find_url_ebi"
  .. ..$ : chr "findFa"
  .. ..$ : chr "findMapORFs"
  .. ..$ : chr "findORFs"
  .. ..$ : chr "findORFsFasta"
  .. ..$ : chr "findPeaksPerGene"
  .. ..$ : chr "findUORFs"
  .. ..$ : chr "findUORFs_exp"
  .. ..$ : chr "firstEndPerGroup"
  .. ..$ : chr "firstExonPerGroup"
  .. ..$ : chr "firstStartPerGroup"
  .. ..$ : chr "fix_malformed_gff"
  .. ..$ : chr "flankPerGroup"
  .. ..$ : chr "floss"
  .. ..$ : chr "fpkm"
  .. ..$ : chr "fractionLength"
  .. ..$ : chr "fread.bed"
  .. ..$ : chr "gcContent"
  .. ..$ : chr "geneToSymbol"
  .. .. [list output truncated]
  ..$ _help             :List of 311
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ORFik-package"
  .. .. ..$ title : chr "ORFik for analysis of open reading frames."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "ORFik-package"
  .. .. .. ..$ : chr "ORFik"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "add_pseudo_5utrs_txdb_if_needed"
  .. .. ..$ title : chr "add_pseudo_5utrs_txdb_if_needed"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "add_pseudo_5utrs_txdb_if_needed"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "append_gene_symbols"
  .. .. ..$ title : chr "Append gene symbols to a data.table with tx ids"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "append_gene_symbols"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "artificial.orfs"
  .. .. ..$ title : chr "Create small artificial orfs from cds"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "artificial.orfs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "as.character-GRangesList-method"
  .. .. ..$ title : chr "Convert GRangesList to character vector"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "as.character,GRangesList-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "assignTSSByCage"
  .. .. ..$ title : chr "Input a txdb and add a 5' leader for each transcript, that does not have one."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "assignTSSByCage"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "asTX"
  .. .. ..$ title : chr "Map genomic to transcript coordinates by reference"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "asTX"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "bamVarName"
  .. .. ..$ title : chr "Get library variable names from ORFik 'experiment'"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "bamVarName"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "browseSRA"
  .. .. ..$ title : chr "Open SRA in browser for specific bioproject"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "browseSRA"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "canonical_isoforms"
  .. .. ..$ title : chr "Get canonical isoforms of organism"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "canonical_isoforms"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "canonical_isoforms-experiment-method"
  .. .. ..$ title : chr "Get canonical isoforms of organism"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "canonical_isoforms,experiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "codon_usage"
  .. .. ..$ title : chr "Codon usage"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "codon_usage"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "codon_usage_exp"
  .. .. ..$ title : chr "Codon analysis for ORFik experiment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "codon_usage_exp"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "codon_usage_plot"
  .. .. ..$ title : chr "Plot codon_usage"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "codon_usage_plot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapse.fastq"
  .. .. ..$ title : chr "Very fast fastq/fasta collapser"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapse.fastq"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapseDuplicatedReads"
  .. .. ..$ title : chr "Collapse duplicated reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapseDuplicatedReads"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapseDuplicatedReads-data.table-method"
  .. .. ..$ title : chr "Collapse duplicated reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapseDuplicatedReads,data.table-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapseDuplicatedReads-GAlignmentPairs-method"
  .. .. ..$ title : chr "Collapse duplicated reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapseDuplicatedReads,GAlignmentPairs-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapseDuplicatedReads-GAlignments-method"
  .. .. ..$ title : chr "Collapse duplicated reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapseDuplicatedReads,GAlignments-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "collapseDuplicatedReads-GRanges-method"
  .. .. ..$ title : chr "Collapse duplicated reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "collapseDuplicatedReads,GRanges-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "combn.pairs"
  .. .. ..$ title : chr "Create all unique combinations pairs possible"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "combn.pairs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "computeFeatures"
  .. .. ..$ title : chr "Get all main features in ORFik"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "computeFeatures"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "computeFeaturesCage"
  .. .. ..$ title : chr "Get all main features in ORFik"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "computeFeaturesCage"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "config"
  .. .. ..$ title : chr "Read directory config for ORFik experiments"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "config"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "config_file"
  .. .. ..$ title : chr "Get path for ORFik config in cache"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "config_file"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "config.exper"
  .. .. ..$ title : chr "Set directories for experiment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "config.exper"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "config.save"
  .. .. ..$ title : chr "Save/update directory config for ORFik experiments"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "config.save"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert_bam_to_ofst"
  .. .. ..$ title : chr "Convert libraries to ofst"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert_bam_to_ofst"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert_to_bigWig"
  .. .. ..$ title : chr "Convert to BigWig"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert_to_bigWig"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert_to_covRle"
  .. .. ..$ title : chr "Convert libraries to covRle"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert_to_covRle"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert_to_covRleList"
  .. .. ..$ title : chr "Convert libraries to covRleList objects"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert_to_covRleList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert_to_fstWig"
  .. .. ..$ title : chr "Convert to fstwig"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert_to_fstWig"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convertLibs"
  .. .. ..$ title : chr "Converted format of NGS libraries"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convertLibs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convertToOneBasedRanges"
  .. .. ..$ title : chr "Convert a GRanges Object to 1 width reads"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convertToOneBasedRanges"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "cor_plot"
  .. .. ..$ title : chr "Get correlation between columns"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "cor_plot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "cor_table"
  .. .. ..$ title : chr "Get correlation between columns"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "cor_table"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "correlation.plots"
  .. .. ..$ title : chr "Correlation plots between all samples"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "correlation.plots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "countOverlapsW"
  .. .. ..$ title : chr "CountOverlaps with weights"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "countOverlapsW"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "countTable"
  .. .. ..$ title : chr "Extract count table directly from experiment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "countTable"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "countTable_regions"
  .. .. ..$ title : chr "Make a list of count matrices from experiment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "countTable_regions"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverage_to_dt"
  .. .. ..$ title : chr "Convert coverage RleList to data.table"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverage_to_dt"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverageByTranscriptC"
  .. .. ..$ title : chr "coverageByTranscript with coverage input"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverageByTranscriptC"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverageByTranscriptFST"
  .. .. ..$ title : chr "Get coverage from fst large coverage format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverageByTranscriptFST"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverageByTranscriptW"
  .. .. ..$ title : chr "coverageByTranscript with weights"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverageByTranscriptW"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverageHeatMap"
  .. .. ..$ title : chr "Create a heatmap of coverage"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverageHeatMap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coveragePerTiling"
  .. .. ..$ title : chr "Get coverage per group"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coveragePerTiling"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "coverageScorings"
  .. .. ..$ title : chr "Add a coverage scoring scheme"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "coverageScorings"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "covRle"
  .. .. ..$ title : chr "Coverage Rlelist for both strands"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "covRle"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "covRle-class"
  .. .. ..$ title : chr "Coverage Rle for both strands or single"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "covRle-class"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "covRleFromGR"
  .. .. ..$ title : chr "Convert GRanges to covRle"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "covRleFromGR"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "covRleList"
  .. .. ..$ title : chr "Coverage Rlelist for both strands"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "covRleList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "covRleList-class"
  .. .. ..$ title : chr "List of covRle"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "covRleList-class"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "create.experiment"
  .. .. ..$ title : chr "Create an ORFik 'experiment'"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "create.experiment"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "defineTrailer"
  .. .. ..$ title : chr "Defines trailers for ORF."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "defineTrailer"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_gorilla"
  .. .. ..$ title : chr "GO analysis with GOrilla"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_gorilla"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_gorilla_copy_to_local"
  .. .. ..$ title : chr "Copy GOrilla result htmls to local"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_gorilla_copy_to_local"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_gorilla_local_load_data"
  .. .. ..$ title : chr "Load all GOrilla xls files from study"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_gorilla_local_load_data"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_gorilla_plot"
  .. .. ..$ title : chr "Plot GOrilla xls results"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_gorilla_plot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_model"
  .. .. ..$ title : chr "Get DESeq2 model without running results"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_model"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_model_results"
  .. .. ..$ title : chr "Get DESeq2 model results from DESeqDataSet"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_model_results"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG_model_simple"
  .. .. ..$ title : chr "Simple Fpkm ratio test DEG"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG_model_simple"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG.analysis"
  .. .. ..$ title : chr "Run differential TE analysis"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG.analysis"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DEG.plot.static"
  .. .. ..$ title : chr "Plot DEG result"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DEG.plot.static"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "design-experiment-method"
  .. .. ..$ title : chr "Get experimental design Find the column/columns that create a separation between samples, by default skips repl"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "design,experiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "detect_drive"
  .. .. ..$ title : chr "Detects the mounted drive based on a mounted path"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "detect_drive"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "detect_ribo_orfs"
  .. .. ..$ title : chr "Detect ORFs by Ribosome profiling data"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "detect_ribo_orfs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "detectRibosomeShifts"
  .. .. ..$ title : chr "Detect ribosome shifts"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "detectRibosomeShifts"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "disengagementScore"
  .. .. ..$ title : chr "Disengagement score (DS)"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "disengagementScore"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "distanceToFollowing"
  .. .. ..$ title : chr "Distance to following range group"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "distanceToFollowing"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "distanceToPreceding"
  .. .. ..$ title : chr "Distance to preceding range group"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "distanceToPreceding"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "distToCds"
  .. .. ..$ title : chr "Get distances between ORF ends and starts of their transcripts cds."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "distToCds"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "distToTSS"
  .. .. ..$ title : chr "Get distances between ORF Start and TSS of its transcript"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "distToTSS"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "download_gene_homologues"
  .. .. ..$ title : chr "Download homologue information of a gene"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "download_gene_homologues"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "download_gene_info"
  .. .. ..$ title : chr "Download summary information of a gene"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "download_gene_info"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "download.SRA"
  .. .. ..$ title : chr "Download read libraries from SRA"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "download.SRA"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "download.SRA.metadata"
  .. .. ..$ title : chr "Downloads metadata from SRA"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "download.SRA.metadata"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DTEG.analysis"
  .. .. ..$ title : chr "Run differential TE analysis"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DTEG.analysis"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "DTEG.plot"
  .. .. ..$ title : chr "Plot DTEG result"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "DTEG.plot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "entropy"
  .. .. ..$ title : chr "Percentage of maximum entropy"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "entropy"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "envExp"
  .. .. ..$ title : chr "Get ORFik experiment environment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "envExp"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "envExp-experiment-method"
  .. .. ..$ title : chr "Get ORFik experiment environment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "envExp,experiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "envExp-set"
  .. .. ..$ title : chr "Set ORFik experiment environment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "envExp&lt;-"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "envExp-set-experiment-method"
  .. .. ..$ title : chr "Set ORFik experiment environment"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "envExp&lt;-,experiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "exonsWithPseudoIntronsPerGroup"
  .. .. ..$ title : chr "Get exons with pseudo introns per Group"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "exonsWithPseudoIntronsPerGroup"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "experiment-class"
  .. .. ..$ title : chr "experiment class definition"
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "experiment"
  .. .. .. ..$ : chr "experiment-class"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "experiment.colors"
  .. .. ..$ title : chr "Decide color for libraries by grouping"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "experiment.colors"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.bed12"
  .. .. ..$ title : chr "Export as bed12 format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.bed12"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.bedo"
  .. .. ..$ title : chr "Store GRanges object as .bedo"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.bedo"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.bedoc"
  .. .. ..$ title : chr "Store GAlignments object as .bedoc"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.bedoc"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.bigWig"
  .. .. ..$ title : chr "Export as bigWig format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.bigWig"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.fstwig"
  .. .. ..$ title : chr "Export as fstwig (fastwig) format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.fstwig"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.ofst"
  .. .. ..$ title : chr "Store GRanges / GAlignments object as .ofst"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.ofst"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.ofst-GAlignmentPairs-method"
  .. .. ..$ title : chr "Store GRanges / GAlignments object as .ofst"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.ofst,GAlignmentPairs-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.ofst-GAlignments-method"
  .. .. ..$ title : chr "Store GRanges / GAlignments object as .ofst"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.ofst,GAlignments-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.ofst-GRanges-method"
  .. .. ..$ title : chr "Store GRanges / GAlignments object as .ofst"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.ofst,GRanges-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "export.wiggle"
  .. .. ..$ title : chr "Export as wiggle format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "export.wiggle"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "extendLeaders"
  .. .. ..$ title : chr "Extend the leaders transcription start sites."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "extendLeaders"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "extendLeadersUntil"
  .. .. ..$ title : chr "Extend Leaders Until"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "extendLeadersUntil"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "extendTrailers"
  .. .. ..$ title : chr "Extend the Trailers transcription stop sites"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "extendTrailers"
  .. .. [list output truncated]
  ..$ _readme           : chr "https://github.com/bioc/ORFik/raw/HEAD/README.md"
  ..$ _rundeps          :List of 128
  .. ..$ : chr "abind"
  .. ..$ : chr "AnnotationDbi"
  .. ..$ : chr "askpass"
  .. ..$ : chr "BH"
  .. ..$ : chr "Biobase"
  .. ..$ : chr "BiocFileCache"
  .. ..$ : chr "BiocGenerics"
  .. ..$ : chr "BiocIO"
  .. ..$ : chr "BiocParallel"
  .. ..$ : chr "biomaRt"
  .. ..$ : chr "biomartr"
  .. ..$ : chr "Biostrings"
  .. ..$ : chr "bit"
  .. ..$ : chr "bit64"
  .. ..$ : chr "bitops"
  .. ..$ : chr "blob"
  .. ..$ : chr "BSgenome"
  .. ..$ : chr "cachem"
  .. ..$ : chr "cigarillo"
  .. ..$ : chr "cli"
  .. ..$ : chr "clipr"
  .. ..$ : chr "codetools"
  .. ..$ : chr "cowplot"
  .. ..$ : chr "cpp11"
  .. ..$ : chr "crayon"
  .. ..$ : chr "curl"
  .. ..$ : chr "data.table"
  .. ..$ : chr "DBI"
  .. ..$ : chr "dbplyr"
  .. ..$ : chr "DelayedArray"
  .. ..$ : chr "DESeq2"
  .. ..$ : chr "digest"
  .. ..$ : chr "downloader"
  .. ..$ : chr "dplyr"
  .. ..$ : chr "farver"
  .. ..$ : chr "fastmap"
  .. ..$ : chr "filelock"
  .. ..$ : chr "formatR"
  .. ..$ : chr "fs"
  .. ..$ : chr "fst"
  .. ..$ : chr "fstcore"
  .. ..$ : chr "futile.logger"
  .. ..$ : chr "futile.options"
  .. ..$ : chr "generics"
  .. ..$ : chr "GenomeInfoDb"
  .. ..$ : chr "GenomicAlignments"
  .. ..$ : chr "GenomicFeatures"
  .. ..$ : chr "GenomicRanges"
  .. ..$ : chr "ggplot2"
  .. ..$ : chr "glue"
  .. ..$ : chr "gridExtra"
  .. ..$ : chr "gtable"
  .. ..$ : chr "hms"
  .. ..$ : chr "httr"
  .. ..$ : chr "httr2"
  .. ..$ : chr "IRanges"
  .. ..$ : chr "isoband"
  .. ..$ : chr "jsonlite"
  .. ..$ : chr "KEGGREST"
  .. ..$ : chr "KernSmooth"
  .. ..$ : chr "labeling"
  .. ..$ : chr "lambda.r"
  .. ..$ : chr "lattice"
  .. ..$ : chr "lifecycle"
  .. ..$ : chr "locfit"
  .. ..$ : chr "magrittr"
  .. ..$ : chr "Matrix"
  .. ..$ : chr "MatrixGenerics"
  .. ..$ : chr "matrixStats"
  .. ..$ : chr "memoise"
  .. ..$ : chr "mime"
  .. ..$ : chr "openssl"
  .. ..$ : chr "philentropy"
  .. ..$ : chr "pillar"
  .. ..$ : chr "pkgconfig"
  .. ..$ : chr "plogr"
  .. ..$ : chr "png"
  .. ..$ : chr "poorman"
  .. ..$ : chr "prettyunits"
  .. ..$ : chr "progress"
  .. ..$ : chr "purrr"
  .. ..$ : chr "qs"
  .. ..$ : chr "R.methodsS3"
  .. ..$ : chr "R.oo"
  .. ..$ : chr "R.utils"
  .. ..$ : chr "R6"
  .. ..$ : chr "RApiSerialize"
  .. ..$ : chr "rappdirs"
  .. ..$ : chr "RColorBrewer"
  .. ..$ : chr "Rcpp"
  .. ..$ : chr "RcppArmadillo"
  .. ..$ : chr "RcppParallel"
  .. ..$ : chr "RCurl"
  .. ..$ : chr "readr"
  .. ..$ : chr "restfulr"
  .. ..$ : chr "Rhtslib"
  .. ..$ : chr "rjson"
  .. ..$ : chr "rlang"
  .. ..$ : chr "Rsamtools"
  .. .. [list output truncated]
  ..$ _sysdeps          :List of 1
  .. ..$ :List of 7
  .. .. ..$ shlib      : chr "libstdc++"
  .. .. ..$ package    : chr "libstdc++6"
  .. .. ..$ source     : chr "gcc"
  .. .. ..$ version    : chr "14.2.0-4ubuntu2~24.04"
  .. .. ..$ name       : chr "c++"
  .. .. ..$ homepage   : chr "http://gcc.gnu.org/"
  .. .. ..$ description: chr "GNU Standard C++ Library v3"
  ..$ _vignettes        :List of 7
  .. ..$ :List of 9
  .. .. ..$ source  : chr "Annotation_Alignment.Rmd"
  .. .. ..$ filename: chr "Annotation_Alignment.html"
  .. .. ..$ title   : chr "Annotation &amp; Alignment"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 28
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "How ORFik organizes stored files"
  .. .. .. ..$ : chr "Specify output folders"
  .. .. .. ..$ : chr "Download RNA-seq NGS data"
  .. .. .. ..$ : chr "Annotation (Fasta genome and gtf file)"
  .. .. .. ..$ : chr "Download genome and gtf files"
  .. .. .. ..$ : chr "Contaminants"
  .. .. .. ..$ : chr "Local annotation"
  .. .. .. ..$ : chr "Annotation without defined UTRs"
  .. .. .. ..$ : chr "Create 5' UTR annotation from CAGE"
  .. .. .. ..$ : chr "Create Pseudo 5' UTR annotation"
  .. .. .. ..$ : chr "Fixing malformed gtf/gff"
  .. .. .. ..$ : chr "RNA-seq alignment"
  .. .. .. ..$ : chr "Indexing"
  .. .. .. ..$ : chr "Trimming and Aligning the data"
  .. .. .. ..$ : chr "Trimming data"
  .. .. .. ..$ : chr "Aligning the data"
  .. .. .. ..$ : chr "UMIs"
  .. .. .. ..$ : chr "Collapsing duplicated reads"
  .. .. .. ..$ : chr "Solving errors when aligning:"
  .. .. .. ..$ : chr "RAM usage warnings"
  .. .. .. ..$ : chr "Systems that restricts max open files"
  .. .. .. ..$ : chr "Restart STAR from crashed step"
  .. .. .. ..$ : chr "Create an ORFik experiment of the Yeast data"
  .. .. .. ..$ : chr "Post alignment QC report"
  .. .. .. ..$ : chr "Convert libraries to new formats"
  .. .. .. ..$ : chr "Outputting libraries to R"
  .. .. .. ..$ : chr "FPKM values (normalized counts)"
  .. .. ..$ created : chr "2022-10-27 15:44:24"
  .. .. ..$ modified: chr "2025-05-06 07:07:41"
  .. .. ..$ commits : int 7
  .. ..$ :List of 9
  .. .. ..$ source  : chr "ORFikExperiment.Rmd"
  .. .. ..$ filename: chr "ORFikExperiment.html"
  .. .. ..$ title   : chr "Data management"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 33
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Motivation"
  .. .. .. ..$ : chr "What is an ORFik experiment?"
  .. .. .. ..$ : chr "creating an ORFik experiment"
  .. .. .. ..$ : chr "A minimal experiment"
  .. .. .. ..$ : chr "Fixing or updating an experiment"
  .. .. .. ..$ : chr "ORFik example experiment"
  .. .. .. ..$ : chr "The experiment object"
  .. .. .. ..$ : chr "Accessing library file paths"
  .. .. .. ..$ : chr "Loading data from experiment"
  .. .. .. ..$ : chr "Loading NGS data to environment"
  .. .. .. ..$ : chr "Loading NGS data as list"
  .. .. .. ..$ : chr "Loading NGS data by fimport"
  .. .. .. ..$ : chr "Loading Annotation and specific regions"
  .. .. .. ..$ : chr "Plotting with ORFik experiments"
  .. .. .. ..$ : chr "P-site shifting experiment"
  .. .. .. ..$ : chr "Converting bam files to faster formats"
  .. .. .. ..$ : chr "ofst: ORFik serialized format"
  .. .. .. ..$ : chr "bedo: bed ORFik file (to be deprecated!)"
  .. .. .. ..$ : chr "bedoc: bed ORFik file with cigar (to be deprecated!)"
  .. .. .. ..$ : chr "ORFik QC report"
  .. .. .. ..$ : chr "General report"
  .. .. .. ..$ : chr "How to run QC:"
  .. .. .. ..$ : chr "How to load optimized output from QC:"
  .. .. .. ..$ : chr "How to load the QC statistics:"
  .. .. .. ..$ : chr "How to see the QC plots:"
  .. .. .. ..$ : chr "Ribo-seq specific pshifting and QC"
  .. .. .. ..$ : chr "Using the ORFik experiment system in your script"
  .. .. .. ..$ : chr "Note for windows users"
  .. .. .. ..$ : chr "Looping over all libraries in experiment"
  .. .. .. ..$ : chr "Reformat output to data.table (merge column-wise)"
  .. .. .. ..$ : chr "Reformat output to data.table (merge row-wise)"
  .. .. .. ..$ : chr "Conclusion"
  .. .. ..$ created : chr "2022-10-27 15:44:24"
  .. .. ..$ modified: chr "2024-12-13 11:59:09"
  .. .. ..$ commits : int 4
  .. ..$ :List of 9
  .. .. ..$ source  : chr "Importing_Data.Rmd"
  .. .. ..$ filename: chr "Importing_Data.html"
  .. .. ..$ title   : chr "Importing data"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 20
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Motivation"
  .. .. .. ..$ : chr "Importing Sequencing reads"
  .. .. .. ..$ : chr "Loading files (general)"
  .. .. .. ..$ : chr "Bam files"
  .. .. .. ..$ : chr "Other formats"
  .. .. .. ..$ : chr "Exporting to new formats"
  .. .. .. ..$ : chr "Files are not preload into R"
  .. .. .. ..$ : chr "BAM to OFST (keep cigar information)"
  .. .. .. ..$ : chr "BAM to bigwig (do not keep cigar information or read lengths)"
  .. .. .. ..$ : chr "Files are preload into R"
  .. .. .. ..$ : chr "Random access"
  .. .. .. ..$ : chr "Importing Annotation"
  .. .. .. ..$ : chr "Loading Genome index (fasta index)"
  .. .. .. ..$ : chr "Loading Gene annotation (Txdb)"
  .. .. .. ..$ : chr "Loading Transcript regions"
  .. .. .. ..$ : chr "Loading Transcript regions (filtering)"
  .. .. .. ..$ : chr "Loading Transcript regions (filtering by length requirements)"
  .. .. .. ..$ : chr "Loading Transcript regions (filtering by canonical isoform)"
  .. .. .. ..$ : chr "Loading uORF annotation"
  .. .. ..$ created : chr "2023-03-23 13:58:56"
  .. .. ..$ modified: chr "2024-10-02 17:27:56"
  .. .. ..$ commits : int 9
  .. ..$ :List of 9
  .. .. ..$ source  : chr "ORFikOverview.Rmd"
  .. .. ..$ filename: chr "ORFikOverview.html"
  .. .. ..$ title   : chr "ORFik Overview"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 23
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Finding Open Reading Frames"
  .. .. .. ..$ : chr "Example of finding ORFs in on 5' UTR of hg19"
  .. .. .. ..$ : chr "Saving coordinates of ORFs to disc"
  .. .. .. ..$ : chr "Getting sequences from ORFs"
  .. .. .. ..$ : chr "Getting DNA fasta sequences of ORFs"
  .. .. .. ..$ : chr "Amino acid sequences of ORFs"
  .. .. .. ..$ : chr "New GRanges and GRangesList utilities for ORFs"
  .. .. .. ..$ : chr "Grouping ORFs"
  .. .. .. ..$ : chr "Filtering example"
  .. .. .. ..$ : chr "ORF interest regions"
  .. .. .. ..$ : chr "When to use which ORF-finding function"
  .. .. .. ..$ : chr "Finding ORFs in spliced transcripts"
  .. .. .. ..$ : chr "Prokaryote/Circular Genomes and fasta transcriptomes."
  .. .. .. ..$ : chr "Filter on strand"
  .. .. .. ..$ : chr "CAGE data for 5' UTR re-annotation"
  .. .. .. ..$ : chr "RiboSeq footprints automatic shift detection and shifting"
  .. .. .. ..$ : chr "Gene identity functions for ORFs or genes"
  .. .. .. ..$ : chr "Calculating Kozak sequence score for ORFs"
  .. .. .. ..$ : chr "Using ORFik in your package or scripts"
  .. .. .. ..$ : chr "Coverage plots made easy with ORFik"
  .. .. .. ..$ : chr "Multiple data sets in one plot"
  .. .. .. ..$ : chr "Conclusion"
  .. .. ..$ created : chr "2017-11-22 15:33:59"
  .. .. ..$ modified: chr "2024-06-05 13:32:38"
  .. .. ..$ commits : int 67
  .. ..$ :List of 9
  .. .. ..$ source  : chr "Ribo-seq_pipeline-human.Rmd"
  .. .. ..$ filename: chr "Ribo-seq_pipeline-human.html"
  .. .. ..$ title   : chr "Ribo-seq pipeline (Yeast)"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 2
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Pipeline"
  .. .. ..$ created : chr "2025-05-06 07:07:41"
  .. .. ..$ modified: chr "2025-05-06 07:07:41"
  .. .. ..$ commits : int 1
  .. ..$ :List of 9
  .. .. ..$ source  : chr "Ribo-seq_pipeline-yeast.Rmd"
  .. .. ..$ filename: chr "Ribo-seq_pipeline-yeast.html"
  .. .. ..$ title   : chr "Ribo-seq pipeline (Yeast)"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 2
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Pipeline"
  .. .. ..$ created : chr "2025-05-06 07:07:41"
  .. .. ..$ modified: chr "2025-05-06 07:07:41"
  .. .. ..$ commits : int 1
  .. ..$ :List of 9
  .. .. ..$ source  : chr "Working_with_transcripts.Rmd"
  .. .. ..$ filename: chr "Working_with_transcripts.html"
  .. .. ..$ title   : chr "Working with transcripts"
  .. .. ..$ author  : chr "Haakon Tjeldnes &amp; Kornel Labun"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 12
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Motivation"
  .. .. .. ..$ : chr "Getting data"
  .. .. .. ..$ : chr "Loading Transcript region (single point locations)"
  .. .. .. ..$ : chr "Loading Transcript region (window locations)"
  .. .. .. ..$ : chr "Extending transcript into genomic flanks"
  .. .. .. ..$ : chr "Coverage over transcript regions"
  .. .. .. ..$ : chr "Loading NGS data"
  .. .. .. ..$ : chr "Total counts per region/window"
  .. .. .. ..$ : chr "Per nucleotide in region/window"
  .. .. .. ..$ : chr "Per nucleotide in region/window (split by read length)"
  .. .. .. ..$ : chr "Advanced details"
  .. .. ..$ created : chr "2023-03-23 13:58:56"
  .. .. ..$ modified: chr "2023-08-04 20:45:20"
  .. .. ..$ commits : int 4
  ..$ _score            : num 10.6
  ..$ _indexed          : logi TRUE
  ..$ _nocasepkg        : chr "orfik"
  ..$ _universes        :List of 2
  .. ..$ : chr "bioc"
  .. ..$ : chr "roleren"
  ..$ _previous         : chr "1.29.9"
  ..$ _binaries         :List of 12
  .. ..$ :List of 11
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:45.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ arch    : chr "aarch64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "32e961971a2545abe00924ba57190a5a68d6cadf14b030057815fead9f7d51b1"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 11
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:54:55.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "8087e75f2619b59c52c796a42a112962cbc2b02a9605b9821bb812d7bb3a620d"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 11
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:11.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ arch    : chr "aarch64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "22686916be9ce27b12967704363346c5f1cbbd34bb0ef78ec2204bb64bc89303"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 11
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:05.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "f35f4f8b399d1071670872704b6318dad85d5757df02fdcac554e54e70a08394"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:53:55.000Z"
  .. .. ..$ arch    : chr "aarch64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "81ca750329140f985006310a9bb3c8f1193a67c56fd997b70fe12266f07548dd"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:58:36.000Z"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "32f95edfcaef50c637adc3833315c96efd2eea5a2f5d2165dcd96d5e91b4fe3d"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:53:41.000Z"
  .. .. ..$ arch    : chr "aarch64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "003aac5dcdb2ffe0cddf27cf7fb0b363fc16946aa5f368f2a64819603224e162"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:56:51.000Z"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "427357d494faa2153a662e40ddbeb2179e11ed9fd15f75ca68559cd698e939eb"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "wasm"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:17.000Z"
  .. .. ..$ arch    : chr "emscripten"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "f4676d5f93ee9dcc8f0cd5626a1e95ae93a674eaa6f5af2ff38deec5d69f5a65"
  .. .. ..$ status  : chr "success"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:55.000Z"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "b5268ac37e579fbe745f5be1a50dc35a177e32270fd2d875280ccb1f179771ed"
  .. .. ..$ status  : chr "failure"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:09.000Z"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "825c34e878d3c04232ec80fda879a9fcab94995fa12a085ed0817a38303b7d28"
  .. .. ..$ status  : chr "failure"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.31.1"
  .. .. ..$ date    : chr "2025-11-21T15:55:52.000Z"
  .. .. ..$ arch    : chr "x86_64"
  .. .. ..$ commit  : chr "7b43c67a171ffa1f2d83347d21d256e54e5eed9e"
  .. .. ..$ fileid  : chr "7845b2d667bc7e1cc7a1a0ea1ae34dbbac5719dbadd01642e1bf5a350a39b9a7"
  .. .. ..$ status  : chr "failure"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19575603413"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>