<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>exploreBiocUniverse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="exploreBiocUniverse_files/libs/clipboard/clipboard.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/popper.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/anchor.min.js"></script>
<link href="exploreBiocUniverse_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="exploreBiocUniverse_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-6650302f782765824de81675458e0c02.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="exploreBiocUniverse_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="exploreBiocUniverse_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">exploreBiocUniverse</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="comparing-bioconductor-packages-in-r-universe-and-the-bbs" class="level1">
<h1>Comparing Bioconductor packages in R-Universe and the BBS</h1>
<p>Skip to <a href="#packages-not-in-agreement">Packages not in agreements</a> or go to <a href="windows.qmd">windows</a>.</p>
<section id="r-universe" class="level2">
<h2 class="anchored" data-anchor-id="r-universe">R Universe</h2>
<p>We can use the <a href="https://github.com/ropensci/universe/">universe</a> package to explore the R-Universe (RU) API. (We might need to increase the <code>max.print</code> to show all packages.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universe)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">max.print =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data for the Bioc Universe (BU).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs_i <span class="ot">&lt;-</span> <span class="fu">universe_all_packages</span>(<span class="st">"bioc"</span>, <span class="at">limit =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What packages are in bioc universe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs <span class="ot">&lt;-</span> <span class="fu">universe_ls</span>(<span class="st">"bioc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bbs" class="level2">
<h2 class="anchored" data-anchor-id="bbs">BBS</h2>
<p>Get Bioconductor packages in BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocPkgTools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bbs_pkgs_i <span class="ot">&lt;-</span> BiocPkgTools<span class="sc">::</span><span class="fu">biocBuildReport</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">version =</span> BiocManager<span class="sc">::</span><span class="fu">version</span>(),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgType =</span> <span class="st">"software"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>bbs_pkgs <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unique</span>(bbs_pkgs_i<span class="sc">$</span>pkg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Packages in RU vs BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">RU =</span> <span class="fu">length</span>(ru_pkgs), <span class="at">BBS =</span> <span class="fu">length</span>(bbs_pkgs),</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"Total"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        RU  BBS
Total 2334 2361</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>not_in_ru <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(bbs_pkgs, ru_pkgs)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>not_in_bbs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(ru_pkgs, bbs_pkgs)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># There's probably a better way to do this</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">Created =</span> <span class="fu">character</span>(), <span class="at">Published =</span> <span class="fu">character</span>(),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">RemoteUrl =</span> <span class="fu">character</span>())</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> not_in_bbs) {</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  pi <span class="ot">&lt;-</span> <span class="fu">universe_one_package</span>(<span class="st">"bioc"</span>, pkg)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  pkgs_not_in_bbs <span class="ot">&lt;-</span> pkgs_not_in_bbs <span class="sc">|&gt;</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> pi<span class="sc">$</span>Package, <span class="at">Version =</span> pi<span class="sc">$</span>Version,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">Created =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_created</span><span class="st">`</span>, <span class="at">Published =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_published</span><span class="st">`</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">RemoteUrl =</span> pi<span class="sc">$</span>RemoteUrl)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-not-in-bbs-but-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-not-in-bbs-but-in-ru">Packages not in BBS but in RU</h3>
<p>Packages in RU but not in the BBS are likely added by maintainers via <a href="https://github.com/r-universe/bioc/blob/master/.remotes.json" class="uri">https://github.com/r-universe/bioc/blob/master/.remotes.json</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="sc">|&gt;</span> knitr<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Created</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Published</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RemoteUrl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Aerith</td>
<td style="text-align: left;">0.99.11</td>
<td style="text-align: left;">2026-01-03T08:33:59.000Z</td>
<td style="text-align: left;">2026-01-05T09:53:52.133Z</td>
<td style="text-align: left;">https://github.com/bioc/Aerith</td>
</tr>
<tr class="even">
<td style="text-align: left;">ClusterGVis</td>
<td style="text-align: left;">0.99.9</td>
<td style="text-align: left;">2026-01-04T15:14:42.000Z</td>
<td style="text-align: left;">2026-01-04T18:25:03.073Z</td>
<td style="text-align: left;">https://github.com/bioc/ClusterGVis</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LACHESIS</td>
<td style="text-align: left;">0.99.4</td>
<td style="text-align: left;">2025-12-23T07:24:20.000Z</td>
<td style="text-align: left;">2025-12-23T08:17:52.495Z</td>
<td style="text-align: left;">https://github.com/bioc/LACHESIS</td>
</tr>
<tr class="even">
<td style="text-align: left;">SpiecEasi</td>
<td style="text-align: left;">1.99.3</td>
<td style="text-align: left;">2025-12-23T06:44:17.000Z</td>
<td style="text-align: left;">2025-12-23T07:10:56.931Z</td>
<td style="text-align: left;">https://github.com/bioc/SpiecEasi</td>
</tr>
<tr class="odd">
<td style="text-align: left;">StatescopeR</td>
<td style="text-align: left;">0.99.33</td>
<td style="text-align: left;">2025-12-25T05:30:37.000Z</td>
<td style="text-align: left;">2025-12-25T06:02:12.332Z</td>
<td style="text-align: left;">https://github.com/bioc/StatescopeR</td>
</tr>
<tr class="even">
<td style="text-align: left;">asuri</td>
<td style="text-align: left;">0.99.18</td>
<td style="text-align: left;">2025-12-22T06:19:40.000Z</td>
<td style="text-align: left;">2025-12-22T06:30:24.152Z</td>
<td style="text-align: left;">https://github.com/bioc/asuri</td>
</tr>
<tr class="odd">
<td style="text-align: left;">damidBind</td>
<td style="text-align: left;">0.99.12</td>
<td style="text-align: left;">2026-01-07T03:56:38.000Z</td>
<td style="text-align: left;">2026-01-07T04:09:22.932Z</td>
<td style="text-align: left;">https://github.com/bioc/damidBind</td>
</tr>
<tr class="even">
<td style="text-align: left;">dominatR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2025-12-23T06:49:50.000Z</td>
<td style="text-align: left;">2025-12-23T07:15:38.700Z</td>
<td style="text-align: left;">https://github.com/bioc/dominatR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ggpattern</td>
<td style="text-align: left;">1.2.1-0</td>
<td style="text-align: left;">2026-01-09T21:40:27.000Z</td>
<td style="text-align: left;">2026-01-09T22:50:38.395Z</td>
<td style="text-align: left;">https://github.com/coolbutuseless/ggpattern</td>
</tr>
<tr class="even">
<td style="text-align: left;">immReferent</td>
<td style="text-align: left;">0.99.6</td>
<td style="text-align: left;">2026-01-05T04:17:54.000Z</td>
<td style="text-align: left;">2026-01-05T04:22:21.195Z</td>
<td style="text-align: left;">https://github.com/bioc/immReferent</td>
</tr>
<tr class="odd">
<td style="text-align: left;">leapR</td>
<td style="text-align: left;">0.99.5</td>
<td style="text-align: left;">2025-12-21T08:40:21.000Z</td>
<td style="text-align: left;">2025-12-21T08:46:35.690Z</td>
<td style="text-align: left;">https://github.com/bioc/leapR</td>
</tr>
<tr class="even">
<td style="text-align: left;">lmms</td>
<td style="text-align: left;">1.3.3</td>
<td style="text-align: left;">2025-12-31T08:01:58.000Z</td>
<td style="text-align: left;">2025-12-31T08:04:58.120Z</td>
<td style="text-align: left;">https://github.com/cran/lmms</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="packages-in-bbs-but-not-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-in-bbs-but-not-in-ru">Packages in BBS but not in RU</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>not_in_ru</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "bgx"                    "BiGGR"                  "biodbHmdb"             
 [4] "biodbNcbi"              "biodbNci"               "biodbUniprot"          
 [7] "ccmap"                  "CellScore"              "CINdex"                
[10] "cisPath"                "DEP"                    "gpuMagic"              
[13] "Harshlight"             "hiAnnotator"            "hiReadsProcessor"      
[16] "interactiveDisplay"     "interactiveDisplayBase" "lapmix"                
[19] "LinTInd"                "lute"                   "MADSEQ"                
[22] "netZooR"                "oppti"                  "PhenStat"              
[25] "qckitfastq"             "ReactomeGraph4R"        "Repitools"             
[28] "Rfastp"                 "rGADEM"                 "rRDP"                  
[31] "SARC"                   "seqArchR"               "seqArchRplus"          
[34] "seqTools"               "Streamer"               "TitanCNA"              
[37] "TransView"              "traviz"                 "XNAString"             </code></pre>
</div>
</div>
<p>The following packages are not deprecated but are not in RU.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>depre_pkgs <span class="ot">&lt;-</span> <span class="fu">filter</span>(bbs_pkgs_i, Deprecated <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(pkg) <span class="sc">|&gt;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(not_in_ru, depre_pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "bgx"          "biodbHmdb"    "biodbNcbi"    "biodbNci"     "biodbUniprot"
[6] "cisPath"      "hiAnnotator"  "seqArchRplus" "traviz"      </code></pre>
</div>
</div>
</section>
</section>
<section id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc." class="level2">
<h2 class="anchored" data-anchor-id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc.">Making sense of R Universe’s build status in respect to R CMD build, check, etc.</h2>
<p>Looking at the status of linux x86_64 binary builds. The build “status” appears to get “success”, “failure”, or “cancelled” from the GitHub Action overall. The “check” value corresponds to <code>R CMD check</code>. I interpret the result of “status” to be “success” OK and “failure” ERROR for <code>R CMD build</code> but it may mean something else. We’ll use CANCELLED as it is. There’s also a special status for documentation, which is handled separately from build/check, which may account for some discrepancies. Not sure if the build status for vignettes is available in the API.</p>
<p>Packages with my comment “Missing R4.6 on Linux” did have data in the API but no x86_64 linux with R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Status =</span> <span class="fu">character</span>(), <span class="at">Build =</span> <span class="fu">character</span>(),</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Check =</span> <span class="fu">character</span>(), <span class="at">Comment =</span> <span class="fu">character</span>())</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#, Buildurl = character(),</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># Commit = character())</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> ru_pkgs_i) {</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> <span class="cn">NA</span>, <span class="at">build =</span> <span class="cn">NA</span>, <span class="at">check =</span> <span class="cn">NA</span>, <span class="at">buildurl =</span> <span class="cn">NA</span>)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  comment <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (b <span class="cf">in</span> p<span class="sc">$</span><span class="st">`</span><span class="at">_binaries</span><span class="st">`</span>) {</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (b<span class="sc">$</span>os <span class="sc">==</span> <span class="st">"linux"</span> <span class="sc">&amp;&amp;</span> b<span class="sc">$</span>r <span class="sc">==</span> <span class="st">"4.6.0"</span>) {</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>      build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> b<span class="sc">$</span>status, <span class="at">check =</span> b<span class="sc">$</span>check, <span class="at">buildurl =</span> b<span class="sc">$</span>buildurl)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(build[<span class="st">"status"</span>])) {</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> <span class="st">"Missing R4.6 on Linux"</span> </span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"success"</span>) {</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"failure"</span>) {</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"ERROR*"</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"cancelled"</span>) {</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"CANCELLED"</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  ru_status <span class="ot">&lt;-</span> ru_status <span class="sc">|&gt;</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> p<span class="sc">$</span>Package, <span class="at">Version =</span> p<span class="sc">$</span>Version, <span class="at">Status =</span> build[<span class="st">"status"</span>],</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">Build =</span> build[<span class="st">"build"</span>], <span class="at">Check =</span> build[<span class="st">"check"</span>], <span class="at">Comment =</span> comment)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>            <span class="co">#, Buildurl = build["buildurl"], Commit = p$`_commit`$id)</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a><span class="co"># order by Package</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> ru_status[<span class="fu">order</span>(ru_status<span class="sc">$</span>Package), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>DEVEL_LINUX_BUILDER <span class="ot">=</span> <span class="st">"nebbiolo2"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper functions</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>get_result <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  entry <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pkg <span class="sc">==</span> x[<span class="st">"pkg"</span>], stage <span class="sc">==</span> x[<span class="st">"stage"</span>], node <span class="sc">==</span> x[<span class="st">"node"</span>])</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(entry)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    entry <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    entry<span class="sc">$</span>result</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>get_install <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"install"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>get_build <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"buildsrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>get_check <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"checksrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>bbs_status <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(node <span class="sc">==</span> DEVEL_LINUX_BUILDER) <span class="sc">|&gt;</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version, Deprecated) <span class="sc">|&gt;</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(pkg, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">install =</span> <span class="fu">map_chr</span>(pkg, get_install)) <span class="sc">|&gt;</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">build =</span> <span class="fu">map_chr</span>(pkg, get_build)) <span class="sc">|&gt;</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">check =</span> <span class="fu">map_chr</span>(pkg, get_check))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-the-status-of-packages-in-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-status-of-packages-in-ru-and-bbs">Compare the status of packages in RU and BBS</h3>
<p>We can only compare <code>check</code> and <code>build</code>, so we’ll check if those columns agree. Note that in the BBS, <code>check</code> can show “WARNINGS” whereas R Universe can show “WARNING” (no s) as well as “OK”, “NOTE”, “ERROR”, or NA.</p>
<p>Note: - Deprecated packages are removed immedately in R Universe. - Packages may have different versions since R-Universe builds new versions after detecting a “push”.</p>
<p>The following includes packages not deprecated in BBS where build and check don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ru_status) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pkg"</span>, <span class="st">"version"</span>, <span class="st">"status"</span>, <span class="st">"build"</span>, <span class="st">"check"</span>, <span class="st">"comment"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>bbs_ru_comparison <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> bbs_status, <span class="at">y =</span> ru_status, <span class="at">by =</span> <span class="st">"pkg"</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">&amp;</span> Deprecated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(pkg, install, build.x, build.y, check.x, check.y, version.x,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                  version.y, status, comment)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>, <span class="st">"RU_build"</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"BBS_version"</span>, <span class="st">"RU_version"</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-without-x86_64-linux-builds-for-r-4.6" class="level4">
<h4 class="anchored" data-anchor-id="packages-without-x86_64-linux-builds-for-r-4.6">Packages without x86_64 Linux builds for R 4.6</h4>
<p>The following are packages in bioc R Universe but without an x86_64 Linux build for R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="sc">|&gt;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Comment <span class="sc">==</span> <span class="st">"Missing R4.6 on Linux"</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages in RU without an x86_64 linux build for R 4.6"</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages in RU without an x86_64 linux build for R 4.6</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ChIPQC</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.46.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">CONFESS</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ddPCRclust</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.30.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">debCAM</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">ERROR</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">skipped</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.28.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GEOmetadb</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.72.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">HilbertVisGUI</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.68.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SamSPECTRAL</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.64.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">SpatialOmicsOverlay</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.10.0</td>
<td style="text-align: left;">1.11.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SwathXtend</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Notes: - Retrying to refresh the build on the dashboard failed for some packages. How can we rebuild a package that hasn’t received a push in 3 months? - paxtoolsr should be in Bioc R Universe, but it isn’t. Why?</p>
</section>
</section>
</section>
<section id="packages-not-in-agreement" class="level2">
<h2 class="anchored" data-anchor-id="packages-not-in-agreement">Packages not in agreement</h2>
<p>Looking at packages with the same version and removing deprecated packages, we have the following packages where the BBS and R Universe’s results don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Deprecated <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> version.x <span class="sc">==</span> version.y <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(status)) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version.x, install, build.x, build.y, check.x, check.y,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>         status, comment)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_version"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"RU_build"</span>, <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="sc">|&gt;</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages with different build results between BBS and RU"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages with different build results between BBS and RU</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Note: <code>ERROR*</code> could be an <code>ERROR</code>, <code>TIMEOUT</code>, or <code>skipped</code>.</p>
<section id="total-packages-not-in-agreement" class="level3">
<h3 class="anchored" data-anchor-id="total-packages-not-in-agreement">Total packages not in agreement</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bbs_ru_not_ok2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="other" class="level1">
<h1>Other</h1>
<p>If a package repo isn’t accessible or an error exists in the DESCRIPTION file, there will be an error in the registry for that submodule and it will show up at <a href="https://github.com/r-universe/bioc/actions/workflows/sync.yml" class="uri">https://github.com/r-universe/bioc/actions/workflows/sync.yml</a>. The red triangle on <a href="https://bioc.r-universe.dev/" class="uri">https://bioc.r-universe.dev/</a> will take you to the specific workflow error.</p>
<p>Currently, this information is available on the BBS but I wonder if this information is available somewhere for R Universe OR if it would be necessary given that developers would be able to inspect their builds sooner.</p>
<section id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api" class="level3">
<h3 class="anchored" data-anchor-id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api">Important: Use the dbdump endpoint with read_bson to dump the API</h3>
<p>Use <code>read_bson</code> to dump API data into a db</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("mongolite")</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mongolite)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>bioc <span class="ot">&lt;-</span> mongolite<span class="sc">::</span><span class="fu">read_bson</span>(<span class="st">'https://bioc.r-universe.dev/api/dbdump'</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: it returns 2268 records rather than 2276 (fewer retrieved through</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># another endpoint)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-a-tarball-from-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="comparing-a-tarball-from-ru-and-bbs">Comparing a tarball from RU and BBS</h3>
<p>Lastly, when I look at the builds, they’re using various repositories, such p3m, r-universe, etc. This might be okay as long as the resulting binaries/tarballs are the generally the same. Looking at XVector for both, both the BBS and R Universe inject some additional information. R Universe also creates an <code>extra</code> directory that has citations, manuals, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2 ruXVector_0.47.2</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2/XVector/DESCRIPTION ruXVector_0.47.2/XVector/DESCRIPTION</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="ex">5c5</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>   <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span>         <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="ex">28,33c28,31</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_url: <span class="ex">https://git.bioconductor.org/packages/XVector</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_branch: <span class="ex">devel</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit: <span class="ex">9f44218</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit_date: <span class="ex">2025-01-07</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Repository: <span class="ex">Bioconductor</span> 3.21</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Date/Publication: <span class="ex">2025-01-08</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Repository: <span class="ex">https://bioc.r-universe.dev</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteUrl: <span class="ex">https://github.com/bioc/XVector</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteRef: <span class="ex">HEAD</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteSha: <span class="ex">9f44218308f9d5867f0b9df5fae60d217f4e7d65</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="ex">35c33</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Packaged: <span class="ex">2025-01-09</span> 00:57:29 UTC<span class="kw">;</span> <span class="ex">biocbuild</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Packaged: <span class="ex">2025-01-08</span> 03:05:41 UTC<span class="kw">;</span> <span class="ex">root</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: extra</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: manual.pdf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="structure-of-a-package-through-universe" class="level3">
<h3 class="anchored" data-anchor-id="structure-of-a-package-through-universe">Structure of a package through universe</h3>
<p>The structure of a package looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">str</span>(ru_pkgs_i[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ :List of 76
  ..$ Package           : chr "singleCellTK"
  ..$ Type              : chr "Package"
  ..$ Title             : chr "Comprehensive and Interactive Analysis of Single Cell RNA-Seq\nData"
  ..$ Version           : chr "2.21.1"
  ..$ Authors@R         : chr "c(person(given=\"Yichen\", family=\"Wang\", email=\"wangych@bu.edu\", role=c(\"aut\"),\ncomment = c(ORCID = \"0"| __truncated__
  ..$ Description       : chr "The Single Cell Toolkit (SCTK) in the singleCellTK package\nprovides an interface to popular tools for importin"| __truncated__
  ..$ License           : chr "MIT + file LICENSE"
  ..$ Encoding          : chr "UTF-8"
  ..$ biocViews         : chr "SingleCell, GeneExpression, DifferentialExpression,\nAlignment, Clustering, ImmunoOncology, BatchEffect,\nNorma"| __truncated__
  ..$ LazyData          : chr "FALSE"
  ..$ RoxygenNote       : chr "7.3.3"
  ..$ VignetteBuilder   : chr "knitr"
  ..$ URL               : chr "https://www.camplab.net/sctk/"
  ..$ BugReports        : chr "https://github.com/compbiomed/singleCellTK/issues"
  ..$ Language          : chr "en-US"
  ..$ Config/pak/sysreqs: chr "libcairo2-dev libfftw3-dev libfontconfig1-dev\nlibfreetype6-dev libfribidi-dev libglpk-dev make\nlibharfbuzz-de"| __truncated__
  ..$ Repository        : chr "https://bioc.r-universe.dev"
  ..$ Date/Publication  : chr "2026-01-11 13:42:53 UTC"
  ..$ RemoteUrl         : chr "https://github.com/bioc/singleCellTK"
  ..$ RemoteRef         : chr "HEAD"
  ..$ RemoteSha         : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  ..$ NeedsCompilation  : chr "no"
  ..$ Packaged          :List of 2
  .. ..$ Date: chr "2026-01-12 04:00:16 UTC"
  .. ..$ User: chr "root"
  ..$ Author            : chr "Yichen Wang [aut] (ORCID: &lt;https://orcid.org/0000-0003-4347-5199&gt;),\nIrzam Sarfraz [aut] (ORCID: &lt;https://orcid"| __truncated__
  ..$ Maintainer        : chr "Joshua David Campbell &lt;camp@bu.edu&gt;"
  ..$ MD5sum            : chr "f8653ee0c13dcd56b5733c3c74dc27f1"
  ..$ _user             : chr "bioc"
  ..$ _type             : chr "src"
  ..$ _file             : chr "singleCellTK_2.21.1.tar.gz"
  ..$ _fileid           : chr "c4233a8ba5c3f710f91d7a79a731790cdc57bea31b44035c898b83e8f9acd48c"
  ..$ _filesize         : int 4033834
  ..$ _sha256           : chr "c4233a8ba5c3f710f91d7a79a731790cdc57bea31b44035c898b83e8f9acd48c"
  ..$ _created          : chr "2026-01-12T04:00:16.000Z"
  ..$ _published        : chr "2026-01-12T04:22:16.643Z"
  ..$ _jobs             :List of 10
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 445
  .. .. ..$ config  : chr "bioc-checks"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "WARNING"
  .. .. ..$ artifact: chr "5093465548"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 1086
  .. .. ..$ config  : chr "linux-devel-x86_64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093516973"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 1034
  .. .. ..$ config  : chr "linux-release-x86_64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093512982"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 883
  .. .. ..$ config  : chr "macos-devel-arm64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093501637"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 1074
  .. .. ..$ config  : chr "macos-release-arm64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093515750"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 692
  .. .. ..$ config  : chr "source"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "5093429657"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 412
  .. .. ..$ config  : chr "wasm-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "5093462978"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 1129
  .. .. ..$ config  : chr "windows-devel"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093520807"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 951
  .. .. ..$ config  : chr "windows-oldrel"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "5093506656"
  .. ..$ :List of 6
  .. .. ..$ job     : num 6.01e+10
  .. .. ..$ time    : int 1214
  .. .. ..$ config  : chr "windows-release"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ artifact: chr "5093527311"
  ..$ _bioccheck        :List of 3
  .. ..$ error  : int 0
  .. ..$ warning: int 2
  .. ..$ note   : int 15
  ..$ _buildurl         : chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  ..$ _status           : chr "success"
  ..$ _host             : chr "GitHub-Actions"
  ..$ _upstream         : chr "https://github.com/bioc/singleCellTK"
  ..$ _commit           :List of 5
  .. ..$ id       : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. ..$ author   : chr "Joshua D. Campbell &lt;joshua.d.campbell@gmail.com&gt;"
  .. ..$ committer: chr "Joshua D. Campbell &lt;joshua.d.campbell@gmail.com&gt;"
  .. ..$ message  : chr "version bump and news\n"
  .. ..$ time     : int 1768138973
  ..$ _maintainer       :List of 5
  .. ..$ name : chr "Joshua David Campbell"
  .. ..$ email: chr "camp@bu.edu"
  .. ..$ login: chr "camplab"
  .. ..$ uuid : int 43044384
  .. ..$ orcid: chr "0000-0003-0780-8662"
  ..$ _distro           : chr "noble"
  ..$ _registered       : logi TRUE
  ..$ _dependencies     :List of 109
  .. ..$ :List of 3
  .. .. ..$ package: chr "R"
  .. .. ..$ version: chr "&gt;= 4.0"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SummarizedExperiment"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SingleCellExperiment"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DelayedArray"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Biobase"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ape"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "anndata"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "AnnotationHub"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "batchelor"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocParallel"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "celldex"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "colourpicker"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "colorspace"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "cowplot"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "cluster"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ComplexHeatmap"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "data.table"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DelayedMatrixStats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DESeq2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "dplyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DT"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ExperimentHub"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ensembldb"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "fields"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggplot2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggplotify"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggrepel"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggtree"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "gridExtra"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "grid"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GSVA"
  .. .. ..$ version: chr "&gt;= 1.50.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GSVAdata"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "igraph"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "KernSmooth"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "limma"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "MAST"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Matrix"
  .. .. ..$ version: chr "&gt;= 1.6-1"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "matrixStats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "methods"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "msigdbr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "multtest"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "plotly"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "plyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ROCR"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Rtsne"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "S4Vectors"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scater"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "scMerge"
  .. .. ..$ version: chr "&gt;= 1.2.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scran"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Seurat"
  .. .. ..$ version: chr "&gt;= 3.1.3"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "shiny"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "shinyjs"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SingleR"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stringr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SoupX"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "sva"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "reshape2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "shinyalert"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "circlize"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "enrichR"
  .. .. ..$ version: chr "&gt;= 3.2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "celda"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "shinycssloaders"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DropletUtils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "scds"
  .. .. ..$ version: chr "&gt;= 1.2.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "reticulate"
  .. .. ..$ version: chr "&gt;= 1.14"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tools"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tximport"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tidyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "eds"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "withr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GSEABase"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "R.utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "zinbwave"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "scRNAseq"
  .. .. ..$ version: chr "&gt;= 2.0.2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TENxPBMCData"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "yaml"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rmarkdown"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "magrittr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scDblFinder"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "metap"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "VAM"
  .. .. ..$ version: chr "&gt;= 0.5.3"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tibble"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rlang"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TSCAN"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TrajectoryUtils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scuttle"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "zellkonverter"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "lifecycle"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "testthat"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Rsubread"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocStyle"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "knitr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "lintr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "spelling"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "org.Mm.eg.db"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "kableExtra"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "shinythemes"
  .. .. ..$ role   : chr "Suggests"
  .. .. [list output truncated]
  ..$ _owner            : chr "bioc"
  ..$ _selfowned        : logi TRUE
  ..$ _usedby           : int 0
  ..$ _updates          :List of 10
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-07"
  .. .. ..$ n   : int 7
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-08"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-13"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-14"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-17"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-19"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-26"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-27"
  .. .. ..$ n   : int 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-39"
  .. .. ..$ n   : int 4
  .. ..$ :List of 2
  .. .. ..$ week: chr "2026-02"
  .. .. ..$ n   : int 3
  ..$ _tags             : list()
  ..$ _bioc             :List of 2
  .. ..$ :List of 3
  .. .. ..$ branch : chr "devel"
  .. .. ..$ version: chr "2.21.0"
  .. .. ..$ bioc   : chr "3.23"
  .. ..$ :List of 3
  .. .. ..$ branch : chr "release"
  .. .. ..$ version: chr "2.20.0"
  .. .. ..$ bioc   : chr "3.22"
  ..$ _topics           :List of 11
  .. ..$ : chr "singlecell"
  .. ..$ : chr "geneexpression"
  .. ..$ : chr "differentialexpression"
  .. ..$ : chr "alignment"
  .. ..$ : chr "clustering"
  .. ..$ : chr "immunooncology"
  .. ..$ : chr "batcheffect"
  .. ..$ : chr "normalization"
  .. ..$ : chr "qualitycontrol"
  .. ..$ : chr "dataimport"
  .. ..$ : chr "gui"
  ..$ _stars            : int 185
  ..$ _contributors     :List of 33
  .. ..$ :List of 3
  .. .. ..$ user : chr "irzamsarfraz"
  .. .. ..$ count: int 795
  .. .. ..$ uuid : int 33160268
  .. ..$ :List of 3
  .. .. ..$ user : chr "joshua-d-campbell"
  .. .. ..$ count: int 664
  .. .. ..$ uuid : int 14353256
  .. ..$ :List of 3
  .. .. ..$ user : chr "mvfki"
  .. .. ..$ count: int 547
  .. .. ..$ uuid : int 28736557
  .. ..$ :List of 3
  .. .. ..$ user : chr "dfjenkins3"
  .. .. ..$ count: int 521
  .. .. ..$ uuid : int 1692620
  .. ..$ :List of 3
  .. .. ..$ user : chr "salamhha"
  .. .. ..$ count: int 365
  .. .. ..$ uuid : int 50881225
  .. ..$ :List of 3
  .. .. ..$ user : chr "rz2333"
  .. .. ..$ count: int 295
  .. .. ..$ uuid : int 16445398
  .. ..$ :List of 3
  .. .. ..$ user : chr "ykoga07"
  .. .. ..$ count: int 292
  .. .. ..$ uuid : int 26311595
  .. ..$ :List of 3
  .. .. ..$ user : chr "liuming1997"
  .. .. ..$ count: int 102
  .. .. ..$ uuid : int 87452797
  .. ..$ :List of 3
  .. .. ..$ user : chr "npervaiz"
  .. .. ..$ count: int 97
  .. .. ..$ uuid : int 36128281
  .. ..$ :List of 3
  .. .. ..$ user : chr "zhewa"
  .. .. ..$ count: int 96
  .. .. ..$ uuid : int 16454489
  .. ..$ :List of 3
  .. .. ..$ user : chr "lateoclock"
  .. .. ..$ count: int 66
  .. .. ..$ uuid : int 22383641
  .. ..$ :List of 3
  .. .. ..$ user : chr "mmkhan19"
  .. .. ..$ count: int 61
  .. .. ..$ uuid : int 39277690
  .. ..$ :List of 3
  .. .. ..$ user : chr "vidyaap"
  .. .. ..$ count: int 58
  .. .. ..$ uuid : int 26637261
  .. ..$ :List of 3
  .. .. ..$ user : chr "akshayarav"
  .. .. ..$ count: int 55
  .. .. ..$ uuid : int 51174148
  .. ..$ :List of 3
  .. .. ..$ user : chr "mingl1997"
  .. .. ..$ count: int 50
  .. .. ..$ uuid : int 129070787
  .. ..$ :List of 3
  .. .. ..$ user : chr "xin-cao"
  .. .. ..$ count: int 40
  .. .. ..$ uuid : int 26312918
  .. ..$ :List of 3
  .. .. ..$ user : chr "ashastry2"
  .. .. ..$ count: int 32
  .. .. ..$ uuid : int 23105488
  .. ..$ :List of 3
  .. .. ..$ user : chr "shruthibandyadka"
  .. .. ..$ count: int 28
  .. .. ..$ uuid : int 20522149
  .. ..$ :List of 3
  .. .. ..$ user : chr "stc1"
  .. .. ..$ count: int 21
  .. .. ..$ uuid : int 25351396
  .. ..$ :List of 3
  .. .. ..$ user : chr "lloydliu717"
  .. .. ..$ count: int 18
  .. .. ..$ uuid : int 21338683
  .. ..$ :List of 3
  .. .. ..$ user : chr "ebriars"
  .. .. ..$ count: int 17
  .. .. ..$ uuid : int 24554622
  .. ..$ :List of 3
  .. .. ..$ user : chr "crelicthecleric"
  .. .. ..$ count: int 15
  .. .. ..$ uuid : int 52921332
  .. ..$ :List of 3
  .. .. ..$ user : chr "nturaga"
  .. .. ..$ count: int 14
  .. .. ..$ uuid : int 2746443
  .. ..$ :List of 3
  .. .. ..$ user : chr "tfaits"
  .. .. ..$ count: int 12
  .. .. ..$ uuid : int 12431337
  .. ..$ :List of 3
  .. .. ..$ user : chr "jwokaty"
  .. .. ..$ count: int 11
  .. .. ..$ uuid : int 1744257
  .. ..$ :List of 3
  .. .. ..$ user : chr "aleshchyk"
  .. .. ..$ count: int 9
  .. .. ..$ uuid : int 26057821
  .. ..$ :List of 3
  .. .. ..$ user : chr "frederickjansen"
  .. .. ..$ count: int 3
  .. .. ..$ uuid : int 864507
  .. ..$ :List of 3
  .. .. ..$ user : chr "irisapo"
  .. .. ..$ count: int 3
  .. .. ..$ uuid : int 16067676
  .. ..$ :List of 3
  .. .. ..$ user : chr "lshep"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 18724380
  .. ..$ :List of 3
  .. .. ..$ user : chr "vobencha"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 2466173
  .. ..$ :List of 3
  .. .. ..$ user : chr "hpages"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 8810451
  .. ..$ :List of 3
  .. .. ..$ user : chr "link-ny"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 4392950
  .. ..$ :List of 3
  .. .. ..$ user : chr "igordot"
  .. .. ..$ count: int 1
  .. .. ..$ uuid : int 6363505
  ..$ _userbio          :List of 4
  .. ..$ uuid       : int 2286807
  .. ..$ type       : chr "organization"
  .. ..$ name       : chr "Bioconductor"
  .. ..$ description: chr "Software for the analysis and comprehension of high-throughput genomic data"
  ..$ _downloads        :List of 2
  .. ..$ count : int 604
  .. ..$ source: chr "https://www.bioconductor.org/packages/stats/bioc/singleCellTK"
  ..$ _mentions         : int 2
  ..$ _devurl           : chr "https://github.com/compbiomed/singlecelltk"
  ..$ _searchresults    : int 248
  ..$ _rbuild           : chr "4.5.2"
  ..$ _assets           :List of 11
  .. ..$ : chr "extra/citation.cff"
  .. ..$ : chr "extra/citation.html"
  .. ..$ : chr "extra/citation.json"
  .. ..$ : chr "extra/citation.txt"
  .. ..$ : chr "extra/contents.json"
  .. ..$ : chr "extra/NEWS.html"
  .. ..$ : chr "extra/NEWS.txt"
  .. ..$ : chr "extra/readme.html"
  .. ..$ : chr "extra/readme.md"
  .. ..$ : chr "extra/singleCellTK.html"
  .. ..$ : chr "manual.pdf"
  ..$ _homeurl          : chr "https://github.com/compbiomed/singlecelltk"
  ..$ _realowner        : chr "bioc"
  ..$ _cranurl          : logi FALSE
  ..$ _exports          :List of 252
  .. ..$ : chr "calcEffectSizes"
  .. ..$ : chr "combineSCE"
  .. ..$ : chr "computeHeatmap"
  .. ..$ : chr "computeZScore"
  .. ..$ : chr "constructSCE"
  .. ..$ : chr "convertSCEToSeurat"
  .. ..$ : chr "convertSeuratToSCE"
  .. ..$ : chr "dedupRowNames"
  .. ..$ : chr "detectCellOutlier"
  .. ..$ : chr "diffAbundanceFET"
  .. ..$ : chr "discreteColorPalette"
  .. ..$ : chr "distinctColors"
  .. ..$ : chr "downSampleCells"
  .. ..$ : chr "downSampleDepth"
  .. ..$ : chr "expData"
  .. ..$ : chr "expData&lt;-"
  .. ..$ : chr "expDataNames"
  .. ..$ : chr "expDeleteDataTag"
  .. ..$ : chr "exportSCE"
  .. ..$ : chr "exportSCEtoAnnData"
  .. ..$ : chr "exportSCEtoFlatFile"
  .. ..$ : chr "exportSCEToSeurat"
  .. ..$ : chr "expSetDataTag"
  .. ..$ : chr "expTaggedData"
  .. ..$ : chr "featureIndex"
  .. ..$ : chr "findMarkerDiffExp"
  .. ..$ : chr "findMarkerTopTable"
  .. ..$ : chr "generateHTANMeta"
  .. ..$ : chr "generateMeta"
  .. ..$ : chr "generateSimulatedData"
  .. ..$ : chr "getBiomarker"
  .. ..$ : chr "getDEGTopTable"
  .. ..$ : chr "getDiffAbundanceResults"
  .. ..$ : chr "getDiffAbundanceResults&lt;-"
  .. ..$ : chr "getEnrichRResult"
  .. ..$ : chr "getEnrichRResult&lt;-"
  .. ..$ : chr "getFindMarkerTopTable"
  .. ..$ : chr "getGenesetNamesFromCollection"
  .. ..$ : chr "getMSigDBTable"
  .. ..$ : chr "getPathwayResultNames"
  .. ..$ : chr "getSampleSummaryStatsTable"
  .. ..$ : chr "getSceParams"
  .. ..$ : chr "getSeuratVariableFeatures"
  .. ..$ : chr "getSoupX"
  .. ..$ : chr "getSoupX&lt;-"
  .. ..$ : chr "getTopHVG"
  .. ..$ : chr "getTSCANResults"
  .. ..$ : chr "getTSCANResults&lt;-"
  .. ..$ : chr "getTSNE"
  .. ..$ : chr "getUMAP"
  .. ..$ : chr "importAlevin"
  .. ..$ : chr "importAnnData"
  .. ..$ : chr "importBUStools"
  .. ..$ : chr "importCellRanger"
  .. ..$ : chr "importCellRangerV2"
  .. ..$ : chr "importCellRangerV2Sample"
  .. ..$ : chr "importCellRangerV3"
  .. ..$ : chr "importCellRangerV3Sample"
  .. ..$ : chr "importDropEst"
  .. ..$ : chr "importExampleData"
  .. ..$ : chr "importFromFiles"
  .. ..$ : chr "importGeneSetsFromCollection"
  .. ..$ : chr "importGeneSetsFromGMT"
  .. ..$ : chr "importGeneSetsFromList"
  .. ..$ : chr "importGeneSetsFromMSigDB"
  .. ..$ : chr "importMitoGeneSet"
  .. ..$ : chr "importMultipleSources"
  .. ..$ : chr "importOptimus"
  .. ..$ : chr "importSEQC"
  .. ..$ : chr "importSTARsolo"
  .. ..$ : chr "iterateSimulations"
  .. ..$ : chr "listSampleSummaryStatsTables"
  .. ..$ : chr "listTSCANResults"
  .. ..$ : chr "listTSCANTerminalNodes"
  .. ..$ : chr "mergeSCEColData"
  .. ..$ : chr "plotBarcodeRankDropsResults"
  .. ..$ : chr "plotBarcodeRankScatter"
  .. ..$ : chr "plotBatchCorrCompare"
  .. ..$ : chr "plotBatchVariance"
  .. ..$ : chr "plotBcdsResults"
  .. ..$ : chr "plotBubble"
  .. ..$ : chr "plotClusterAbundance"
  .. ..$ : chr "plotCxdsResults"
  .. ..$ : chr "plotDecontXResults"
  .. ..$ : chr "plotDEGHeatmap"
  .. ..$ : chr "plotDEGRegression"
  .. ..$ : chr "plotDEGViolin"
  .. ..$ : chr "plotDEGVolcano"
  .. ..$ : chr "plotDimRed"
  .. ..$ : chr "plotDoubletFinderResults"
  .. ..$ : chr "plotEmptyDropsResults"
  .. ..$ : chr "plotEmptyDropsScatter"
  .. ..$ : chr "plotEnrichR"
  .. ..$ : chr "plotFindMarkerHeatmap"
  .. ..$ : chr "plotMarkerDiffExp"
  .. ..$ : chr "plotMASTThresholdGenes"
  .. ..$ : chr "plotPathway"
  .. ..$ : chr "plotPCA"
  .. ..$ : chr "plotRunPerCellQCResults"
  .. .. [list output truncated]
  ..$ _datasets         :List of 6
  .. ..$ :List of 8
  .. .. ..$ name  : chr "MitoGenes"
  .. .. ..$ title : chr "List of mitochondrial genes of multiple reference"
  .. .. ..$ object: chr "MitoGenes"
  .. .. ..$ file  : chr "MitoGenes.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "list"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi TRUE
  .. .. ..$ tojson: logi TRUE
  .. ..$ :List of 8
  .. .. ..$ name  : chr "SEG"
  .. .. ..$ title : chr "Stably Expressed Gene (SEG) list obect, with SEG sets for human and mouse."
  .. .. ..$ object: chr "SEG"
  .. .. ..$ file  : chr "SEG.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "list"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi TRUE
  .. ..$ :List of 8
  .. .. ..$ name  : chr "mouseBrainSubsetSCE"
  .. .. ..$ title : chr "Example Single Cell RNA-Seq data in SingleCellExperiment Object, GSE60361 subset"
  .. .. ..$ object: chr "mouseBrainSubsetSCE"
  .. .. ..$ file  : chr "mouseBrainSubsetSCE.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "SingleCellExperiment"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 9
  .. .. ..$ name  : chr "msigdb_table"
  .. .. ..$ title : chr "MSigDB gene get Category table"
  .. .. ..$ object: chr "msigdb_table"
  .. .. ..$ file  : chr "msigdb_table.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "data.frame"
  .. .. ..$ fields:List of 5
  .. .. .. ..$ : chr "ID"
  .. .. .. ..$ : chr "Category"
  .. .. .. ..$ : chr "Subcategory"
  .. .. .. ..$ : chr "Category_Description"
  .. .. .. ..$ : chr "Subcategory_Description"
  .. .. ..$ rows  : int 17
  .. .. ..$ table : logi TRUE
  .. .. ..$ tojson: logi TRUE
  .. ..$ :List of 8
  .. .. ..$ name  : chr "sce"
  .. .. ..$ title : chr "Example Single Cell RNA-Seq data in SingleCellExperiment Object, subset of 10x public dataset"
  .. .. ..$ object: chr "scExample"
  .. .. ..$ file  : chr "scExample.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "SingleCellExperiment"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 8
  .. .. ..$ name  : chr "sceBatches"
  .. .. ..$ title : chr "Example Single Cell RNA-Seq data in SingleCellExperiment object, with different batches annotated"
  .. .. ..$ object: chr "sceBatches"
  .. .. ..$ file  : chr "sceBatches.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "SingleCellExperiment"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  ..$ _help             :List of 238
  .. ..$ :List of 3
  .. .. ..$ page  : chr "calcEffectSizes"
  .. .. ..$ title : chr "Finds the effect sizes for all genes in the original dataset, regardless of significance."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "calcEffectSizes"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "combineSCE"
  .. .. ..$ title : chr "Combine a list of SingleCellExperiment objects as one SingleCellExperiment object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "combineSCE"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "computeHeatmap"
  .. .. ..$ title : chr "Computes heatmap for a set of features against dimensionality reduction components"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "computeHeatmap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "computeZScore"
  .. .. ..$ title : chr "Compute Z-Score"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "computeZScore"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "constructSCE"
  .. .. ..$ title : chr "Create SingleCellExperiment object from csv or txt input"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "constructSCE"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convertSCEToSeurat"
  .. .. ..$ title : chr "convertSCEToSeurat Converts sce object to seurat while retaining all assays and metadata"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convertSCEToSeurat"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convertSeuratToSCE"
  .. .. ..$ title : chr "convertSeuratToSCE Converts the input seurat object to a sce object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convertSeuratToSCE"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "dedupRowNames"
  .. .. ..$ title : chr "Deduplicate the rownames of a matrix or SingleCellExperiment object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "dedupRowNames"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "detectCellOutlier"
  .. .. ..$ title : chr "Detecting outliers within the SingleCellExperiment object."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "detectCellOutlier"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "diffAbundanceFET"
  .. .. ..$ title : chr "Calculate Differential Abundance with FET"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "diffAbundanceFET"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "discreteColorPalette"
  .. .. ..$ title : chr "Generate given number of color codes"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "discreteColorPalette"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "distinctColors"
  .. .. ..$ title : chr "Generate a distinct palette for coloring different clusters"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "distinctColors"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "downSampleCells"
  .. .. ..$ title : chr "Estimate numbers of detected genes, significantly differentially expressed genes, and median significant effect size"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "downSampleCells"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "downSampleDepth"
  .. .. ..$ title : chr "Estimate numbers of detected genes, significantly differentially expressed genes, and median significant effect size"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "downSampleDepth"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expData"
  .. .. ..$ title : chr "expData Get data item from an input 'SingleCellExperiment' object. The data item can be an 'assay', 'altExp' (s"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expData-ANY-character-method"
  .. .. ..$ title : chr "expData Get data item from an input 'SingleCellExperiment' object. The data item can be an 'assay', 'altExp' (s"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expData,ANY,character-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expData-set"
  .. .. ..$ title : chr "expData Store data items using tags to identify the type of data item stored. To be used as a replacement for a"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expData&lt;-"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expData-set-ANY-character-CharacterOrNullOrMissing-logical-method"
  .. .. ..$ title : chr "expData Store data items using tags to identify the type of data item stored. To be used as a replacement for a"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expData&lt;-,ANY,character,CharacterOrNullOrMissing,logical-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expDataNames"
  .. .. ..$ title : chr "expDataNames Get names of all the data items in the input 'SingleCellExperiment' object including assays, altEx"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expDataNames"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expDataNames-ANY-method"
  .. .. ..$ title : chr "expDataNames Get names of all the data items in the input 'SingleCellExperiment' object including assays, altEx"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expDataNames,ANY-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expDeleteDataTag"
  .. .. ..$ title : chr "expDeleteDataTag Remove tag against an input data from the stored tag information in the metadata of the input object."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expDeleteDataTag"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "exportSCE"
  .. .. ..$ title : chr "Export data in SingleCellExperiment object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "exportSCE"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "exportSCEtoAnnData"
  .. .. ..$ title : chr "Export a SingleCellExperiment R object as Python annData object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "exportSCEtoAnnData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "exportSCEtoFlatFile"
  .. .. ..$ title : chr "Export a SingleCellExperiment object to flat text files"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "exportSCEtoFlatFile"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "exportSCEToSeurat"
  .. .. ..$ title : chr "Export data in Seurat object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "exportSCEToSeurat"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expSetDataTag"
  .. .. ..$ title : chr "expSetDataTag Set tag to an assay or a data item in the input SCE object."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expSetDataTag"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "expTaggedData"
  .. .. ..$ title : chr "expTaggedData Returns a list of names of data items from the input 'SingleCellExperiment' object based upon the"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "expTaggedData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "featureIndex"
  .. .. ..$ title : chr "Retrieve row index for a set of features"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "featureIndex"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "generateHTANMeta"
  .. .. ..$ title : chr "Generate HTAN manifest file for droplet and cell count data"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "generateHTANMeta"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "generateMeta"
  .. .. ..$ title : chr "Generate HTAN manifest file for droplet and cell count data"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "generateMeta"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "generateSimulatedData"
  .. .. ..$ title : chr "Generates a single simulated dataset, bootstrapping from the input counts matrix."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "generateSimulatedData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getBiomarker"
  .. .. ..$ title : chr "Given a list of genes and a SingleCellExperiment object, return the binary or continuous expression of the genes."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getBiomarker"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getDEGTopTable"
  .. .. ..$ title : chr "Get Top Table of a DEG analysis"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getDEGTopTable"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getDiffAbundanceResults"
  .. .. ..$ title : chr "Get/Set diffAbundanceFET result table"
  .. .. ..$ topics:List of 4
  .. .. .. ..$ : chr "getDiffAbundanceResults"
  .. .. .. ..$ : chr "getDiffAbundanceResults,SingleCellExperiment-method"
  .. .. .. ..$ : chr "getDiffAbundanceResults&lt;-"
  .. .. .. ..$ : chr "getDiffAbundanceResults&lt;-,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getEnrichRResult"
  .. .. ..$ title : chr "Get or Set EnrichR Result"
  .. .. ..$ topics:List of 4
  .. .. .. ..$ : chr "getEnrichRResult"
  .. .. .. ..$ : chr "getEnrichRResult,SingleCellExperiment-method"
  .. .. .. ..$ : chr "getEnrichRResult&lt;-"
  .. .. .. ..$ : chr "getEnrichRResult&lt;-,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getFindMarkerTopTable"
  .. .. ..$ title : chr "Fetch the table of top markers that pass the filtering"
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "findMarkerTopTable"
  .. .. .. ..$ : chr "getFindMarkerTopTable"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getGenesetNamesFromCollection"
  .. .. ..$ title : chr "List geneset names from geneSetCollection"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getGenesetNamesFromCollection"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getMSigDBTable"
  .. .. ..$ title : chr "Shows MSigDB categories"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getMSigDBTable"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getPathwayResultNames"
  .. .. ..$ title : chr "List pathway analysis result names"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getPathwayResultNames"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getSampleSummaryStatsTable"
  .. .. ..$ title : chr "Stores and returns table of SCTK QC outputs to metadata."
  .. .. ..$ topics:List of 4
  .. .. .. ..$ : chr "getSampleSummaryStatsTable"
  .. .. .. ..$ : chr "getSampleSummaryStatsTable,SingleCellExperiment-method"
  .. .. .. ..$ : chr "setSampleSummaryStatsTable&lt;-"
  .. .. .. ..$ : chr "setSampleSummaryStatsTable&lt;-,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getSceParams"
  .. .. ..$ title : chr "Extract QC parameters from the SingleCellExperiment object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getSceParams"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getSeuratVariableFeatures"
  .. .. ..$ title : chr "Get variable feature names after running runSeuratFindHVG function"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getSeuratVariableFeatures"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getSoupX"
  .. .. ..$ title : chr "Get or Set SoupX Result"
  .. .. ..$ topics:List of 4
  .. .. .. ..$ : chr "getSoupX"
  .. .. .. ..$ : chr "getSoupX,SingleCellExperiment-method"
  .. .. .. ..$ : chr "getSoupX&lt;-"
  .. .. .. ..$ : chr "getSoupX&lt;-,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getTopHVG"
  .. .. ..$ title : chr "Get or set top HVG after calculation"
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "getTopHVG"
  .. .. .. ..$ : chr "setTopHVG"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getTSCANResults"
  .. .. ..$ title : chr "getTSCANResults accessor function"
  .. .. ..$ topics:List of 8
  .. .. .. ..$ : chr "getTSCANResults"
  .. .. .. ..$ : chr "getTSCANResults,SingleCellExperiment-method"
  .. .. .. ..$ : chr "getTSCANResults&lt;-"
  .. .. .. ..$ : chr "getTSCANResults&lt;-,SingleCellExperiment-method"
  .. .. .. ..$ : chr "listTSCANResults"
  .. .. .. ..$ : chr "listTSCANResults,SingleCellExperiment-method"
  .. .. .. ..$ : chr "listTSCANTerminalNodes"
  .. .. .. ..$ : chr "listTSCANTerminalNodes,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importAlevin"
  .. .. ..$ title : chr "Construct SCE object from Salmon-Alevin output"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importAlevin"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importAnnData"
  .. .. ..$ title : chr "Create a SingleCellExperiment Object from Python AnnData .h5ad files"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importAnnData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importBUStools"
  .. .. ..$ title : chr "Construct SCE object from BUStools output"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importBUStools"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importCellRanger"
  .. .. ..$ title : chr "Construct SCE object from Cell Ranger output"
  .. .. ..$ topics:List of 3
  .. .. .. ..$ : chr "importCellRanger"
  .. .. .. ..$ : chr "importCellRangerV2"
  .. .. .. ..$ : chr "importCellRangerV3"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importCellRangerV2Sample"
  .. .. ..$ title : chr "Construct SCE object from Cell Ranger V2 output for a single sample"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importCellRangerV2Sample"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importCellRangerV3Sample"
  .. .. ..$ title : chr "Construct SCE object from Cell Ranger V3 output for a single sample"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importCellRangerV3Sample"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importDropEst"
  .. .. ..$ title : chr "Create a SingleCellExperiment Object from DropEst output"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importDropEst"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importExampleData"
  .. .. ..$ title : chr "Retrieve example datasets"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importExampleData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importFromFiles"
  .. .. ..$ title : chr "Create a SingleCellExperiment object from files"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importFromFiles"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importGeneSetsFromCollection"
  .. .. ..$ title : chr "Imports gene sets from a GeneSetCollection object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importGeneSetsFromCollection"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importGeneSetsFromGMT"
  .. .. ..$ title : chr "Imports gene sets from a GMT file"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importGeneSetsFromGMT"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importGeneSetsFromList"
  .. .. ..$ title : chr "Imports gene sets from a list"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importGeneSetsFromList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importGeneSetsFromMSigDB"
  .. .. ..$ title : chr "Imports gene sets from MSigDB"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importGeneSetsFromMSigDB"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importMitoGeneSet"
  .. .. ..$ title : chr "Import mitochondrial gene sets"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importMitoGeneSet"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importMultipleSources"
  .. .. ..$ title : chr "Imports samples from different sources and compiles them into a list of SCE objects"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importMultipleSources"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importOptimus"
  .. .. ..$ title : chr "Construct SCE object from Optimus output"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importOptimus"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importSEQC"
  .. .. ..$ title : chr "Construct SCE object from seqc output"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importSEQC"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "importSTARsolo"
  .. .. ..$ title : chr "Construct SCE object from STARsolo outputs"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "importSTARsolo"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "iterateSimulations"
  .. .. ..$ title : chr "Returns significance data from a snapshot."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "iterateSimulations"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "listSampleSummaryStatsTables"
  .. .. ..$ title : chr "Lists the table of SCTK QC outputs stored within the metadata."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "listSampleSummaryStatsTables"
  .. .. .. ..$ : chr "listSampleSummaryStatsTables,SingleCellExperiment-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "mergeSCEColData"
  .. .. ..$ title : chr "Merging colData from two singleCellExperiment objects"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "mergeSCEColData"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "MitoGenes"
  .. .. ..$ title : chr "List of mitochondrial genes of multiple reference"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "MitoGenes"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "mouseBrainSubsetSCE"
  .. .. ..$ title : chr "Example Single Cell RNA-Seq data in SingleCellExperiment Object, GSE60361 subset"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "mouseBrainSubsetSCE"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "msigdb_table"
  .. .. ..$ title : chr "MSigDB gene get Category table"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "msigdb_table"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBarcodeRankDropsResults"
  .. .. ..$ title : chr "Plots for runBarcodeRankDrops outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBarcodeRankDropsResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBarcodeRankScatter"
  .. .. ..$ title : chr "Plots for runBarcodeRankDrops outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBarcodeRankScatter"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBatchCorrCompare"
  .. .. ..$ title : chr "Plot comparison of batch corrected result against original assay"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBatchCorrCompare"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBatchVariance"
  .. .. ..$ title : chr "Plot the percent of the variation that is explained by batch and condition in the data"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBatchVariance"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBcdsResults"
  .. .. ..$ title : chr "Plots for runBcds outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBcdsResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBubble"
  .. .. ..$ title : chr "Plot Bubble plot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBubble"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotClusterAbundance"
  .. .. ..$ title : chr "Plot the differential Abundance"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotClusterAbundance"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotCxdsResults"
  .. .. ..$ title : chr "Plots for runCxds outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotCxdsResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDecontXResults"
  .. .. ..$ title : chr "Plots for runDecontX outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDecontXResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDEGHeatmap"
  .. .. ..$ title : chr "Heatmap visualization of DEG result"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDEGHeatmap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDEGRegression"
  .. .. ..$ title : chr "Create linear regression plot to show the expression the of top DEGs"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDEGRegression"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDEGViolin"
  .. .. ..$ title : chr "Generate violin plot to show the expression of top DEGs"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDEGViolin"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDEGVolcano"
  .. .. ..$ title : chr "Generate volcano plot for DEGs"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDEGVolcano"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDimRed"
  .. .. ..$ title : chr "Plot dimensionality reduction from computed metrics including PCA, ICA, tSNE and UMAP"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDimRed"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotDoubletFinderResults"
  .. .. ..$ title : chr "Plots for runDoubletFinder outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotDoubletFinderResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotEmptyDropsResults"
  .. .. ..$ title : chr "Plots for runEmptyDrops outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotEmptyDropsResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotEmptyDropsScatter"
  .. .. ..$ title : chr "Plots for runEmptyDrops outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotEmptyDropsScatter"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotEnrichR"
  .. .. ..$ title : chr "Plot EnrichR results"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotEnrichR"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotFindMarkerHeatmap"
  .. .. ..$ title : chr "Plot a heatmap to visualize the result of 'runFindMarker'"
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "plotFindMarkerHeatmap"
  .. .. .. ..$ : chr "plotMarkerDiffExp"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotMASTThresholdGenes"
  .. .. ..$ title : chr "MAST Identify adaptive thresholds"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotMASTThresholdGenes"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotPathway"
  .. .. ..$ title : chr "Generate violin plots for pathway analysis results"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotPathway"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotPCA"
  .. .. ..$ title : chr "Plot PCA run data from its components."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotPCA"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotRunPerCellQCResults"
  .. .. ..$ title : chr "Plots for runPerCellQC outputs."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotRunPerCellQCResults"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyDotPlot"
  .. .. ..$ title : chr "plotScanpyDotPlot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyDotPlot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyEmbedding"
  .. .. ..$ title : chr "plotScanpyEmbedding"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyEmbedding"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyHeatmap"
  .. .. ..$ title : chr "plotScanpyHeatmap"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyHeatmap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyHVG"
  .. .. ..$ title : chr "plotScanpyHVG"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyHVG"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyMarkerGenes"
  .. .. ..$ title : chr "plotScanpyMarkerGenes"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyMarkerGenes"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyMarkerGenesDotPlot"
  .. .. ..$ title : chr "plotScanpyMarkerGenesDotPlot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyMarkerGenesDotPlot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotScanpyMarkerGenesHeatmap"
  .. .. ..$ title : chr "plotScanpyMarkerGenesHeatmap"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotScanpyMarkerGenesHeatmap"
  .. .. [list output truncated]
  ..$ _readme           : chr "https://github.com/bioc/singleCellTK/raw/HEAD/README.md"
  ..$ _rundeps          :List of 382
  .. ..$ : chr "abind"
  .. ..$ : chr "alabaster.base"
  .. ..$ : chr "alabaster.matrix"
  .. ..$ : chr "alabaster.ranges"
  .. ..$ : chr "alabaster.sce"
  .. ..$ : chr "alabaster.schemas"
  .. ..$ : chr "alabaster.se"
  .. ..$ : chr "anndata"
  .. ..$ : chr "annotate"
  .. ..$ : chr "AnnotationDbi"
  .. ..$ : chr "AnnotationFilter"
  .. ..$ : chr "AnnotationHub"
  .. ..$ : chr "ape"
  .. ..$ : chr "aplot"
  .. ..$ : chr "askpass"
  .. ..$ : chr "assertthat"
  .. ..$ : chr "assorthead"
  .. ..$ : chr "babelgene"
  .. ..$ : chr "backports"
  .. ..$ : chr "base64enc"
  .. ..$ : chr "basilisk"
  .. ..$ : chr "batchelor"
  .. ..$ : chr "bbmle"
  .. ..$ : chr "bdsmatrix"
  .. ..$ : chr "beachmat"
  .. ..$ : chr "beeswarm"
  .. ..$ : chr "BH"
  .. ..$ : chr "Biobase"
  .. ..$ : chr "BiocBaseUtils"
  .. ..$ : chr "BiocFileCache"
  .. ..$ : chr "BiocGenerics"
  .. ..$ : chr "BiocIO"
  .. ..$ : chr "BiocManager"
  .. ..$ : chr "BiocNeighbors"
  .. ..$ : chr "BiocParallel"
  .. ..$ : chr "BiocSingular"
  .. ..$ : chr "BiocVersion"
  .. ..$ : chr "Biostrings"
  .. ..$ : chr "bit"
  .. ..$ : chr "bit64"
  .. ..$ : chr "bitops"
  .. ..$ : chr "blob"
  .. ..$ : chr "bluster"
  .. ..$ : chr "bslib"
  .. ..$ : chr "cachem"
  .. ..$ : chr "Cairo"
  .. ..$ : chr "callr"
  .. ..$ : chr "caTools"
  .. ..$ : chr "celda"
  .. ..$ : chr "celldex"
  .. ..$ : chr "checkmate"
  .. ..$ : chr "cigarillo"
  .. ..$ : chr "circlize"
  .. ..$ : chr "cli"
  .. ..$ : chr "clue"
  .. ..$ : chr "cluster"
  .. ..$ : chr "codetools"
  .. ..$ : chr "colorspace"
  .. ..$ : chr "colourpicker"
  .. ..$ : chr "combinat"
  .. ..$ : chr "commonmark"
  .. ..$ : chr "ComplexHeatmap"
  .. ..$ : chr "cowplot"
  .. ..$ : chr "cpp11"
  .. ..$ : chr "crayon"
  .. ..$ : chr "crosstalk"
  .. ..$ : chr "curl"
  .. ..$ : chr "cvTools"
  .. ..$ : chr "data.table"
  .. ..$ : chr "DBI"
  .. ..$ : chr "dbplyr"
  .. ..$ : chr "dbscan"
  .. ..$ : chr "DelayedArray"
  .. ..$ : chr "DelayedMatrixStats"
  .. ..$ : chr "deldir"
  .. ..$ : chr "dendextend"
  .. ..$ : chr "densEstBayes"
  .. ..$ : chr "DEoptimR"
  .. ..$ : chr "desc"
  .. ..$ : chr "DESeq2"
  .. ..$ : chr "digest"
  .. ..$ : chr "dir.expiry"
  .. ..$ : chr "distr"
  .. ..$ : chr "distributional"
  .. ..$ : chr "doParallel"
  .. ..$ : chr "dotCall64"
  .. ..$ : chr "dplyr"
  .. ..$ : chr "dqrng"
  .. ..$ : chr "DropletUtils"
  .. ..$ : chr "DT"
  .. ..$ : chr "edgeR"
  .. ..$ : chr "eds"
  .. ..$ : chr "enrichR"
  .. ..$ : chr "ensembldb"
  .. ..$ : chr "evaluate"
  .. ..$ : chr "ExperimentHub"
  .. ..$ : chr "farver"
  .. ..$ : chr "fastDummies"
  .. ..$ : chr "fastICA"
  .. .. [list output truncated]
  ..$ _vignettes        :List of 1
  .. ..$ :List of 9
  .. .. ..$ source  : chr "singleCellTK.Rmd"
  .. .. ..$ filename: chr "singleCellTK.html"
  .. .. ..$ title   : chr "Introduction to singleCellTK"
  .. .. ..$ author  : chr "Yichen Wang, Irzam Sarfraz, Rui Hong, Yusuke Koga, Salam Abdullatif, Nida Pervaiz, David Jenkins, Vidya Akavoor"| __truncated__
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 2
  .. .. .. ..$ : chr "Vignettes"
  .. .. .. ..$ : chr "Session info"
  .. .. ..$ created : chr "2020-10-12 13:15:30"
  .. .. ..$ modified: chr "2023-07-26 18:31:43"
  .. .. ..$ commits : int 4
  ..$ _score            : num 10.2
  ..$ _indexed          : logi TRUE
  ..$ _nocasepkg        : chr "singlecelltk"
  ..$ _universes        :List of 3
  .. ..$ : chr "bioc"
  .. ..$ : chr "camplab"
  .. ..$ : chr "compbiomed"
  ..$ _previous         : chr "2.21.0"
  ..$ _binaries         :List of 8
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:07:36.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "848bc88ee2ffb2903368223ce2a64d5a2884f96dbd893d7636e0e928a6da2b72"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:07:16.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "bdd55c3cc1fe27b2c2c81c8f80c5b21df908fce160e4617e01c25c1fb73ce7e8"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:04:34.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "558fa24201af3165fdcf6789049cda6ebd749c16316c53bf3b8bf6f12754918f"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:04:54.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "18d0595a8b1aa839a348d8afe4e8a95e463de5e01d74bd72ccea0024ec5bb0d3"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 8
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "wasm"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:08:23.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "773585d59b68a20b0088ba636c16c61bf4641b31c7f706f824efe857a2a9d868"
  .. .. ..$ status  : chr "success"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:05:49.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "934af5ec34ef3826eae6bee043e5e939f258b038eb360291b0424a7fe0a72015"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:07:55.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "2124e861ae3d5f53eb8be51b503467aec9cc03fefe047b1c9c8674a6f17cae18"
  .. .. ..$ status  : chr "failure"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "2.21.1"
  .. .. ..$ date    : chr "2026-01-12T04:06:39.000Z"
  .. .. ..$ commit  : chr "15d4a132fee6e116f5b5ac3ce78ca266d010faaa"
  .. .. ..$ fileid  : chr "b747ca2d50c96a8503606293beaaae940b644a323ea9931bb22523cf66a089a1"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "NOTE"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/20907471105"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>