<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>exploreBiocUniverse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="exploreBiocUniverse_files/libs/clipboard/clipboard.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/popper.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/anchor.min.js"></script>
<link href="exploreBiocUniverse_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="exploreBiocUniverse_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-6650302f782765824de81675458e0c02.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="exploreBiocUniverse_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="exploreBiocUniverse_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">exploreBiocUniverse</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="comparing-bioconductor-packages-in-r-universe-and-the-bbs" class="level1">
<h1>Comparing Bioconductor packages in R-Universe and the BBS</h1>
<p>Skip to <a href="#packages-not-in-agreement">Packages not in agreements</a> or go to <a href="windows.qmd">windows</a>.</p>
<section id="r-universe" class="level2">
<h2 class="anchored" data-anchor-id="r-universe">R Universe</h2>
<p>We can use the <a href="https://github.com/ropensci/universe/">universe</a> package to explore the R-Universe (RU) API. (We might need to increase the <code>max.print</code> to show all packages.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universe)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">max.print =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data for the Bioc Universe (BU).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs_i <span class="ot">&lt;-</span> <span class="fu">universe_all_packages</span>(<span class="st">"bioc"</span>, <span class="at">limit =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What packages are in bioc universe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs <span class="ot">&lt;-</span> <span class="fu">universe_ls</span>(<span class="st">"bioc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bbs" class="level2">
<h2 class="anchored" data-anchor-id="bbs">BBS</h2>
<p>Get Bioconductor packages in BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocPkgTools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bbs_pkgs_i <span class="ot">&lt;-</span> BiocPkgTools<span class="sc">::</span><span class="fu">biocBuildReport</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">version =</span> BiocManager<span class="sc">::</span><span class="fu">version</span>(),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgType =</span> <span class="st">"software"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>bbs_pkgs <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unique</span>(bbs_pkgs_i<span class="sc">$</span>pkg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Packages in RU vs BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">RU =</span> <span class="fu">length</span>(ru_pkgs), <span class="at">BBS =</span> <span class="fu">length</span>(bbs_pkgs),</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"Total"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        RU  BBS
Total 2325 2361</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>not_in_ru <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(bbs_pkgs, ru_pkgs)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>not_in_bbs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(ru_pkgs, bbs_pkgs)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># There's probably a better way to do this</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">Created =</span> <span class="fu">character</span>(), <span class="at">Published =</span> <span class="fu">character</span>(),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">RemoteUrl =</span> <span class="fu">character</span>())</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> not_in_bbs) {</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  pi <span class="ot">&lt;-</span> <span class="fu">universe_one_package</span>(<span class="st">"bioc"</span>, pkg)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  pkgs_not_in_bbs <span class="ot">&lt;-</span> pkgs_not_in_bbs <span class="sc">|&gt;</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> pi<span class="sc">$</span>Package, <span class="at">Version =</span> pi<span class="sc">$</span>Version,</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">Created =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_created</span><span class="st">`</span>, <span class="at">Published =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_published</span><span class="st">`</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">RemoteUrl =</span> pi<span class="sc">$</span>RemoteUrl)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-not-in-bbs-but-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-not-in-bbs-but-in-ru">Packages not in BBS but in RU</h3>
<p>Packages in RU but not in the BBS are likely added by maintainers via <a href="https://github.com/r-universe/bioc/blob/master/.remotes.json" class="uri">https://github.com/r-universe/bioc/blob/master/.remotes.json</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="sc">|&gt;</span> knitr<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Created</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Published</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RemoteUrl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">SpiecEasi</td>
<td style="text-align: left;">1.99.3</td>
<td style="text-align: left;">2025-11-06T03:26:26.000Z</td>
<td style="text-align: left;">2025-11-06T03:38:14.161Z</td>
<td style="text-align: left;">https://github.com/bioc/SpiecEasi</td>
</tr>
<tr class="even">
<td style="text-align: left;">dominatR</td>
<td style="text-align: left;">0.99.4</td>
<td style="text-align: left;">2025-11-06T03:25:15.000Z</td>
<td style="text-align: left;">2025-11-06T03:31:42.129Z</td>
<td style="text-align: left;">https://github.com/bioc/dominatR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ggpattern</td>
<td style="text-align: left;">1.1.4</td>
<td style="text-align: left;">2025-11-11T06:57:27.000Z</td>
<td style="text-align: left;">2025-11-11T07:10:04.396Z</td>
<td style="text-align: left;">https://github.com/coolbutuseless/ggpattern</td>
</tr>
<tr class="even">
<td style="text-align: left;">lmms</td>
<td style="text-align: left;">1.3.3</td>
<td style="text-align: left;">2025-10-23T03:46:27.000Z</td>
<td style="text-align: left;">2025-10-23T03:52:01.356Z</td>
<td style="text-align: left;">https://github.com/cran/lmms</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="packages-in-bbs-but-not-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-in-bbs-but-not-in-ru">Packages in BBS but not in RU</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>not_in_ru</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "bgx"                    "BiGGR"                  "biodbHmdb"             
 [4] "biodbNcbi"              "biodbNci"               "biodbUniprot"          
 [7] "ccmap"                  "CellScore"              "CINdex"                
[10] "cisPath"                "DEP"                    "gpuMagic"              
[13] "Harshlight"             "hiAnnotator"            "hiReadsProcessor"      
[16] "hypeR"                  "interactiveDisplay"     "interactiveDisplayBase"
[19] "lapmix"                 "LinTInd"                "lute"                  
[22] "MADSEQ"                 "netZooR"                "oppti"                 
[25] "PhenStat"               "qckitfastq"             "ReactomeGraph4R"       
[28] "Repitools"              "Rfastp"                 "rGADEM"                
[31] "rRDP"                   "SARC"                   "seqArchR"              
[34] "seqArchRplus"           "seqTools"               "Streamer"              
[37] "TitanCNA"               "TransView"              "traviz"                
[40] "XNAString"             </code></pre>
</div>
</div>
<p>The following packages are not deprecated but are not in RU.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>depre_pkgs <span class="ot">&lt;-</span> <span class="fu">filter</span>(bbs_pkgs_i, Deprecated <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(pkg) <span class="sc">|&gt;</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(not_in_ru, depre_pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "bgx"          "biodbHmdb"    "biodbNcbi"    "biodbNci"     "biodbUniprot"
[6] "cisPath"      "hiAnnotator"  "seqArchRplus" "traviz"      </code></pre>
</div>
</div>
</section>
</section>
<section id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc." class="level2">
<h2 class="anchored" data-anchor-id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc.">Making sense of R Universe’s build status in respect to R CMD build, check, etc.</h2>
<p>Looking at the status of linux x86_64 binary builds. The build “status” appears to get “success”, “failure”, or “cancelled” from the GitHub Action overall. The “check” value corresponds to <code>R CMD check</code>. I interpret the result of “status” to be “success” OK and “failure” ERROR for <code>R CMD build</code> but it may mean something else. We’ll use CANCELLED as it is. There’s also a special status for documentation, which is handled separately from build/check, which may account for some discrepancies. Not sure if the build status for vignettes is available in the API.</p>
<p>Packages with my comment “Missing R4.6 on Linux” did have data in the API but no x86_64 linux with R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Status =</span> <span class="fu">character</span>(), <span class="at">Build =</span> <span class="fu">character</span>(),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Check =</span> <span class="fu">character</span>(), <span class="at">Comment =</span> <span class="fu">character</span>())</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#, Buildurl = character(),</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># Commit = character())</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> ru_pkgs_i) {</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> <span class="cn">NA</span>, <span class="at">build =</span> <span class="cn">NA</span>, <span class="at">check =</span> <span class="cn">NA</span>, <span class="at">buildurl =</span> <span class="cn">NA</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  comment <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (b <span class="cf">in</span> p<span class="sc">$</span><span class="st">`</span><span class="at">_binaries</span><span class="st">`</span>) {</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (b<span class="sc">$</span>os <span class="sc">==</span> <span class="st">"linux"</span> <span class="sc">&amp;&amp;</span> b<span class="sc">$</span>r <span class="sc">==</span> <span class="st">"4.6.0"</span>) {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>      build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> b<span class="sc">$</span>status, <span class="at">check =</span> b<span class="sc">$</span>check, <span class="at">buildurl =</span> b<span class="sc">$</span>buildurl)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(build[<span class="st">"status"</span>])) {</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> <span class="st">"Missing R4.6 on Linux"</span> </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"success"</span>) {</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"failure"</span>) {</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"ERROR*"</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"cancelled"</span>) {</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"CANCELLED"</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  ru_status <span class="ot">&lt;-</span> ru_status <span class="sc">|&gt;</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> p<span class="sc">$</span>Package, <span class="at">Version =</span> p<span class="sc">$</span>Version, <span class="at">Status =</span> build[<span class="st">"status"</span>],</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">Build =</span> build[<span class="st">"build"</span>], <span class="at">Check =</span> build[<span class="st">"check"</span>], <span class="at">Comment =</span> comment)</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>            <span class="co">#, Buildurl = build["buildurl"], Commit = p$`_commit`$id)</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co"># order by Package</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> ru_status[<span class="fu">order</span>(ru_status<span class="sc">$</span>Package), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>DEVEL_LINUX_BUILDER <span class="ot">=</span> <span class="st">"nebbiolo2"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper functions</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>get_result <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  entry <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pkg <span class="sc">==</span> x[<span class="st">"pkg"</span>], stage <span class="sc">==</span> x[<span class="st">"stage"</span>], node <span class="sc">==</span> x[<span class="st">"node"</span>])</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(entry)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    entry <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    entry<span class="sc">$</span>result</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>get_install <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"install"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>get_build <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"buildsrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>get_check <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"checksrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>bbs_status <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(node <span class="sc">==</span> DEVEL_LINUX_BUILDER) <span class="sc">|&gt;</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version, Deprecated) <span class="sc">|&gt;</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(pkg, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">install =</span> <span class="fu">map_chr</span>(pkg, get_install)) <span class="sc">|&gt;</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">build =</span> <span class="fu">map_chr</span>(pkg, get_build)) <span class="sc">|&gt;</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">check =</span> <span class="fu">map_chr</span>(pkg, get_check))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-the-status-of-packages-in-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-status-of-packages-in-ru-and-bbs">Compare the status of packages in RU and BBS</h3>
<p>We can only compare <code>check</code> and <code>build</code>, so we’ll check if those columns agree. Note that in the BBS, <code>check</code> can show “WARNINGS” whereas R Universe can show “WARNING” (no s) as well as “OK”, “NOTE”, “ERROR”, or NA.</p>
<p>Note: - Deprecated packages are removed immedately in R Universe. - Packages may have different versions since R-Universe builds new versions after detecting a “push”.</p>
<p>The following includes packages not deprecated in BBS where build and check don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ru_status) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pkg"</span>, <span class="st">"version"</span>, <span class="st">"status"</span>, <span class="st">"build"</span>, <span class="st">"check"</span>, <span class="st">"comment"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>bbs_ru_comparison <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> bbs_status, <span class="at">y =</span> ru_status, <span class="at">by =</span> <span class="st">"pkg"</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">&amp;</span> Deprecated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(pkg, install, build.x, build.y, check.x, check.y, version.x,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                  version.y, status, comment)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>, <span class="st">"RU_build"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"BBS_version"</span>, <span class="st">"RU_version"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-without-x86_64-linux-builds-for-r-4.6" class="level4">
<h4 class="anchored" data-anchor-id="packages-without-x86_64-linux-builds-for-r-4.6">Packages without x86_64 Linux builds for R 4.6</h4>
<p>The following are packages in bioc R Universe but without an x86_64 Linux build for R 4.6.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Comment <span class="sc">==</span> <span class="st">"Missing R4.6 on Linux"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages in RU without an x86_64 linux build for R 4.6"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages in RU without an x86_64 linux build for R 4.6</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CONFESS</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">ddPCRclust</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.30.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">enhancerHomologSearch</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.16.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">GEOmetadb</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.72.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HilbertVisGUI</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.68.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">motifbreakR</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.24.0</td>
<td style="text-align: left;">2.23.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MungeSumstats</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.18.0</td>
<td style="text-align: left;">1.19.1</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">RadioGx</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.14.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SamSPECTRAL</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">WARNINGS</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">1.64.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
<tr class="even">
<td style="text-align: left;">SwathXtend</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">OK</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">2.32.0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Missing R4.6 on Linux</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Notes: - Retrying to refresh the build on the dashboard failed for some packages. How can we rebuild a package that hasn’t received a push in 3 months? - paxtoolsr should be in Bioc R Universe, but it isn’t. Why?</p>
</section>
</section>
</section>
<section id="packages-not-in-agreement" class="level2">
<h2 class="anchored" data-anchor-id="packages-not-in-agreement">Packages not in agreement</h2>
<p>Looking at packages with the same version and removing deprecated packages, we have the following packages where the BBS and R Universe’s results don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Deprecated <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> version.x <span class="sc">==</span> version.y <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(status)) <span class="sc">|&gt;</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version.x, install, build.x, build.y, check.x, check.y,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         status, comment)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_version"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"RU_build"</span>, <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="sc">|&gt;</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages with different build results between BBS and RU"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages with different build results between BBS and RU</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Note: <code>ERROR*</code> could be an <code>ERROR</code>, <code>TIMEOUT</code>, or <code>skipped</code>.</p>
<section id="total-packages-not-in-agreement" class="level3">
<h3 class="anchored" data-anchor-id="total-packages-not-in-agreement">Total packages not in agreement</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bbs_ru_not_ok2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="other" class="level1">
<h1>Other</h1>
<p>If a package repo isn’t accessible or an error exists in the DESCRIPTION file, there will be an error in the registry for that submodule and it will show up at <a href="https://github.com/r-universe/bioc/actions/workflows/sync.yml" class="uri">https://github.com/r-universe/bioc/actions/workflows/sync.yml</a>. The red triangle on <a href="https://bioc.r-universe.dev/" class="uri">https://bioc.r-universe.dev/</a> will take you to the specific workflow error.</p>
<p>Currently, this information is available on the BBS but I wonder if this information is available somewhere for R Universe OR if it would be necessary given that developers would be able to inspect their builds sooner.</p>
<section id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api" class="level3">
<h3 class="anchored" data-anchor-id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api">Important: Use the dbdump endpoint with read_bson to dump the API</h3>
<p>Use <code>read_bson</code> to dump API data into a db</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("mongolite")</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mongolite)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>bioc <span class="ot">&lt;-</span> mongolite<span class="sc">::</span><span class="fu">read_bson</span>(<span class="st">'https://bioc.r-universe.dev/api/dbdump'</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: it returns 2268 records rather than 2276 (fewer retrieved through</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co"># another endpoint)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-a-tarball-from-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="comparing-a-tarball-from-ru-and-bbs">Comparing a tarball from RU and BBS</h3>
<p>Lastly, when I look at the builds, they’re using various repositories, such p3m, r-universe, etc. This might be okay as long as the resulting binaries/tarballs are the generally the same. Looking at XVector for both, both the BBS and R Universe inject some additional information. R Universe also creates an <code>extra</code> directory that has citations, manuals, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2 ruXVector_0.47.2</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2/XVector/DESCRIPTION ruXVector_0.47.2/XVector/DESCRIPTION</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">5c5</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>   <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span>         <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="ex">28,33c28,31</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_url: <span class="ex">https://git.bioconductor.org/packages/XVector</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_branch: <span class="ex">devel</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit: <span class="ex">9f44218</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit_date: <span class="ex">2025-01-07</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Repository: <span class="ex">Bioconductor</span> 3.21</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Date/Publication: <span class="ex">2025-01-08</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Repository: <span class="ex">https://bioc.r-universe.dev</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteUrl: <span class="ex">https://github.com/bioc/XVector</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteRef: <span class="ex">HEAD</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteSha: <span class="ex">9f44218308f9d5867f0b9df5fae60d217f4e7d65</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="ex">35c33</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Packaged: <span class="ex">2025-01-09</span> 00:57:29 UTC<span class="kw">;</span> <span class="ex">biocbuild</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Packaged: <span class="ex">2025-01-08</span> 03:05:41 UTC<span class="kw">;</span> <span class="ex">root</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: extra</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: manual.pdf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="structure-of-a-package-through-universe" class="level3">
<h3 class="anchored" data-anchor-id="structure-of-a-package-through-universe">Structure of a package through universe</h3>
<p>The structure of a package looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">str</span>(ru_pkgs_i[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ :List of 70
  ..$ Package            : chr "ChIPpeakAnno"
  ..$ Type               : chr "Package"
  ..$ Title              : chr "Batch annotation of the peaks identified from either ChIP-seq,\nChIP-chip experiments, or any experiments that "| __truncated__
  ..$ Version            : chr "3.45.2"
  ..$ Encoding           : chr "UTF-8"
  ..$ Author             : chr "Lihua Julie Zhu, Jianhong Ou, Jun Yu, Kai Hu, Haibo Liu, Junhui\nLi, Hervé Pagès, Claude Gazin, Nathan Lawson, "| __truncated__
  ..$ Maintainer         : chr "Jianhong Ou &lt;jou@morgridge.org&gt;, Lihua Julie Zhu\n&lt;julie.zhu@umassmed.edu&gt;, Kai Hu &lt;kai.hu@umassmed.edu&gt;, Junhu"| __truncated__
  ..$ Description        : chr "The package encompasses a range of functions for\nidentifying the closest gene, exon, miRNA, or custom\nfeature"| __truncated__
  ..$ License            : chr "GPL (&gt;= 2)"
  ..$ LazyLoad           : chr "yes"
  ..$ LazyData           : chr "true"
  ..$ LazyDataCompression: chr "xz"
  ..$ biocViews          : chr "Annotation, ChIPSeq, ChIPchip"
  ..$ VignetteBuilder    : chr "knitr"
  ..$ RoxygenNote        : chr "7.3.3"
  ..$ Config/pak/sysreqs : chr "make libbz2-dev libicu-dev liblzma-dev libpng-dev\nlibxml2-dev libssl-dev xz-utils zlib1g-dev"
  ..$ Repository         : chr "https://bioc.r-universe.dev"
  ..$ Date/Publication   : chr "2025-11-12 01:01:42 UTC"
  ..$ RemoteUrl          : chr "https://github.com/bioc/ChIPpeakAnno"
  ..$ RemoteRef          : chr "HEAD"
  ..$ RemoteSha          : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  ..$ NeedsCompilation   : chr "no"
  ..$ Packaged           :List of 2
  .. ..$ Date: chr "2025-11-12 04:00:39 UTC"
  .. ..$ User: chr "root"
  ..$ MD5sum             : chr "b0ccc762cec1e26f541e21f82a456fee"
  ..$ _user              : chr "bioc"
  ..$ _type              : chr "src"
  ..$ _file              : chr "ChIPpeakAnno_3.45.2.tar.gz"
  ..$ _fileid            : chr "26e140df240710c49a8335fb0ec5e56c2f9704e57b46b7fa65a0e0de9ae52c84"
  ..$ _filesize          : int 22692720
  ..$ _sha256            : chr "26e140df240710c49a8335fb0ec5e56c2f9704e57b46b7fa65a0e0de9ae52c84"
  ..$ _created           : chr "2025-11-12T04:00:39.000Z"
  ..$ _published         : chr "2025-11-12T07:19:05.699Z"
  ..$ _distro            : chr "noble"
  ..$ _jobs              :List of 9
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1123
  .. .. ..$ config  : chr "linux-devel-x86_64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538939017"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1098
  .. .. ..$ config  : chr "linux-release-x86_64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538936596"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 762
  .. .. ..$ config  : chr "macos-oldrel-arm64"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4539937099"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1058
  .. .. ..$ config  : chr "macos-release-arm64"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4539977802"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1446
  .. .. ..$ config  : chr "source"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538823092"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 671
  .. .. ..$ config  : chr "wasm-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538896065"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1077
  .. .. ..$ config  : chr "windows-devel"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538934594"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1013
  .. .. ..$ config  : chr "windows-oldrel"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538927156"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.51e+10
  .. .. ..$ time    : int 1008
  .. .. ..$ config  : chr "windows-release"
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4538928077"
  ..$ _buildurl          : chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  ..$ _status            : chr "success"
  ..$ _host              : chr "GitHub-Actions"
  ..$ _upstream          : chr "https://github.com/bioc/ChIPpeakAnno"
  ..$ _commit            :List of 5
  .. ..$ id       : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. ..$ author   : chr "Jianhong Ou &lt;jianhong.ou@gmail.com&gt;"
  .. ..$ committer: chr "Jianhong Ou &lt;jianhong.ou@gmail.com&gt;"
  .. ..$ message  : chr "fix a typo\n"
  .. ..$ time     : int 1762909302
  ..$ _maintainer        :List of 2
  .. ..$ name : chr "Jianhong Ou"
  .. ..$ email: chr "junhui.li@umassmed.edu"
  ..$ _registered        : logi TRUE
  ..$ _dependencies      :List of 72
  .. ..$ :List of 3
  .. .. ..$ package: chr "R"
  .. .. ..$ version: chr "&gt;= 3.5"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "methods"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "IRanges"
  .. .. ..$ version: chr "&gt;= 2.13.12"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "GenomicRanges"
  .. .. ..$ version: chr "&gt;= 1.31.8"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 3
  .. .. ..$ package: chr "S4Vectors"
  .. .. ..$ version: chr "&gt;= 0.17.25"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "AnnotationDbi"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "BiocGenerics"
  .. .. ..$ version: chr "&gt;= 0.1.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "Biostrings"
  .. .. ..$ version: chr "&gt;= 2.47.6"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "pwalign"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DBI"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "dplyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GenomeInfoDb"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GenomicAlignments"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GenomicFeatures"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "RBGL"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Rsamtools"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SummarizedExperiment"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "VennDiagram"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "biomaRt"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ggplot2"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "grDevices"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "graph"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "graphics"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "grid"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "InteractionSet"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "KEGGREST"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "matrixStats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "multtest"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "regioneR"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rtracklayer"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "universalmotif"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stringr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tibble"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tidyr"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "data.table"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "scales"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ensembldb"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "AnnotationHub"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "limma"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "reactome.db"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocManager"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocStyle"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Ecoli.NCBI.20080805"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Hsapiens.UCSC.hg19"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "org.Ce.eg.db"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "org.Hs.eg.db"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Celegans.UCSC.ce10"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Drerio.UCSC.danRer7"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Hsapiens.UCSC.hg38"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "DelayedArray"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "idr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "seqinr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "EnsDb.Hsapiens.v75"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "EnsDb.Hsapiens.v79"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "EnsDb.Hsapiens.v86"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TxDb.Hsapiens.UCSC.hg18.knownGene"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TxDb.Hsapiens.UCSC.hg19.knownGene"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TxDb.Hsapiens.UCSC.hg38.knownGene"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GO.db"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "gplots"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "UpSetR"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "knitr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rmarkdown"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "reshape2"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "testthat"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "trackViewer"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "motifStack"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "OrganismDbi"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocFileCache"
  .. .. ..$ role   : chr "Suggests"
  ..$ _owner             : chr "bioc"
  ..$ _selfowned         : logi TRUE
  ..$ _usedby            : int 6
  ..$ _updates           :List of 6
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-01"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-16"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-28"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-42"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-44"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-46"
  .. .. ..$ n   : int 3
  ..$ _tags              : list()
  ..$ _bioc              :List of 2
  .. ..$ :List of 3
  .. .. ..$ branch : chr "devel"
  .. .. ..$ version: chr "3.45.0"
  .. .. ..$ bioc   : chr "3.23"
  .. ..$ :List of 3
  .. .. ..$ branch : chr "release"
  .. .. ..$ version: chr "3.44.0"
  .. .. ..$ bioc   : chr "3.22"
  ..$ _topics            :List of 3
  .. ..$ : chr "annotation"
  .. ..$ : chr "chipseq"
  .. ..$ : chr "chipchip"
  ..$ _userbio           :List of 4
  .. ..$ uuid       : int 2286807
  .. ..$ type       : chr "organization"
  .. ..$ name       : chr "Bioconductor"
  .. ..$ description: chr "Software for the analysis and comprehension of high-throughput genomic data"
  ..$ _downloads         :List of 2
  .. ..$ count : int 2004
  .. ..$ source: chr "https://www.bioconductor.org/packages/stats/bioc/ChIPpeakAnno"
  ..$ _mentions          : int 145
  ..$ _searchresults     : int 768
  ..$ _rbuild            : chr "4.5.2"
  ..$ _assets            :List of 11
  .. ..$ : chr "extra/ChIPpeakAnno.html"
  .. ..$ : chr "extra/citation.cff"
  .. ..$ : chr "extra/citation.html"
  .. ..$ : chr "extra/citation.json"
  .. ..$ : chr "extra/citation.txt"
  .. ..$ : chr "extra/contents.json"
  .. ..$ : chr "extra/NEWS.html"
  .. ..$ : chr "extra/NEWS.txt"
  .. ..$ : chr "extra/readme.html"
  .. ..$ : chr "extra/readme.md"
  .. ..$ : chr "manual.pdf"
  ..$ _homeurl           : chr "https://github.com/jianhong/chippeakanno"
  ..$ _realowner         : chr "bioc"
  ..$ _cranurl           : logi FALSE
  ..$ _exports           :List of 59
  .. ..$ : chr "addAncestors"
  .. ..$ : chr "addGeneIDs"
  .. ..$ : chr "addMetadata"
  .. ..$ : chr "annoGR"
  .. ..$ : chr "annoPeaks"
  .. ..$ : chr "annotatePeakInBatch"
  .. ..$ : chr "assignChromosomeRegion"
  .. ..$ : chr "binOverFeature"
  .. ..$ : chr "binOverGene"
  .. ..$ : chr "binOverRegions"
  .. ..$ : chr "coerce"
  .. ..$ : chr "condenseMatrixByColnames"
  .. ..$ : chr "convert2EntrezID"
  .. ..$ : chr "countPatternInSeqs"
  .. ..$ : chr "cumulativePercentage"
  .. ..$ : chr "downstreams"
  .. ..$ : chr "egOrgMap"
  .. ..$ : chr "enrichmentPlot"
  .. ..$ : chr "estFragmentLength"
  .. ..$ : chr "estLibSize"
  .. ..$ : chr "featureAlignedDistribution"
  .. ..$ : chr "featureAlignedExtendSignal"
  .. ..$ : chr "featureAlignedHeatmap"
  .. ..$ : chr "featureAlignedSignal"
  .. ..$ : chr "findEnhancers"
  .. ..$ : chr "findMotifsInPromoterSeqs"
  .. ..$ : chr "findOverlappingPeaks"
  .. ..$ : chr "findOverlapsOfPeaks"
  .. ..$ : chr "genomicElementDistribution"
  .. ..$ : chr "genomicElementUpSetR"
  .. ..$ : chr "getAllPeakSequence"
  .. ..$ : chr "getAnnotation"
  .. ..$ : chr "getEnrichedGO"
  .. ..$ : chr "getEnrichedPATH"
  .. ..$ : chr "getGO"
  .. ..$ : chr "getUniqueGOidCount"
  .. ..$ : chr "getVennCounts"
  .. ..$ : chr "hyperGtest"
  .. ..$ : chr "IDRfilter"
  .. ..$ : chr "info"
  .. ..$ : chr "makeVennDiagram"
  .. ..$ : chr "mergePlusMinusPeaks"
  .. ..$ : chr "metagenePlot"
  .. ..$ : chr "oligoFrequency"
  .. ..$ : chr "oligoSummary"
  .. ..$ : chr "peakPermTest"
  .. ..$ : chr "peaksNearBDP"
  .. ..$ : chr "pie1"
  .. ..$ : chr "plotBinOverRegions"
  .. ..$ : chr "preparePool"
  .. ..$ : chr "reCenterPeaks"
  .. ..$ : chr "summarizeOverlapsByBins"
  .. ..$ : chr "summarizePatternInPeaks"
  .. ..$ : chr "tileCount"
  .. ..$ : chr "tileGRanges"
  .. ..$ : chr "toGRanges"
  .. ..$ : chr "translatePattern"
  .. ..$ : chr "write2FASTA"
  .. ..$ : chr "xget"
  ..$ _datasets          :List of 21
  .. ..$ :List of 7
  .. .. ..$ name  : chr "ExonPlusUtr.human.GRCh37"
  .. .. ..$ title : chr "Gene model with exon, 5' UTR and 3' UTR information for human sapiens (GRCh37) obtained from biomaRt"
  .. .. ..$ object: chr "ExonPlusUtr.human.GRCh37"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "HOT.spots"
  .. .. ..$ title : chr "High Occupancy of Transcription Related Factors regions"
  .. .. ..$ object: chr "HOT.spots"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "CompressedGRangesList"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "Peaks.Ste12.Replicate1"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 1 in yeast (see reference)"
  .. .. ..$ object: chr "Peaks.Ste12.Replicate1"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "Peaks.Ste12.Replicate2"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 2 in yeast (see reference)"
  .. .. ..$ object: chr "Peaks.Ste12.Replicate2"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "Peaks.Ste12.Replicate3"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 3 in yeast (see reference)"
  .. .. ..$ object: chr "Peaks.Ste12.Replicate3"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.human.GRCh37"
  .. .. ..$ title : chr "TSS annotation for human sapiens (GRCh37) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.human.GRCh37"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.human.GRCh38"
  .. .. ..$ title : chr "TSS annotation for human sapiens (GRCh38) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.human.GRCh38"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.human.NCBI36"
  .. .. ..$ title : chr "TSS annotation for human sapiens (NCBI36) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.human.NCBI36"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.mouse.GRCm38"
  .. .. ..$ title : chr "TSS annotation data for Mus musculus (GRCm38.p1) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.mouse.GRCm38"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.mouse.NCBIM37"
  .. .. ..$ title : chr "TSS annotation data for mouse (NCBIM37) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.mouse.NCBIM37"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.rat.RGSC3.4"
  .. .. ..$ title : chr "TSS annotation data for rat (RGSC3.4) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.rat.RGSC3.4"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.rat.Rnor_5.0"
  .. .. ..$ title : chr "TSS annotation data for Rattus norvegicus (Rnor_5.0) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.rat.Rnor_5.0"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.zebrafish.Zv8"
  .. .. ..$ title : chr "TSS annotation data for zebrafish (Zv8) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.zebrafish.Zv8"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "TSS.zebrafish.Zv9"
  .. .. ..$ title : chr "TSS annotation for Danio rerio (Zv9) obtained from biomaRt"
  .. .. ..$ object: chr "TSS.zebrafish.Zv9"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "annotatedPeak"
  .. .. ..$ title : chr "Annotated Peaks"
  .. .. ..$ object: chr "annotatedPeak"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "enrichedGO"
  .. .. ..$ title : chr "Enriched Gene Ontology terms used as example"
  .. .. ..$ object: chr "enrichedGO"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "list"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi TRUE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "myPeakList"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ object: chr "myPeakList"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "peaks1"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ object: chr "peaks1"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "peaks2"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ object: chr "peaks2"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "peaks3"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ object: chr "peaks3"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  .. ..$ :List of 7
  .. .. ..$ name  : chr "wgEncodeTfbsV3"
  .. .. ..$ title : chr "transcription factor binding site clusters (V3) from ENCODE"
  .. .. ..$ object: chr "wgEncodeTfbsV3"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "GRanges"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi FALSE
  ..$ _help              :List of 84
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ChIPpeakAnno-package"
  .. .. ..$ title : chr "Batch annotation of the peaks identified from either ChIP-seq or ChIP-chip experiments."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "ChIPpeakAnno-package"
  .. .. .. ..$ : chr "ChIPpeakAnno"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "addAncestors"
  .. .. ..$ title : chr "Add GO IDs of the ancestors for a given vector of GO ids"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "addAncestors"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "addGeneIDs"
  .. .. ..$ title : chr "Add common IDs to annotated peaks such as gene symbol, entrez ID, ensemble gene id and refseq id."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "addGeneIDs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "addMetadata"
  .. .. ..$ title : chr "Add metadata of the GRanges objects used for findOverlapsOfPeaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "addMetadata"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "annoGR"
  .. .. ..$ title : chr "Class 'annoGR'"
  .. .. ..$ topics:List of 10
  .. .. .. ..$ : chr "annoGR"
  .. .. .. ..$ : chr "annoGR,EnsDb-method"
  .. .. .. ..$ : chr "annoGR,GRanges-method"
  .. .. .. ..$ : chr "annoGR,TxDb-method"
  .. .. .. ..$ : chr "annoGR-class"
  .. .. .. ..$ : chr "coerce"
  .. .. .. ..$ : chr "coerce,annoGR,GRanges-method"
  .. .. .. ..$ : chr "coerce,GRanges,annoGR-method"
  .. .. .. ..$ : chr "info"
  .. .. .. ..$ : chr "info,annoGR-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "annoPeaks"
  .. .. ..$ title : chr "Annotate peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "annoPeaks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "annotatedPeak"
  .. .. ..$ title : chr "Annotated Peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "annotatedPeak"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "annotatePeakInBatch"
  .. .. ..$ title : chr "Obtain the distance to the nearest TSS, miRNA, and/or exon for a list of peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "annotatePeakInBatch"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "assignChromosomeRegion"
  .. .. ..$ title : chr "Summarize peak distribution over exon, intron, enhancer, proximal promoter, 5 prime UTR and 3 prime UTR"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "assignChromosomeRegion"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "bdp"
  .. .. ..$ title : chr "obtain the peaks near bi-directional promoters"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "bdp"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "bindist"
  .. .. ..$ title : chr "Class '\"bindist\"'"
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "$,bindist-method"
  .. .. .. ..$ : chr "$&lt;-,bindist-method"
  .. .. .. ..$ : chr "bindist"
  .. .. .. ..$ : chr "bindist-class"
  .. .. .. ..$ : chr "bindist-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "binOverFeature"
  .. .. ..$ title : chr "Aggregate peaks over bins from the TSS"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "binOverFeature"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "binOverGene"
  .. .. ..$ title : chr "coverage of gene body"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "binOverGene"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "binOverRegions"
  .. .. ..$ title : chr "coverage of chromosome regions"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "binOverRegions"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ChIPpeakAnno-deprecated"
  .. .. ..$ title : chr "Deprecated Functions in Package ChIPpeakAnno"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "ChIPpeakAnno-deprecated"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "cntOverlaps"
  .. .. ..$ title : chr "count overlaps"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "cntOverlaps"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "condenseMatrixByColnames"
  .. .. ..$ title : chr "Condense matrix by colnames"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "condenseMatrixByColnames"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "convert2EntrezID"
  .. .. ..$ title : chr "Convert other common IDs to entrez gene ID."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "convert2EntrezID"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "countPatternInSeqs"
  .. .. ..$ title : chr "Output total number of patterns found in the input sequences"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "countPatternInSeqs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "cumulativePercentage"
  .. .. ..$ title : chr "Plot the cumulative percentage tag allocation in sample"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "cumulativePercentage"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "downstreams"
  .. .. ..$ title : chr "Get downstream coordinates"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "downstreams"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "egOrgMap"
  .. .. ..$ title : chr "Convert between the name of the organism annotation package (\"OrgDb\") and the name of the organism."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "egOrgMap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "enrichedGO"
  .. .. ..$ title : chr "Enriched Gene Ontology terms used as example"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "enrichedGO"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "enrichmentPlot"
  .. .. ..$ title : chr "plot enrichment results"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "enrichmentPlot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "EnsDb2GR"
  .. .. ..$ title : chr "EnsDb object to GRanges"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "EnsDb2GR"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "estFragmentLength"
  .. .. ..$ title : chr "estimate the fragment length"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "estFragmentLength"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "estLibSize"
  .. .. ..$ title : chr "estimate the library size"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "estLibSize"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "ExonPlusUtr.human.GRCh37"
  .. .. ..$ title : chr "Gene model with exon, 5' UTR and 3' UTR information for human sapiens (GRCh37) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "ExonPlusUtr.human.GRCh37"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "featureAlignedDistribution"
  .. .. ..$ title : chr "plot distribution in given ranges"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "featureAlignedDistribution"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "featureAlignedExtendSignal"
  .. .. ..$ title : chr "extract signals in given ranges from bam files"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "featureAlignedExtendSignal"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "featureAlignedHeatmap"
  .. .. ..$ title : chr "Heatmap representing signals in given ranges"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "featureAlignedHeatmap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "featureAlignedSignal"
  .. .. ..$ title : chr "extract signals in given ranges"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "featureAlignedSignal"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "findEnhancers"
  .. .. ..$ title : chr "Find possible enhancers depend on DNA interaction data"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "findEnhancers"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "findMotifsInPromoterSeqs"
  .. .. ..$ title : chr "Find occurence of input motifs in the promoter regions of the input gene list"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "findMotifsInPromoterSeqs"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "findOverlappingPeaks"
  .. .. ..$ title : chr "Find the overlapping peaks for two peak ranges."
  .. .. ..$ topics:List of 2
  .. .. .. ..$ : chr "findOverlappingPeaks"
  .. .. .. ..$ : chr "findOverlappingPeaks-deprecated"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "findOverlapsOfPeaks"
  .. .. ..$ title : chr "Find the overlapped peaks among two or more set of peaks."
  .. .. ..$ topics:List of 3
  .. .. .. ..$ : chr "findOverlapsOfPeaks"
  .. .. .. ..$ : chr "overlappingPeaks"
  .. .. .. ..$ : chr "overlappingPeaks-class"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "genomicElementDistribution"
  .. .. ..$ title : chr "Genomic Element distribution"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "genomicElementDistribution"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "genomicElementUpSetR"
  .. .. ..$ title : chr "Genomic Element data for upset plot"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "genomicElementUpSetR"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getAllPeakSequence"
  .. .. ..$ title : chr "Obtain genomic sequences around the peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getAllPeakSequence"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getAnnotation"
  .. .. ..$ title : chr "Obtain the TSS, exon or miRNA annotation for the specified species"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getAnnotation"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getEnrichedGO"
  .. .. ..$ title : chr "Obtain enriched gene ontology (GO) terms that near the peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getEnrichedGO"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getEnrichedPATH"
  .. .. ..$ title : chr "Obtain enriched PATH that near the peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getEnrichedPATH"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getGO"
  .. .. ..$ title : chr "Obtain gene ontology (GO) terms for given genes"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getGO"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "getVennCounts"
  .. .. ..$ title : chr "Obtain Venn Counts for Venn Diagram, internal function for makeVennDigram"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "getVennCounts"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "HOT.spots"
  .. .. ..$ title : chr "High Occupancy of Transcription Related Factors regions"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "HOT.spots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "IDRfilter"
  .. .. ..$ title : chr "Filter peaks by IDR (irreproducible discovery rate)"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "IDRfilter"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "makeVennDiagram"
  .. .. ..$ title : chr "Make Venn Diagram from a list of peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "makeVennDiagram"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "mergePlusMinusPeaks"
  .. .. ..$ title : chr "Merge peaks from plus strand and minus strand"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "mergePlusMinusPeaks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "metagenePlot"
  .. .. ..$ title : chr "peak distance to features"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "metagenePlot"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "myPeakList"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "myPeakList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "oligoFrequency"
  .. .. ..$ title : chr "get the oligonucleotide frequency"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "oligoFrequency"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "oligoSummary"
  .. .. ..$ title : chr "Output a summary of consensus in the peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "oligoSummary"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "peakPermTest"
  .. .. ..$ title : chr "Permutation Test for two given peak lists"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "peakPermTest"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "Peaks.Ste12.Replicate1"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 1 in yeast (see reference)"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "Peaks.Ste12.Replicate1"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "Peaks.Ste12.Replicate2"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 2 in yeast (see reference)"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "Peaks.Ste12.Replicate2"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "Peaks.Ste12.Replicate3"
  .. .. ..$ title : chr "Ste12-binding sites from biological replicate 3 in yeast (see reference)"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "Peaks.Ste12.Replicate3"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "peaks1"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "peaks1"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "peaks2"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "peaks2"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "peaks3"
  .. .. ..$ title : chr "An example GRanges object representing a ChIP-seq peak dataset"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "peaks3"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "peaksNearBDP"
  .. .. ..$ title : chr "obtain the peaks near bi-directional promoters"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "peaksNearBDP"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "permPool"
  .. .. ..$ title : chr "Class '\"permPool\"'"
  .. .. ..$ topics:List of 5
  .. .. .. ..$ : chr "$,permPool-method"
  .. .. .. ..$ : chr "$&lt;-,permPool-method"
  .. .. .. ..$ : chr "permPool"
  .. .. .. ..$ : chr "permPool-class"
  .. .. .. ..$ : chr "permPool-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "pie1"
  .. .. ..$ title : chr "Pie Charts"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "pie1"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "plotBinOverRegions"
  .. .. ..$ title : chr "plot the coverage of regions"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "plotBinOverRegions"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "preparePool"
  .. .. ..$ title : chr "prepare data for permutation test"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "preparePool"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "reCenterPeaks"
  .. .. ..$ title : chr "re-center the peaks"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "reCenterPeaks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "summarizeOverlapsByBins"
  .. .. ..$ title : chr "Perform overlap queries between reads and genomic features by bins"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "summarizeOverlapsByBins"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "summarizePatternInPeaks"
  .. .. ..$ title : chr "Output a summary of the occurrence and enrichment of each pattern in the sequences."
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "summarizePatternInPeaks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "tileCount"
  .. .. ..$ title : chr "Perform overlap queries between reads and genome by windows"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "tileCount"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "tileGRanges"
  .. .. ..$ title : chr "Slide windows on a given GRanges object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "tileGRanges"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "toGRanges"
  .. .. ..$ title : chr "Convert dataset to GRanges"
  .. .. ..$ topics:List of 6
  .. .. .. ..$ : chr "toGRanges"
  .. .. .. ..$ : chr "toGRanges,character-method"
  .. .. .. ..$ : chr "toGRanges,connection-method"
  .. .. .. ..$ : chr "toGRanges,data.frame-method"
  .. .. .. ..$ : chr "toGRanges,EnsDb-method"
  .. .. .. ..$ : chr "toGRanges,TxDb-method"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "translatePattern"
  .. .. ..$ title : chr "translate pattern from IUPAC Extended Genetic Alphabet to regular expression"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "translatePattern"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.human.GRCh37"
  .. .. ..$ title : chr "TSS annotation for human sapiens (GRCh37) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.human.GRCh37"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.human.GRCh38"
  .. .. ..$ title : chr "TSS annotation for human sapiens (GRCh38) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.human.GRCh38"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.human.NCBI36"
  .. .. ..$ title : chr "TSS annotation for human sapiens (NCBI36) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.human.NCBI36"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.mouse.GRCm38"
  .. .. ..$ title : chr "TSS annotation data for Mus musculus (GRCm38.p1) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.mouse.GRCm38"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.mouse.NCBIM37"
  .. .. ..$ title : chr "TSS annotation data for mouse (NCBIM37) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.mouse.NCBIM37"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.rat.RGSC3.4"
  .. .. ..$ title : chr "TSS annotation data for rat (RGSC3.4) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.rat.RGSC3.4"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.rat.Rnor_5.0"
  .. .. ..$ title : chr "TSS annotation data for Rattus norvegicus (Rnor_5.0) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.rat.Rnor_5.0"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.zebrafish.Zv8"
  .. .. ..$ title : chr "TSS annotation data for zebrafish (Zv8) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.zebrafish.Zv8"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TSS.zebrafish.Zv9"
  .. .. ..$ title : chr "TSS annotation for Danio rerio (Zv9) obtained from biomaRt"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TSS.zebrafish.Zv9"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TxDb2GR"
  .. .. ..$ title : chr "TxDb object to GRanges"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TxDb2GR"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "wgEncodeTfbsV3"
  .. .. ..$ title : chr "transcription factor binding site clusters (V3) from ENCODE"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "wgEncodeTfbsV3"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "write2FASTA"
  .. .. ..$ title : chr "Write sequences to a file in fasta format"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "write2FASTA"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "xget"
  .. .. ..$ title : chr "Return the value from a Bimap objects"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "xget"
  ..$ _readme            : chr "https://github.com/bioc/ChIPpeakAnno/raw/HEAD/README.md"
  ..$ _rundeps           :List of 117
  .. ..$ : chr "abind"
  .. ..$ : chr "AnnotationDbi"
  .. ..$ : chr "AnnotationFilter"
  .. ..$ : chr "askpass"
  .. ..$ : chr "BH"
  .. ..$ : chr "Biobase"
  .. ..$ : chr "BiocFileCache"
  .. ..$ : chr "BiocGenerics"
  .. ..$ : chr "BiocIO"
  .. ..$ : chr "BiocParallel"
  .. ..$ : chr "biomaRt"
  .. ..$ : chr "Biostrings"
  .. ..$ : chr "bit"
  .. ..$ : chr "bit64"
  .. ..$ : chr "bitops"
  .. ..$ : chr "blob"
  .. ..$ : chr "BSgenome"
  .. ..$ : chr "cachem"
  .. ..$ : chr "cigarillo"
  .. ..$ : chr "cli"
  .. ..$ : chr "codetools"
  .. ..$ : chr "cpp11"
  .. ..$ : chr "crayon"
  .. ..$ : chr "curl"
  .. ..$ : chr "data.table"
  .. ..$ : chr "DBI"
  .. ..$ : chr "dbplyr"
  .. ..$ : chr "DelayedArray"
  .. ..$ : chr "dplyr"
  .. ..$ : chr "ensembldb"
  .. ..$ : chr "farver"
  .. ..$ : chr "fastmap"
  .. ..$ : chr "filelock"
  .. ..$ : chr "formatR"
  .. ..$ : chr "futile.logger"
  .. ..$ : chr "futile.options"
  .. ..$ : chr "generics"
  .. ..$ : chr "GenomeInfoDb"
  .. ..$ : chr "GenomicAlignments"
  .. ..$ : chr "GenomicFeatures"
  .. ..$ : chr "GenomicRanges"
  .. ..$ : chr "ggplot2"
  .. ..$ : chr "glue"
  .. ..$ : chr "graph"
  .. ..$ : chr "gtable"
  .. ..$ : chr "hms"
  .. ..$ : chr "httr"
  .. ..$ : chr "httr2"
  .. ..$ : chr "InteractionSet"
  .. ..$ : chr "IRanges"
  .. ..$ : chr "isoband"
  .. ..$ : chr "jsonlite"
  .. ..$ : chr "KEGGREST"
  .. ..$ : chr "labeling"
  .. ..$ : chr "lambda.r"
  .. ..$ : chr "lattice"
  .. ..$ : chr "lazyeval"
  .. ..$ : chr "lifecycle"
  .. ..$ : chr "magrittr"
  .. ..$ : chr "MASS"
  .. ..$ : chr "Matrix"
  .. ..$ : chr "MatrixGenerics"
  .. ..$ : chr "matrixStats"
  .. ..$ : chr "memoise"
  .. ..$ : chr "mime"
  .. ..$ : chr "multtest"
  .. ..$ : chr "openssl"
  .. ..$ : chr "pillar"
  .. ..$ : chr "pkgconfig"
  .. ..$ : chr "plogr"
  .. ..$ : chr "png"
  .. ..$ : chr "prettyunits"
  .. ..$ : chr "progress"
  .. ..$ : chr "ProtGenerics"
  .. ..$ : chr "purrr"
  .. ..$ : chr "pwalign"
  .. ..$ : chr "R6"
  .. ..$ : chr "rappdirs"
  .. ..$ : chr "RBGL"
  .. ..$ : chr "RColorBrewer"
  .. ..$ : chr "Rcpp"
  .. ..$ : chr "RcppThread"
  .. ..$ : chr "RCurl"
  .. ..$ : chr "regioneR"
  .. ..$ : chr "restfulr"
  .. ..$ : chr "Rhtslib"
  .. ..$ : chr "rjson"
  .. ..$ : chr "rlang"
  .. ..$ : chr "Rsamtools"
  .. ..$ : chr "RSQLite"
  .. ..$ : chr "rtracklayer"
  .. ..$ : chr "S4Arrays"
  .. ..$ : chr "S4Vectors"
  .. ..$ : chr "S7"
  .. ..$ : chr "scales"
  .. ..$ : chr "Seqinfo"
  .. ..$ : chr "snow"
  .. ..$ : chr "SparseArray"
  .. ..$ : chr "stringi"
  .. .. [list output truncated]
  ..$ _vignettes         :List of 1
  .. ..$ :List of 9
  .. .. ..$ source  : chr "ChIPpeakAnno.Rmd"
  .. .. ..$ filename: chr "ChIPpeakAnno.html"
  .. .. ..$ title   : chr "ChIPpeakAnno: annotate, visualize, and compare peak data"
  .. .. ..$ author  : chr "Jianhong Ou, Kai Hu, Haibo Liu, Junhui Li, Jun Yu, Lihua Julie Zhu"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 84
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Quick demo"
  .. .. .. ..$ : chr "Convert peaks to GRanges"
  .. .. .. ..$ : chr "Prepare annotations"
  .. .. .. ..$ : chr "Annotate peaks"
  .. .. .. ..$ : chr "Add gene symbols"
  .. .. .. ..$ : chr "Convert peaks to GRanges"
  .. .. .. ..$ : chr "Handle replicates"
  .. .. .. ..$ : chr "Obtain final peak set from replicates"
  .. .. .. ..$ : chr "Assess the concordance of peak replicates"
  .. .. .. ..$ : chr "Find overlapping peaks"
  .. .. .. ..$ : chr "Visualize the overlaps using Venn diagram"
  .. .. .. ..$ : chr "Use other tools to plot Venn diagram"
  .. .. .. ..$ : chr "Visualize the distribution of relative positions for overlapping peaks"
  .. .. .. ..$ : chr "Prepare annotation file"
  .. .. .. ..$ : chr "Commonly used annotations"
  .. .. .. ..$ : chr "Bioconductor supported TxDb and EnsDb"
  .. .. .. ..$ : chr "Obtain EnsDb and TxDb from AnnotationHub"
  .. .. .. ..$ : chr "Build custom EnsDb and TxDb"
  .. .. .. ..$ : chr "Use biomaRt"
  .. .. .. ..$ : chr "Convert annotations into GRanges"
  .. .. .. ..$ : chr "Use toGRanges function"
  .. .. .. ..$ : chr "Use accessor functions"
  .. .. .. ..$ : chr "Use getAnnotation function"
  .. .. .. ..$ : chr "Visualize peak distributions"
  .. .. .. ..$ : chr "Plot peak distributions relative to genomic features"
  .. .. .. ..$ : chr "Plot peak distributions over different feature levels"
  .. .. .. ..$ : chr "Pie graph with one peak set"
  .. .. .. ..$ : chr "Bar graph with a list of peak sets"
  .. .. .. ..$ : chr "Plot peak overlaps for multiple features"
  .. .. .. ..$ : chr "Annotate peaks"
  .. .. .. ..$ : chr "Peak-centric method"
  .. .. .. ..$ : chr "Feature-centric method"
  .. .. .. ..$ : chr "Relative peak-to-feature location"
  .. .. .. ..$ : chr "More customization with bindingRegion"
  .. .. .. ..$ : chr "Example 1: find the nearest features"
  .. .. .. ..$ : chr "Example 2: find the nearest and overlapping features"
  .. .. .. ..$ : chr "Visualize peak-to-feature distances"
  .. .. .. ..$ : chr "Use custom annotation data"
  .. .. .. ..$ : chr "Add other feature IDs"
  .. .. .. ..$ : chr "Example1: find gene symbols for a vector of entrez IDs"
  .. .. .. ..$ : chr "Example2: add gene symbols to annotated peaks"
  .. .. .. ..$ : chr "Enrichment analysis"
  .. .. .. ..$ : chr "Motif analysis"
  .. .. .. ..$ : chr "Obtain sequences surrounding the peaks"
  .. .. .. ..$ : chr "Discover consensus sequences (motifs) in the peaks"
  .. .. .. ..$ : chr "Scan pre-defined sequence patterns in the peaks"
  .. .. .. ..$ : chr "Alternative tools"
  .. .. .. ..$ : chr "Peak profile comparison"
  .. .. .. ..$ : chr "Use hypergeometric test to determine overlap among peak sets"
  .. .. .. ..$ : chr "Use permutation test to determine overlap among peak sets"
  .. .. .. ..$ : chr "Use custom peak pool"
  .. .. .. ..$ : chr "Visualize TSS enrichment signal of multiple experiments"
  .. .. .. ..$ : chr "Heatmap"
  .. .. .. ..$ : chr "Density plot"
  .. .. .. ..$ : chr "Common workflow 1: single TF with replicates"
  .. .. .. ..$ : chr "Step1: import data"
  .. .. .. ..$ : chr "Step2: prepare annotation file"
  .. .. .. ..$ : chr "Step3: visualize peak distribution"
  .. .. .. ..$ : chr "Peak distribution relative to features"
  .. .. .. ..$ : chr "Peak distribution over multiple feature levels"
  .. .. .. ..$ : chr "Peak overlappings for multiple features"
  .. .. .. ..$ : chr "Step4: annotate peaks"
  .. .. .. ..$ : chr "Find peaks located in bi-directional promoters"
  .. .. .. ..$ : chr "Find peaks located in enhancers"
  .. .. .. ..$ : chr "Step5: perform enrichment analysis"
  .. .. .. ..$ : chr "Step6: conduct motif analysis"
  .. .. .. ..$ : chr "Common workflow 2: comparing binding profiles for multiple TFs"
  .. .. .. ..$ : chr "Step2: determine if there is significant overlap among peak sets"
  .. .. .. ..$ : chr "Step3: visualize and compare the binding patterns"
  .. .. .. ..$ : chr "Have questions?"
  .. .. .. ..$ : chr "Selected Q &amp; A"
  .. .. .. ..$ : chr "How to import peaks?"
  .. .. .. ..$ : chr "How to properly annotate peaks from ChIP-seq data?"
  .. .. .. ..$ : chr "Should I use annotatePeakInBatch or annoPeaks?"
  .. .. .. ..$ : chr "How to prepare annotation data for annotatePeakInBatch?"
  .. .. .. ..$ : chr "Can I output either the nearest or overlapping features?"
  .. .. .. ..$ : chr "Why is the sum of peak numbers in the Venn diagram NOT equal to the sum of the peaks in the original peak lists?"
  .. .. .. ..$ : chr "How does findOverlapsOfPeaks count the number of overlapping peaks?"
  .. .. .. ..$ : chr "Is there a way to show the number of peaks in original peak lists?"
  .. .. .. ..$ : chr "How to extract the original peak IDs of the overlapping peaks?"
  .. .. .. ..$ : chr "How to select the proper number for totalTest when using makeVennDiagram?"
  .. .. .. ..$ : chr "How to cite ChIPpeakAnno"
  .. .. .. ..$ : chr "Session info"
  .. .. ..$ created : chr "2015-10-07 21:01:48"
  .. .. ..$ modified: chr "2024-08-15 12:19:40"
  .. .. ..$ commits : int 17
  ..$ _score             : num 8.92
  ..$ _indexed           : logi TRUE
  ..$ _nocasepkg         : chr "chippeakanno"
  ..$ _universes         :List of 1
  .. ..$ : chr "bioc"
  ..$ _previous          : chr "3.45.0"
  ..$ _binaries          :List of 8
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:13:35.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "c4300cc5caeeb84b90e77edc2d1b5c61b96b8ee77f223710ac5962604701efe0"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:13:19.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "961308a27bd8b73f443baa284738de7921e582d8ac3f24ab329e3aed0fce17f5"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T07:04:57.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "6c0cf7ce5c98d0672176641101a055eeda02a78300912abe17f9a5fcf69cbc96"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T07:08:57.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "64e0f5a836f55487a3cfa541ae227a193b71ea05ea1c2deac13766bb897a5526"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 8
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "wasm"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:13:26.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "dc978ba5c532db98803275aff9753ddd67f50447ef1d6d6fc376177a76a69810"
  .. .. ..$ status  : chr "success"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:11:19.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "c7cd21375d72552b9454fadbaf29284b1a39d294ffff468e78e8bcaa91c89aa6"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:10:26.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "2fd8f605344376b0afae5c5c3f2f2c02c64fbb7f84358a0519e00089417239b4"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.2"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "3.45.2"
  .. .. ..$ date    : chr "2025-11-12T04:10:57.000Z"
  .. .. ..$ commit  : chr "a293c4f1b41269208ce7e8d9ff5750ef5cf330e4"
  .. .. ..$ fileid  : chr "86d4688b85aef0f58aeeb544730323de5e8ca65f8f62c8732f1b3dfbacfc457e"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/19285513998"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>