<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>exploreBiocUniverse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="exploreBiocUniverse_files/libs/clipboard/clipboard.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/quarto.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/popper.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="exploreBiocUniverse_files/libs/quarto-html/anchor.min.js"></script>
<link href="exploreBiocUniverse_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="exploreBiocUniverse_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="exploreBiocUniverse_files/libs/bootstrap/bootstrap-8b443b48ee49e16bb685065ccc83795d.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="exploreBiocUniverse_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="exploreBiocUniverse_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">exploreBiocUniverse</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="comparing-bioconductor-packages-in-r-universe-and-the-bbs" class="level1">
<h1>Comparing Bioconductor packages in R-Universe and the BBS</h1>
<p>Skip to <a href="#packages-not-in-agreement">Packages not in agreements</a> or go to <a href="windows.qmd">windows</a>.</p>
<section id="r-universe" class="level2">
<h2 class="anchored" data-anchor-id="r-universe">R Universe</h2>
<p>We can use the <a href="https://github.com/ropensci/universe/">universe</a> package to explore the R-Universe (RU) API. (We might need to increase the <code>max.print</code> to show all packages.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"universe"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"universe"</span>, <span class="at">repos =</span> <span class="st">"https://ropensci.r-universe.dev"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universe)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">max.print =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data for the Bioc Universe (BU).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs_i <span class="ot">&lt;-</span> <span class="fu">universe_all_packages</span>(<span class="st">"bioc"</span>, <span class="at">limit =</span> <span class="dv">3000</span>L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What packages are in bioc universe?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ru_pkgs <span class="ot">&lt;-</span> <span class="fu">universe_ls</span>(<span class="st">"bioc"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bbs" class="level2">
<h2 class="anchored" data-anchor-id="bbs">BBS</h2>
<p>Get Bioconductor packages in BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocPkgTools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bbs_pkgs_i <span class="ot">&lt;-</span> BiocPkgTools<span class="sc">::</span><span class="fu">biocBuildReport</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">version =</span> BiocManager<span class="sc">::</span><span class="fu">version</span>(),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgType =</span> <span class="st">"software"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>bbs_pkgs <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unique</span>(bbs_pkgs_i<span class="sc">$</span>pkg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Packages in RU vs BBS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">RU =</span> <span class="fu">length</span>(ru_pkgs), <span class="at">BBS =</span> <span class="fu">length</span>(bbs_pkgs),</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"Total"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        RU  BBS
Total 2332 2361</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>not_in_ru <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(bbs_pkgs, ru_pkgs)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>not_in_bbs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(ru_pkgs, bbs_pkgs)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># There's probably a better way to do this</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">Created =</span> <span class="fu">character</span>(), <span class="at">Published =</span> <span class="fu">character</span>(),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">RemoteUrl =</span> <span class="fu">character</span>())</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> not_in_bbs) {</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  pi <span class="ot">&lt;-</span> <span class="fu">universe_one_package</span>(<span class="st">"bioc"</span>, pkg)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  pkgs_not_in_bbs <span class="ot">&lt;-</span> pkgs_not_in_bbs <span class="sc">|&gt;</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> pi<span class="sc">$</span>Package, <span class="at">Version =</span> pi<span class="sc">$</span>Version,</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">Created =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_created</span><span class="st">`</span>, <span class="at">Published =</span> pi<span class="sc">$</span><span class="st">`</span><span class="at">_published</span><span class="st">`</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">RemoteUrl =</span> pi<span class="sc">$</span>RemoteUrl)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-not-in-bbs-but-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-not-in-bbs-but-in-ru">Packages not in BBS but in RU</h3>
<p>Packages in RU but not in the BBS are likely added by maintainers via <a href="https://github.com/r-universe/bioc/blob/master/.remotes.json" class="uri">https://github.com/r-universe/bioc/blob/master/.remotes.json</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pkgs_not_in_bbs <span class="sc">|&gt;</span> knitr<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Created</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Published</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RemoteUrl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ggpattern</td>
<td style="text-align: left;">1.1.4</td>
<td style="text-align: left;">2025-10-12T07:14:19.000Z</td>
<td style="text-align: left;">2025-10-12T07:17:44.041Z</td>
<td style="text-align: left;">https://github.com/coolbutuseless/ggpattern</td>
</tr>
<tr class="even">
<td style="text-align: left;">lmms</td>
<td style="text-align: left;">1.3.3</td>
<td style="text-align: left;">2025-10-23T03:46:27.000Z</td>
<td style="text-align: left;">2025-10-23T03:52:01.356Z</td>
<td style="text-align: left;">https://github.com/cran/lmms</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tidydr</td>
<td style="text-align: left;">0.0.5.001</td>
<td style="text-align: left;">2025-08-23T03:15:52.000Z</td>
<td style="text-align: left;">2025-10-25T02:45:43.018Z</td>
<td style="text-align: left;">https://github.com/YuLab-SMU/tidydr</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
</section>
<section id="packages-in-bbs-but-not-in-ru" class="level3">
<h3 class="anchored" data-anchor-id="packages-in-bbs-but-not-in-ru">Packages in BBS but not in RU</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>not_in_ru</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "BiGGR"                  "ccmap"                  "CellScore"             
 [4] "CINdex"                 "DEP"                    "gpuMagic"              
 [7] "Harshlight"             "hiReadsProcessor"       "hypeR"                 
[10] "interactiveDisplay"     "interactiveDisplayBase" "lapmix"                
[13] "LinTInd"                "lute"                   "MADSEQ"                
[16] "netZooR"                "oppti"                  "PhenStat"              
[19] "qckitfastq"             "ReactomeGraph4R"        "Repitools"             
[22] "Rfastp"                 "rGADEM"                 "rRDP"                  
[25] "SARC"                   "scafari"                "seqArchR"              
[28] "seqTools"               "Streamer"               "TitanCNA"              
[31] "TransView"              "XNAString"             </code></pre>
</div>
</div>
<p>The following packages are not deprecated but are not in RU.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>depre_pkgs <span class="ot">&lt;-</span> <span class="fu">filter</span>(bbs_pkgs_i, Deprecated <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(pkg) <span class="sc">|&gt;</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(not_in_ru, depre_pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "scafari"</code></pre>
</div>
</div>
</section>
</section>
<section id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc." class="level2">
<h2 class="anchored" data-anchor-id="making-sense-of-r-universes-build-status-in-respect-to-r-cmd-build-check-etc.">Making sense of R Universe’s build status in respect to R CMD build, check, etc.</h2>
<p>Looking at the status of linux x86_64 binary builds. The build “status” appears to get “success”, “failure”, or “cancelled” from the GitHub Action overall. The “check” value corresponds to <code>R CMD check</code>. I interpret the result of “status” to be “success” OK and “failure” ERROR for <code>R CMD build</code> but it may mean something else. We’ll use CANCELLED as it is. There’s also a special status for documentation, which is handled separately from build/check, which may account for some discrepancies. Not sure if the build status for vignettes is available in the API.</p>
<p>Packages with my comment “Missing R4.5 on Linux” did have data in the API but no x86_64 linux with R 4.5.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Package =</span> <span class="fu">character</span>(), <span class="at">Version =</span> <span class="fu">character</span>(), </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Status =</span> <span class="fu">character</span>(), <span class="at">Build =</span> <span class="fu">character</span>(),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Check =</span> <span class="fu">character</span>(), <span class="at">Comment =</span> <span class="fu">character</span>())</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#, Buildurl = character(),</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># Commit = character())</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> ru_pkgs_i) {</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> <span class="cn">NA</span>, <span class="at">build =</span> <span class="cn">NA</span>, <span class="at">check =</span> <span class="cn">NA</span>, <span class="at">buildurl =</span> <span class="cn">NA</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  comment <span class="ot">=</span> <span class="st">""</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (b <span class="cf">in</span> p<span class="sc">$</span><span class="st">`</span><span class="at">_binaries</span><span class="st">`</span>) {</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (b<span class="sc">$</span>os <span class="sc">==</span> <span class="st">"linux"</span> <span class="sc">&amp;&amp;</span> b<span class="sc">$</span>r <span class="sc">==</span> <span class="st">"4.5.0"</span>) {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>      build <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">status =</span> b<span class="sc">$</span>status, <span class="at">check =</span> b<span class="sc">$</span>check, <span class="at">buildurl =</span> b<span class="sc">$</span>buildurl)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.na</span>(build[<span class="st">"status"</span>])) {</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> <span class="st">"Missing R4.5 on Linux"</span> </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"success"</span>) {</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"failure"</span>) {</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"ERROR*"</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (build[<span class="st">"status"</span>] <span class="sc">==</span> <span class="st">"cancelled"</span>) {</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    build[<span class="st">"build"</span>] <span class="ot">&lt;-</span> <span class="st">"CANCELLED"</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    comment <span class="ot">=</span> build[<span class="st">"buildurl"</span>]</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  ru_status <span class="ot">&lt;-</span> ru_status <span class="sc">|&gt;</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_row</span>(<span class="at">Package =</span> p<span class="sc">$</span>Package, <span class="at">Version =</span> p<span class="sc">$</span>Version, <span class="at">Status =</span> build[<span class="st">"status"</span>],</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">Build =</span> build[<span class="st">"build"</span>], <span class="at">Check =</span> build[<span class="st">"check"</span>], <span class="at">Comment =</span> comment)</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>            <span class="co">#, Buildurl = build["buildurl"], Commit = p$`_commit`$id)</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co"># order by Package</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>ru_status <span class="ot">&lt;-</span> ru_status[<span class="fu">order</span>(ru_status<span class="sc">$</span>Package), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>DEVEL_LINUX_BUILDER <span class="ot">=</span> <span class="st">"nebbiolo2"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper functions</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>get_result <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  entry <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pkg <span class="sc">==</span> x[<span class="st">"pkg"</span>], stage <span class="sc">==</span> x[<span class="st">"stage"</span>], node <span class="sc">==</span> x[<span class="st">"node"</span>])</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(entry)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    entry <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    entry<span class="sc">$</span>result</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>get_install <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"install"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>get_build <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"buildsrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>get_check <span class="ot">&lt;-</span> <span class="cf">function</span>(pkg) {</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_result</span>(<span class="fu">c</span>(<span class="at">pkg =</span> pkg, <span class="at">stage =</span> <span class="st">"checksrc"</span>, <span class="at">node =</span> DEVEL_LINUX_BUILDER))</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>bbs_status <span class="ot">&lt;-</span> bbs_pkgs_i <span class="sc">|&gt;</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(node <span class="sc">==</span> DEVEL_LINUX_BUILDER) <span class="sc">|&gt;</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version, Deprecated) <span class="sc">|&gt;</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(pkg, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">install =</span> <span class="fu">map_chr</span>(pkg, get_install)) <span class="sc">|&gt;</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">build =</span> <span class="fu">map_chr</span>(pkg, get_build)) <span class="sc">|&gt;</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">check =</span> <span class="fu">map_chr</span>(pkg, get_check))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-the-status-of-packages-in-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="compare-the-status-of-packages-in-ru-and-bbs">Compare the status of packages in RU and BBS</h3>
<p>We can only compare <code>check</code> and <code>build</code>, so we’ll check if those columns agree. Note that in the BBS, <code>check</code> can show “WARNINGS” whereas R Universe can show “WARNING” (no s) as well as “OK”, “NOTE”, “ERROR”, or NA.</p>
<p>Note: - Deprecated packages are removed immedately in R Universe. - Packages may have different versions since R-Universe builds new versions after detecting a “push”.</p>
<p>The following includes packages not deprecated in BBS where build and check don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ru_status) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pkg"</span>, <span class="st">"version"</span>, <span class="st">"status"</span>, <span class="st">"build"</span>, <span class="st">"check"</span>, <span class="st">"comment"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>bbs_ru_comparison <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> bbs_status, <span class="at">y =</span> ru_status, <span class="at">by =</span> <span class="st">"pkg"</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">&amp;</span> Deprecated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(pkg, install, build.x, build.y, check.x, check.y, version.x,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                  version.y, status, comment)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>, <span class="st">"RU_build"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"BBS_version"</span>, <span class="st">"RU_version"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="packages-without-x86_64-linux-builds-for-r-4.5" class="level4">
<h4 class="anchored" data-anchor-id="packages-without-x86_64-linux-builds-for-r-4.5">Packages without x86_64 Linux builds for R 4.5</h4>
<p>The following are packages in bioc R Universe but without an x86_64 Linux build for R 4.5.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Comment <span class="sc">==</span> <span class="st">"Missing R4.5 on Linux"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages in RU without an x86_64 linux build for R 4.5"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages in RU without an x86_64 linux build for R 4.5</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<p>Notes: - Retrying to refresh the build on the dashboard failed for some packages. How can we rebuild a package that hasn’t received a push in 3 months? - paxtoolsr should be in Bioc R Universe, but it isn’t. Why?</p>
</section>
</section>
</section>
<section id="packages-not-in-agreement" class="level2">
<h2 class="anchored" data-anchor-id="packages-not-in-agreement">Packages not in agreement</h2>
<p>Looking at packages with the same version and removing deprecated packages, we have the following packages where the BBS and R Universe’s results don’t agree.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="ot">&lt;-</span> bbs_ru_comparison <span class="sc">|&gt;</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(build.x <span class="sc">!=</span> build.y <span class="sc">|</span> <span class="sc">!</span>(check.x <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNINGS"</span>)) <span class="sc">|</span> <span class="sc">!</span>(check.y <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"OK"</span>, <span class="st">"WARNING"</span>, <span class="st">"NOTE"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Deprecated <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> version.x <span class="sc">==</span> version.y <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(status)) <span class="sc">|&gt;</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pkg, version.x, install, build.x, build.y, check.x, check.y,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         status, comment)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bbs_ru_not_ok2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Package"</span>, <span class="st">"BBS_version"</span>, <span class="st">"BBS_install"</span>, <span class="st">"BBS_build"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"RU_build"</span>, <span class="st">"BBS_check"</span>, <span class="st">"RU_check"</span>, <span class="st">"RU_status"</span>, <span class="st">"Comment"</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>bbs_ru_not_ok2 <span class="sc">|&gt;</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"Packages with different build results between BBS and RU"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"hover"</span>, <span class="st">"responsive"</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-hover caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Packages with different build results between BBS and RU</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Package</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_version</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_install</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_build</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">BBS_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_check</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RU_status</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Comment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">:-------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:-----------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:--------</td>
<td style="text-align: left;">:---------</td>
<td style="text-align: left;">:-------</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<p>Note: <code>ERROR*</code> could be an <code>ERROR</code>, <code>TIMEOUT</code>, or <code>skipped</code>.</p>
<section id="total-packages-not-in-agreement" class="level3">
<h3 class="anchored" data-anchor-id="total-packages-not-in-agreement">Total packages not in agreement</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bbs_ru_not_ok2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="other" class="level1">
<h1>Other</h1>
<p>If a package repo isn’t accessible or an error exists in the DESCRIPTION file, there will be an error in the registry for that submodule and it will show up at <a href="https://github.com/r-universe/bioc/actions/workflows/sync.yml" class="uri">https://github.com/r-universe/bioc/actions/workflows/sync.yml</a>. The red triangle on <a href="https://bioc.r-universe.dev/" class="uri">https://bioc.r-universe.dev/</a> will take you to the specific workflow error.</p>
<p>Currently, this information is available on the BBS but I wonder if this information is available somewhere for R Universe OR if it would be necessary given that developers would be able to inspect their builds sooner.</p>
<section id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api" class="level3">
<h3 class="anchored" data-anchor-id="important-use-the-dbdump-endpoint-with-read_bson-to-dump-the-api">Important: Use the dbdump endpoint with read_bson to dump the API</h3>
<p>Use <code>read_bson</code> to dump API data into a db</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("mongolite")</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mongolite)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>bioc <span class="ot">&lt;-</span> mongolite<span class="sc">::</span><span class="fu">read_bson</span>(<span class="st">'https://bioc.r-universe.dev/api/dbdump'</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: it returns 2268 records rather than 2276 (fewer retrieved through</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co"># another endpoint)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-a-tarball-from-ru-and-bbs" class="level3">
<h3 class="anchored" data-anchor-id="comparing-a-tarball-from-ru-and-bbs">Comparing a tarball from RU and BBS</h3>
<p>Lastly, when I look at the builds, they’re using various repositories, such p3m, r-universe, etc. This might be okay as long as the resulting binaries/tarballs are the generally the same. Looking at XVector for both, both the BBS and R Universe inject some additional information. R Universe also creates an <code>extra</code> directory that has citations, manuals, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2 ruXVector_0.47.2</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">diff</span> <span class="at">-r</span> bbsXVector_0.47.2/XVector/DESCRIPTION ruXVector_0.47.2/XVector/DESCRIPTION</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">5c5</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>   <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span>         <span class="st">"externally"</span> <span class="kw">(</span><span class="ex">e.g.</span> behind an R external pointer, or on disk<span class="kw">)</span><span class="bu">.</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="ex">28,33c28,31</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_url: <span class="ex">https://git.bioconductor.org/packages/XVector</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_branch: <span class="ex">devel</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit: <span class="ex">9f44218</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> git_last_commit_date: <span class="ex">2025-01-07</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Repository: <span class="ex">Bioconductor</span> 3.21</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Date/Publication: <span class="ex">2025-01-08</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Repository: <span class="ex">https://bioc.r-universe.dev</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteUrl: <span class="ex">https://github.com/bioc/XVector</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteRef: <span class="ex">HEAD</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> RemoteSha: <span class="ex">9f44218308f9d5867f0b9df5fae60d217f4e7d65</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="ex">35c33</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span> Packaged: <span class="ex">2025-01-09</span> 00:57:29 UTC<span class="kw">;</span> <span class="ex">biocbuild</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="ex">---</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> Packaged: <span class="ex">2025-01-08</span> 03:05:41 UTC<span class="kw">;</span> <span class="ex">root</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: extra</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="ex">Only</span> in ruXVector_0.47.2/XVector: manual.pdf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="structure-of-a-package-through-universe" class="level3">
<h3 class="anchored" data-anchor-id="structure-of-a-package-through-universe">Structure of a package through universe</h3>
<p>The structure of a package looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">str</span>(ru_pkgs_i[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ :List of 73
  ..$ Package           : chr "TENET"
  ..$ Title             : chr "R package for TENET (Tracing regulatory Element Networks using\nEpigenetic Traits) to identify key transcription factors"
  ..$ Description       : chr "TENET identifies key transcription factors (TFs) and\nregulatory elements (REs) linked to a specific cell type "| __truncated__
  ..$ Version           : chr "1.1.91"
  ..$ Authors@R         : chr "c(person(\"Rhie Lab at the University of Southern California\", email = \"rhielab@gmail.com\", role = \"cre\"),"| __truncated__
  ..$ URL               : chr "https://github.com/rhielab/TENET"
  ..$ BugReports        : chr "https://github.com/rhielab/TENET/issues"
  ..$ License           : chr "GPL-2"
  ..$ Encoding          : chr "UTF-8"
  ..$ Roxygen           : chr "list(markdown = TRUE)"
  ..$ RoxygenNote       : chr "7.3.3"
  ..$ BiocType          : chr "Software"
  ..$ biocViews         : chr "Software, BiomedicalInformatics, CellBiology, Genetics,\nEpigenetics, MultipleComparison, GeneExpression,\nDiff"| __truncated__
  ..$ VignetteBuilder   : chr "knitr"
  ..$ Config/pak/sysreqs: chr "cmake make libbz2-dev libicu-dev libjpeg-dev\nliblzma-dev libpng-dev libxml2-dev libssl-dev libx11-dev\nxz-utils zlib1g-dev"
  ..$ Repository        : chr "https://bioc.r-universe.dev"
  ..$ Date/Publication  : chr "2025-10-27 12:03:12 UTC"
  ..$ RemoteUrl         : chr "https://github.com/bioc/TENET"
  ..$ RemoteRef         : chr "HEAD"
  ..$ RemoteSha         : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  ..$ NeedsCompilation  : chr "no"
  ..$ Packaged          :List of 2
  .. ..$ Date: chr "2025-10-27 13:57:05 UTC"
  .. ..$ User: chr "root"
  ..$ Author            : chr "Rhie Lab at the University of Southern California [cre],\nDaniel Mullen [aut] (ORCID: &lt;https://orcid.org/0000-0"| __truncated__
  ..$ Maintainer        : chr "Rhie Lab at the University of Southern California &lt;rhielab@gmail.com&gt;"
  ..$ MD5sum            : chr "d6c69b47772eb0c75abfc2270d2cac75"
  ..$ _user             : chr "bioc"
  ..$ _type             : chr "src"
  ..$ _file             : chr "TENET_1.1.91.tar.gz"
  ..$ _fileid           : chr "f9bdf0297ac426afed5559d219277fbe427dc77c8f90f3a995c7e399988209a1"
  ..$ _filesize         : int 1948113
  ..$ _sha256           : chr "f9bdf0297ac426afed5559d219277fbe427dc77c8f90f3a995c7e399988209a1"
  ..$ _created          : chr "2025-10-27T13:57:05.000Z"
  ..$ _published        : chr "2025-10-27T14:13:24.513Z"
  ..$ _distro           : chr "noble"
  ..$ _jobs             :List of 9
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 535
  .. .. ..$ config  : chr "linux-devel-x86_64"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382235056"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 544
  .. .. ..$ config  : chr "linux-release-x86_64"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382236845"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 260
  .. .. ..$ config  : chr "macos-oldrel-arm64"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "FAIL"
  .. .. ..$ artifact: chr ""
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 489
  .. .. ..$ config  : chr "macos-release-arm64"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382308147"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 1314
  .. .. ..$ config  : chr "source"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382124105"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 278
  .. .. ..$ config  : chr "wasm-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382182848"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 690
  .. .. ..$ config  : chr "windows-devel"
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ artifact: chr "4382265501"
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 170
  .. .. ..$ config  : chr "windows-oldrel"
  .. .. ..$ r       : chr "4.4.3"
  .. .. ..$ check   : chr "FAIL"
  .. .. ..$ artifact: chr ""
  .. ..$ :List of 6
  .. .. ..$ job     : num 5.38e+10
  .. .. ..$ time    : int 486
  .. .. ..$ config  : chr "windows-release"
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ artifact: chr "4382223365"
  ..$ _buildurl         : chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  ..$ _status           : chr "success"
  ..$ _host             : chr "GitHub-Actions"
  ..$ _upstream         : chr "https://github.com/bioc/TENET"
  ..$ _commit           :List of 5
  .. ..$ id       : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. ..$ author   : chr "Ethan Nelson-Moore &lt;nelsonmo@usc.edu&gt;"
  .. ..$ committer: chr "Ethan Nelson-Moore &lt;nelsonmo@usc.edu&gt;"
  .. ..$ message  : chr "v1.1.91: Numerous documentation improvements and small fixes\n"
  .. ..$ time     : int 1761566592
  ..$ _maintainer       :List of 2
  .. ..$ name : chr "Rhie Lab at the University of Southern California"
  .. ..$ email: chr "rhielab@gmail.com"
  ..$ _registered       : logi TRUE
  ..$ _dependencies     :List of 36
  .. ..$ :List of 3
  .. .. ..$ package: chr "R"
  .. .. ..$ version: chr "&gt;= 4.4"
  .. .. ..$ role   : chr "Depends"
  .. ..$ :List of 2
  .. .. ..$ package: chr "graphics"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "grDevices"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "stats"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "tools"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "S4Vectors"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "GenomicRanges"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "IRanges"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "parallel"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "pastecs"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 3
  .. .. ..$ package: chr "ggplot2"
  .. .. ..$ version: chr "&gt;= 4.0"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "RCircos"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "survival"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BSgenome.Hsapiens.UCSC.hg38"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "seqLogo"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "Biostrings"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "matlab"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TCGAbiolinks"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "methods"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "R.utils"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "MultiAssayExperiment"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "SummarizedExperiment"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "sesame"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "sesameData"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "AnnotationHub"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "ExperimentHub"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TENET.ExperimentHub"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rtracklayer"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "MotifDb"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BAMMtools"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "survminer"
  .. .. ..$ role   : chr "Imports"
  .. ..$ :List of 2
  .. .. ..$ package: chr "TENET.AnnotationHub"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "knitr"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "rmarkdown"
  .. .. ..$ role   : chr "Suggests"
  .. ..$ :List of 2
  .. .. ..$ package: chr "BiocStyle"
  .. .. ..$ role   : chr "Suggests"
  ..$ _owner            : chr "bioc"
  ..$ _selfowned        : logi TRUE
  ..$ _usedby           : int 0
  ..$ _updates          :List of 10
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-03"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-07"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-09"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-11"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-14"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-15"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-16"
  .. .. ..$ n   : int 2
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-19"
  .. .. ..$ n   : int 3
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-22"
  .. .. ..$ n   : int 1
  .. ..$ :List of 2
  .. .. ..$ week: chr "2025-44"
  .. .. ..$ n   : int 1
  ..$ _tags             : list()
  ..$ _bioc             :List of 2
  .. ..$ :List of 3
  .. .. ..$ branch : chr "devel"
  .. .. ..$ version: chr "1.1.90"
  .. .. ..$ bioc   : chr "3.22"
  .. ..$ :List of 3
  .. .. ..$ branch : chr "release"
  .. .. ..$ version: chr "1.0.1"
  .. .. ..$ bioc   : chr "3.21"
  ..$ _topics           :List of 22
  .. ..$ : chr "software"
  .. ..$ : chr "biomedicalinformatics"
  .. ..$ : chr "cellbiology"
  .. ..$ : chr "genetics"
  .. ..$ : chr "epigenetics"
  .. ..$ : chr "multiplecomparison"
  .. ..$ : chr "geneexpression"
  .. ..$ : chr "differentialexpression"
  .. ..$ : chr "dnamethylation"
  .. ..$ : chr "differentialmethylation"
  .. ..$ : chr "methylationarray"
  .. ..$ : chr "sequencing"
  .. ..$ : chr "methylseq"
  .. ..$ : chr "rnaseq"
  .. ..$ : chr "functionalgenomics"
  .. ..$ : chr "generegulation"
  .. ..$ : chr "genetarget"
  .. ..$ : chr "histonemodification"
  .. ..$ : chr "transcription"
  .. ..$ : chr "transcriptomics"
  .. ..$ : chr "survival"
  .. ..$ : chr "visualization"
  ..$ _stars            : int 1
  ..$ _contributors     :List of 2
  .. ..$ :List of 3
  .. .. ..$ user : chr "ethannelson-moore"
  .. .. ..$ count: int 12
  .. .. ..$ uuid : int 87036004
  .. ..$ :List of 3
  .. .. ..$ user : chr "danieljmullen"
  .. .. ..$ count: int 2
  .. .. ..$ uuid : int 26636973
  ..$ _userbio          :List of 4
  .. ..$ uuid       : int 2286807
  .. ..$ type       : chr "organization"
  .. ..$ name       : chr "Bioconductor"
  .. ..$ description: chr "Software for the analysis and comprehension of high-throughput genomic data"
  ..$ _downloads        :List of 2
  .. ..$ count : int 224
  .. ..$ source: chr "https://www.bioconductor.org/packages/stats/bioc/TENET"
  ..$ _devurl           : chr "https://github.com/rhielab/tenet"
  ..$ _searchresults    : int 19
  ..$ _rbuild           : chr "4.5.1"
  ..$ _assets           :List of 11
  .. ..$ : chr "extra/citation.cff"
  .. ..$ : chr "extra/citation.html"
  .. ..$ : chr "extra/citation.json"
  .. ..$ : chr "extra/citation.txt"
  .. ..$ : chr "extra/contents.json"
  .. ..$ : chr "extra/NEWS.html"
  .. ..$ : chr "extra/NEWS.txt"
  .. ..$ : chr "extra/readme.html"
  .. ..$ : chr "extra/readme.md"
  .. ..$ : chr "extra/TENET.html"
  .. ..$ : chr "manual.pdf"
  ..$ _homeurl          : chr "https://github.com/rhielab/tenet"
  ..$ _realowner        : chr "bioc"
  ..$ _cranurl          : logi FALSE
  ..$ _exports          :List of 23
  .. ..$ : chr "easyTENET"
  .. ..$ : chr "step1MakeExternalDatasets"
  .. ..$ : chr "step2GetDifferentiallyMethylatedSites"
  .. ..$ : chr "step3GetAnalysisZScores"
  .. ..$ : chr "step4SelectMostSignificantLinksPerDNAMethylationSite"
  .. ..$ : chr "step5OptimizeLinks"
  .. ..$ : chr "step6DNAMethylationSitesPerGeneTabulation"
  .. ..$ : chr "step7ExpressionVsDNAMethylationScatterplots"
  .. ..$ : chr "step7LinkedDNAMethylationSiteCountHistograms"
  .. ..$ : chr "step7LinkedDNAMethylationSitesMotifSearching"
  .. ..$ : chr "step7SelectedDNAMethylationSitesCaseVsControlBoxplots"
  .. ..$ : chr "step7StatesForLinks"
  .. ..$ : chr "step7TopGenesCaseVsControlExpressionBoxplots"
  .. ..$ : chr "step7TopGenesCircosPlots"
  .. ..$ : chr "step7TopGenesDNAMethylationHeatmaps"
  .. ..$ : chr "step7TopGenesExpressionCorrelationHeatmaps"
  .. ..$ : chr "step7TopGenesOverlappingLinkedDNAMethylationSitesHeatmaps"
  .. ..$ : chr "step7TopGenesSurvival"
  .. ..$ : chr "step7TopGenesTADTables"
  .. ..$ : chr "step7TopGenesUCSCBedFiles"
  .. ..$ : chr "step7TopGenesUserPeakOverlap"
  .. ..$ : chr "TCGADownloader"
  .. ..$ : chr "TENETCacheAllData"
  ..$ _datasets         :List of 2
  .. ..$ :List of 9
  .. .. ..$ name  : chr "humanTranscriptionFactorDb"
  .. .. ..$ title : chr "Human transcription factor database"
  .. .. ..$ object: chr "humanTranscriptionFactorDb"
  .. .. ..$ file  : chr "humanTranscriptionFactorDb.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "data.frame"
  .. .. ..$ fields:List of 28
  .. .. .. ..$ : chr "Ensembl.ID"
  .. .. .. ..$ : chr "HGNC.symbol"
  .. .. .. ..$ : chr "DBD"
  .. .. .. ..$ : chr "Is.TF."
  .. .. .. ..$ : chr "TF.assessment"
  .. .. .. ..$ : chr "Binding.mode"
  .. .. .. ..$ : chr "Motif.status"
  .. .. .. ..$ : chr "Final.Notes"
  .. .. .. ..$ : chr "Final.Comments"
  .. .. .. ..$ : chr "Interpro.ID.s."
  .. .. .. ..$ : chr "EntrezGene.ID"
  .. .. .. ..$ : chr "EntrezGene.Description"
  .. .. .. ..$ : chr "PDB.ID"
  .. .. .. ..$ : chr "TF.tested.by.HT.SELEX."
  .. .. .. ..$ : chr "TF.tested.by.PBM."
  .. .. .. ..$ : chr "Conditional.Binding.Requirements"
  .. .. .. ..$ : chr "Original.Comments"
  .. .. .. ..$ : chr "Vaquerizas.2009.classification"
  .. .. .. ..$ : chr "CisBP.considers.it.a.TF."
  .. .. .. ..$ : chr "TFCat.classification"
  .. .. .. ..$ : chr "Is.a.GO.TF."
  .. .. .. ..$ : chr "Initial.assessment"
  .. .. .. ..$ : chr "Curator.1"
  .. .. .. ..$ : chr "Curator.2"
  .. .. .. ..$ : chr "TFclass.considers.it.a.TF."
  .. .. .. ..$ : chr "Go.Evidence"
  .. .. .. ..$ : chr "Pfam.Domains..By.ENSP.ID."
  .. .. .. ..$ : chr "Is.C2H2.ZF.KRAB.."
  .. .. ..$ rows  : int 2765
  .. .. ..$ table : logi TRUE
  .. .. ..$ tojson: logi TRUE
  .. ..$ :List of 8
  .. .. ..$ name  : chr "humanTranscriptionFactorList"
  .. .. ..$ title : chr "Human transcription factor list"
  .. .. ..$ object: chr "humanTranscriptionFactorList"
  .. .. ..$ file  : chr "humanTranscriptionFactorList.rda"
  .. .. ..$ class :List of 1
  .. .. .. ..$ : chr "character"
  .. .. ..$ fields: list()
  .. .. ..$ table : logi FALSE
  .. .. ..$ tojson: logi TRUE
  ..$ _help             :List of 25
  .. ..$ :List of 3
  .. .. ..$ page  : chr "easyTENET"
  .. .. ..$ title : chr "Run the step 1 through step 6 functions with default arguments"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "easyTENET"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "humanTranscriptionFactorDb"
  .. .. ..$ title : chr "Human transcription factor database"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "humanTranscriptionFactorDb"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "humanTranscriptionFactorList"
  .. .. ..$ title : chr "Human transcription factor list"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "humanTranscriptionFactorList"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step1MakeExternalDatasets"
  .. .. ..$ title : chr "Create a GRanges object representing putative regulatory element regions, based on the data sources selected fo"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step1MakeExternalDatasets"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step2GetDifferentiallyMethylatedSites"
  .. .. ..$ title : chr "Identify differentially methylated RE DNA methylation sites"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step2GetDifferentiallyMethylatedSites"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step3GetAnalysisZScores"
  .. .. ..$ title : chr "Calculate Z-scores comparing the mean expression of each gene in the case samples that are hyper- and/or hypome"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step3GetAnalysisZScores"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step4SelectMostSignificantLinksPerDNAMethylationSite"
  .. .. ..$ title : chr "Select the most significant RE DNA methylation site-gene links to each RE DNA methylation site"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step4SelectMostSignificantLinksPerDNAMethylationSite"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step5OptimizeLinks"
  .. .. ..$ title : chr "Find final RE DNA methylation site-gene links using various optimization metrics"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step5OptimizeLinks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step6DNAMethylationSitesPerGeneTabulation"
  .. .. ..$ title : chr "Tabulate the total number of RE DNA methylation sites linked to each gene"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step6DNAMethylationSitesPerGeneTabulation"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7ExpressionVsDNAMethylationScatterplots"
  .. .. ..$ title : chr "Create scatterplots displaying the expression of the top genes and the methylation levels of each of their link"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7ExpressionVsDNAMethylationScatterplots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7LinkedDNAMethylationSiteCountHistograms"
  .. .. ..$ title : chr "Create histograms displaying the number of total genes and transcription factor genes linked to a given number "| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7LinkedDNAMethylationSiteCountHistograms"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7LinkedDNAMethylationSitesMotifSearching"
  .. .. ..$ title : chr "Search for transcription factor motifs in the vicinity of DNA methylation sites and/or within custom regions de"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7LinkedDNAMethylationSitesMotifSearching"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7SelectedDNAMethylationSitesCaseVsControlBoxplots"
  .. .. ..$ title : chr "Generate boxplots or violin plots comparing the methylation level of the specified RE DNA methylation sites in "| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7SelectedDNAMethylationSitesCaseVsControlBoxplots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7StatesForLinks"
  .. .. ..$ title : chr "Identify which of the case samples harbor each of the identified regulatory element DNA methylation site-gene links"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7StatesForLinks"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesCaseVsControlExpressionBoxplots"
  .. .. ..$ title : chr "Generate boxplots or violin plots comparing the expression level of the top genes and transcription factors in "| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesCaseVsControlExpressionBoxplots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesCircosPlots"
  .. .. ..$ title : chr "Generate Circos plots displaying the links between the top identified genes and each of the RE DNA methylation "| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesCircosPlots"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesDNAMethylationHeatmaps"
  .. .. ..$ title : chr "Generate heatmaps displaying the methylation level of all RE DNA methylation sites linked to the top genes and "| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesDNAMethylationHeatmaps"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesExpressionCorrelationHeatmaps"
  .. .. ..$ title : chr "Generate mirrored heatmaps displaying the correlation of the expression values of the top genes and TFs"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesExpressionCorrelationHeatmaps"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesOverlappingLinkedDNAMethylationSitesHeatmaps"
  .. .. ..$ title : chr "Generate binary heatmaps displaying which of the top genes and transcription factors share links with each of t"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesOverlappingLinkedDNAMethylationSitesHeatmaps"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesSurvival"
  .. .. ..$ title : chr "Perform Kaplan-Meier and Cox regression analyses to assess the association of top gene expression and linked RE"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesSurvival"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesTADTables"
  .. .. ..$ title : chr "Create tables using user-supplied topologically associating domain (TAD) information which identify the TADs co"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesTADTables"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesUCSCBedFiles"
  .. .. ..$ title : chr "Create BED-formatted interact files which can be loaded on the UCSC Genome Browser to display links between top"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesUCSCBedFiles"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "step7TopGenesUserPeakOverlap"
  .. .. ..$ title : chr "Identify if RE DNA methylation sites linked to top genes and transcription factors are located within a specifi"| __truncated__
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "step7TopGenesUserPeakOverlap"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TCGADownloader"
  .. .. ..$ title : chr "Download TCGA gene expression, DNA methylation, and clinical datasets and compile them into a MultiAssayExperiment object"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TCGADownloader"
  .. ..$ :List of 3
  .. .. ..$ page  : chr "TENETCacheAllData"
  .. .. ..$ title : chr "Cache all online datasets required by TENET examples and optional features"
  .. .. ..$ topics:List of 1
  .. .. .. ..$ : chr "TENETCacheAllData"
  ..$ _readme           : chr "https://github.com/bioc/TENET/raw/HEAD/README.md"
  ..$ _rundeps          :List of 195
  .. ..$ : chr "abind"
  .. ..$ : chr "AnnotationDbi"
  .. ..$ : chr "AnnotationHub"
  .. ..$ : chr "ape"
  .. ..$ : chr "askpass"
  .. ..$ : chr "backports"
  .. ..$ : chr "BAMMtools"
  .. ..$ : chr "BH"
  .. ..$ : chr "Biobase"
  .. ..$ : chr "BiocBaseUtils"
  .. ..$ : chr "BiocFileCache"
  .. ..$ : chr "BiocGenerics"
  .. ..$ : chr "BiocIO"
  .. ..$ : chr "BiocManager"
  .. ..$ : chr "BiocParallel"
  .. ..$ : chr "BiocVersion"
  .. ..$ : chr "biomaRt"
  .. ..$ : chr "Biostrings"
  .. ..$ : chr "bit"
  .. ..$ : chr "bit64"
  .. ..$ : chr "bitops"
  .. ..$ : chr "blob"
  .. ..$ : chr "boot"
  .. ..$ : chr "broom"
  .. ..$ : chr "BSgenome"
  .. ..$ : chr "BSgenome.Hsapiens.UCSC.hg38"
  .. ..$ : chr "cachem"
  .. ..$ : chr "car"
  .. ..$ : chr "carData"
  .. ..$ : chr "caTools"
  .. ..$ : chr "cli"
  .. ..$ : chr "clipr"
  .. ..$ : chr "codetools"
  .. ..$ : chr "colorspace"
  .. ..$ : chr "commonmark"
  .. ..$ : chr "corrplot"
  .. ..$ : chr "cowplot"
  .. ..$ : chr "cpp11"
  .. ..$ : chr "crayon"
  .. ..$ : chr "curl"
  .. ..$ : chr "data.table"
  .. ..$ : chr "DBI"
  .. ..$ : chr "dbplyr"
  .. ..$ : chr "DelayedArray"
  .. ..$ : chr "Deriv"
  .. ..$ : chr "digest"
  .. ..$ : chr "doBy"
  .. ..$ : chr "downloader"
  .. ..$ : chr "dplyr"
  .. ..$ : chr "evaluate"
  .. ..$ : chr "exactRankTests"
  .. ..$ : chr "ExperimentHub"
  .. ..$ : chr "farver"
  .. ..$ : chr "fastmap"
  .. ..$ : chr "filelock"
  .. ..$ : chr "formatR"
  .. ..$ : chr "Formula"
  .. ..$ : chr "futile.logger"
  .. ..$ : chr "futile.options"
  .. ..$ : chr "generics"
  .. ..$ : chr "GenomeInfoDb"
  .. ..$ : chr "GenomicAlignments"
  .. ..$ : chr "GenomicRanges"
  .. ..$ : chr "ggplot2"
  .. ..$ : chr "ggpubr"
  .. ..$ : chr "ggrepel"
  .. ..$ : chr "ggsci"
  .. ..$ : chr "ggsignif"
  .. ..$ : chr "ggtext"
  .. ..$ : chr "glue"
  .. ..$ : chr "gplots"
  .. ..$ : chr "gridExtra"
  .. ..$ : chr "gridtext"
  .. ..$ : chr "gtable"
  .. ..$ : chr "gtools"
  .. ..$ : chr "highr"
  .. ..$ : chr "hms"
  .. ..$ : chr "httr"
  .. ..$ : chr "httr2"
  .. ..$ : chr "IRanges"
  .. ..$ : chr "isoband"
  .. ..$ : chr "jpeg"
  .. ..$ : chr "jsonlite"
  .. ..$ : chr "KEGGREST"
  .. ..$ : chr "KernSmooth"
  .. ..$ : chr "km.ci"
  .. ..$ : chr "KMsurv"
  .. ..$ : chr "knitr"
  .. ..$ : chr "labeling"
  .. ..$ : chr "lambda.r"
  .. ..$ : chr "lattice"
  .. ..$ : chr "lifecycle"
  .. ..$ : chr "litedown"
  .. ..$ : chr "lme4"
  .. ..$ : chr "magrittr"
  .. ..$ : chr "markdown"
  .. ..$ : chr "MASS"
  .. ..$ : chr "matlab"
  .. ..$ : chr "Matrix"
  .. .. [list output truncated]
  ..$ _vignettes        :List of 1
  .. ..$ :List of 9
  .. .. ..$ source  : chr "TENET_vignette.Rmd"
  .. .. ..$ filename: chr "TENET_vignette.html"
  .. .. ..$ title   : chr "Using TENET"
  .. .. ..$ author  : chr "Rhie Lab at the University of Southern California"
  .. .. ..$ engine  : chr "knitr::rmarkdown"
  .. .. ..$ headings:List of 39
  .. .. .. ..$ : chr "Introduction"
  .. .. .. ..$ : chr "Acquiring and installing TENET and associated packages"
  .. .. .. ..$ : chr "Loading TENET"
  .. .. .. ..$ : chr "Running TENET without internet access"
  .. .. .. ..$ : chr "Input data"
  .. .. .. ..$ : chr "Expression SummarizedExperiment object"
  .. .. .. ..$ : chr "Methylation SummarizedExperiment object"
  .. .. .. ..$ : chr "MultiAssayExperiment colData object"
  .. .. .. ..$ : chr "MultiAssayExperiment sampleMap object"
  .. .. .. ..$ : chr "Overview of main TENET functions"
  .. .. .. ..$ : chr "easyTENET: Run the step 1 through step 6 functions with default arguments"
  .. .. .. ..$ : chr "step1MakeExternalDatasets: Create a GRanges object representing putative regulatory element regions, based on t"| __truncated__
  .. .. .. ..$ : chr "step2GetDifferentiallyMethylatedSites: Identify differentially methylated RE DNA methylation sites"
  .. .. .. ..$ : chr "step3GetAnalysisZScores: Calculate Z-scores comparing the mean expression of each gene in the case samples that"| __truncated__
  .. .. .. ..$ : chr "step4SelectMostSignificantLinksPerDNAMethylationSite: Select the most significant RE DNA methylation site-gene "| __truncated__
  .. .. .. ..$ : chr "step5OptimizeLinks: Find final RE DNA methylation site-gene links using various optimization metrics"
  .. .. .. ..$ : chr "step6DNAMethylationSitesPerGeneTabulation: Tabulate the total number of RE DNA methylation sites linked to each gene"
  .. .. .. ..$ : chr "TCGADownloader: Download TCGA gene expression, DNA methylation, and clinical datasets and compile them into a M"| __truncated__
  .. .. .. ..$ : chr "TENETCacheAllData: Cache all online datasets required by TENET examples and optional features"
  .. .. .. ..$ : chr "Overview of downstream step 7 functions"
  .. .. .. ..$ : chr "step7ExpressionVsDNAMethylationScatterplots: Create scatterplots displaying the expression of the top genes and"| __truncated__
  .. .. .. ..$ : chr "step7LinkedDNAMethylationSiteCountHistograms: Create histograms displaying the number of total genes and transc"| __truncated__
  .. .. .. ..$ : chr "step7LinkedDNAMethylationSitesMotifSearching: Search for transcription factor motifs in the vicinity of DNA met"| __truncated__
  .. .. .. ..$ : chr "step7SelectedDNAMethylationSitesCaseVsControlBoxplots: Generate boxplots or violin plots comparing the methylat"| __truncated__
  .. .. .. ..$ : chr "step7StatesForLinks: Identify which of the case samples harbor each of the identified regulatory element DNA me"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesCaseVsControlExpressionBoxplots: Generate boxplots or violin plots comparing the expression level "| __truncated__
  .. .. .. ..$ : chr "step7TopGenesCircosPlots: Generate Circos plots displaying the links between top identified genes and each of t"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesExpressionCorrelationHeatmaps: Generate mirrored heatmaps displaying the correlation of the expres"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesDNAMethylationHeatmaps: Generate heatmaps displaying the methylation level of all RE DNA methylati"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesOverlappingLinkedDNAMethylationSitesHeatmaps: Generate binary heatmaps displaying which of the top"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesSurvival: Perform Kaplan-Meier and Cox regression analyses to assess the association of top gene e"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesTADTables: Create tables using user-supplied topologically associating domain (TAD) information wh"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesUCSCBedFiles: Create BED-formatted interact files which can be loaded on the UCSC Genome Browser t"| __truncated__
  .. .. .. ..$ : chr "step7TopGenesUserPeakOverlap: Identify if RE DNA methylation sites linked to top genes and transcription factor"| __truncated__
  .. .. .. ..$ : chr "Datasets included in the TENET package"
  .. .. .. ..$ : chr "humanTranscriptionFactorList: Human transcription factor list"
  .. .. .. ..$ : chr "humanTranscriptionFactorDb: Human transcription factor database"
  .. .. .. ..$ : chr "Acknowledgements"
  .. .. .. ..$ : chr "Session info"
  .. .. ..$ created : chr "2024-10-15 22:54:53"
  .. .. ..$ modified: chr "2025-10-27 12:03:12"
  .. .. ..$ commits : int 9
  ..$ _score            : num 4.98
  ..$ _indexed          : logi TRUE
  ..$ _nocasepkg        : chr "tenet"
  ..$ _universes        :List of 2
  .. ..$ : chr "bioc"
  .. ..$ : chr "rhielab"
  ..$ _previous         : chr "1.1.90"
  ..$ _binaries         :List of 6
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:02:18.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "114547b50d092d22d78c3b46a537e51a5f804525e72b6270fc0b0964f068bc1e"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  .. ..$ :List of 10
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "linux"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:02:25.000Z"
  .. .. ..$ distro  : chr "noble"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "a7762abf77a072e74a7eb75455f8ff087ed9ebdbef4a1630429cd25bbe351211"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "mac"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:07:18.000Z"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "73ed83fb678ca1143d9771450c0b27d1ccd1cb9384647c3c4735d367324f22ea"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  .. ..$ :List of 8
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "wasm"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:02:43.000Z"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "ca7363dd929802994fa15115f92c4baa59799cfd9eb04398b3c1209160204935"
  .. .. ..$ status  : chr "success"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.6.0"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:05:29.000Z"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "e7ea25c51f8620cde7ee71f7555362f292a280b84d66ad2c4a68a9ffde204615"
  .. .. ..$ status  : chr "failure"
  .. .. ..$ check   : chr "ERROR"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"
  .. ..$ :List of 9
  .. .. ..$ r       : chr "4.5.1"
  .. .. ..$ os      : chr "win"
  .. .. ..$ version : chr "1.1.91"
  .. .. ..$ date    : chr "2025-10-27T14:00:46.000Z"
  .. .. ..$ commit  : chr "7417d0dbb1630c3474e555ff1ab0e46eb8982b75"
  .. .. ..$ fileid  : chr "7562ad0ae868913ca08377bf86b2b1045774c275e3cc7090e5d26f3f842153c3"
  .. .. ..$ status  : chr "success"
  .. .. ..$ check   : chr "OK"
  .. .. ..$ buildurl: chr "https://github.com/r-universe/bioc/actions/runs/18842861945"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>